{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "06cf2fc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8824abec",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Desktop/Covered_regions.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL6_annotated_indel.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL6_annotated_indel.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8b7b0563",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "136948bd",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:127:138:138:100:38:27.54%:1.8855E-13:60:45...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:114:114:0:114:100%:4.392E-68:0:59:0:0:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:56:64:64:47:17:26.56%:2.243E-6:55:58:33:14...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686266</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:85:79:79:54:25:31.65%:3.1101E-9:55:54:43:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:58:11:11:0:11:100%:1.4176E-6:0:63:0:0:10:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36647</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9528488</td>\n",
       "      <td>rs1179775265</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:130:186:186:146:40:21.51%:8.6616E-14:52:49...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36648</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13360045</td>\n",
       "      <td>rs2032674</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:54:54:0:54:100%:4.0229E-32:0:55:0:0:35:19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36649</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14941891</td>\n",
       "      <td>rs17269816</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:182:32:32:0:32:100%:5.4567E-19:0:62:0:0:28:4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36650</th>\n",
       "      <td>chrY</td>\n",
       "      <td>15174113</td>\n",
       "      <td>rs17307398</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:164:29:29:0:29:100%:3.3259E-17:0:58:0:0:24:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:123:22:22:0:22:100%:4.7526E-13:0:56:0:0:14:8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36652 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270   rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511     rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290             .   A   C    .   PASS   \n",
       "3      chr1    686266             .   A   C    .   PASS   \n",
       "4      chr1    924533   rs112703963   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "36647  chrY   9528488  rs1179775265   T   G    .   PASS   \n",
       "36648  chrY  13360045     rs2032674   T   C    .   PASS   \n",
       "36649  chrY  14941891    rs17269816   C   T    .   PASS   \n",
       "36650  chrY  15174113    rs17307398   T   C    .   PASS   \n",
       "36651  chrY  20577481     rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "3      ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...   \n",
       "4      ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....   \n",
       "...                                                  ...   \n",
       "36647  ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...   \n",
       "36648  ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....   \n",
       "36649  ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...   \n",
       "36650  ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....   \n",
       "\n",
       "                                                  FORMAT  \\\n",
       "0      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4      GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                  ...   \n",
       "36647  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "36648  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "36649  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "36650  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "36651  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                  SAMPLE  \n",
       "0      0/1:127:138:138:100:38:27.54%:1.8855E-13:60:45...  \n",
       "1      1/1:255:114:114:0:114:100%:4.392E-68:0:59:0:0:...  \n",
       "2      0/1:56:64:64:47:17:26.56%:2.243E-6:55:58:33:14...  \n",
       "3      0/1:85:79:79:54:25:31.65%:3.1101E-9:55:54:43:1...  \n",
       "4         1/1:58:11:11:0:11:100%:1.4176E-6:0:63:0:0:10:1  \n",
       "...                                                  ...  \n",
       "36647  0/1:130:186:186:146:40:21.51%:8.6616E-14:52:49...  \n",
       "36648  1/1:255:54:54:0:54:100%:4.0229E-32:0:55:0:0:35:19  \n",
       "36649   1/1:182:32:32:0:32:100%:5.4567E-19:0:62:0:0:28:4  \n",
       "36650   1/1:164:29:29:0:29:100%:3.3259E-17:0:58:0:0:24:5  \n",
       "36651   1/1:123:22:22:0:22:100%:4.7526E-13:0:56:0:0:14:8  \n",
       "\n",
       "[36652 rows x 10 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Desktop/output_filtered_recently.vcf', comment='#', sep=\"\\t\", header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "9d4dd857",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686266</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>54</td>\n",
       "      <td>25</td>\n",
       "      <td>31.65%</td>\n",
       "      <td>3.1101E-9</td>\n",
       "      <td>43</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36647</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9528488</td>\n",
       "      <td>rs1179775265</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>130</td>\n",
       "      <td>186</td>\n",
       "      <td>186</td>\n",
       "      <td>146</td>\n",
       "      <td>40</td>\n",
       "      <td>21.51%</td>\n",
       "      <td>8.6616E-14</td>\n",
       "      <td>77</td>\n",
       "      <td>69</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36648</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13360045</td>\n",
       "      <td>rs2032674</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>54</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.0229E-32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36649</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14941891</td>\n",
       "      <td>rs17269816</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>182</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4567E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36650</th>\n",
       "      <td>chrY</td>\n",
       "      <td>15174113</td>\n",
       "      <td>rs17307398</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>164</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.3259E-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36652 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270   rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511     rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290             .   A   C    .   PASS   \n",
       "3      chr1    686266             .   A   C    .   PASS   \n",
       "4      chr1    924533   rs112703963   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "36647  chrY   9528488  rs1179775265   T   G    .   PASS   \n",
       "36648  chrY  13360045     rs2032674   T   C    .   PASS   \n",
       "36649  chrY  14941891    rs17269816   C   T    .   PASS   \n",
       "36650  chrY  15174113    rs17307398   T   C    .   PASS   \n",
       "36651  chrY  20577481     rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "3      ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   85   79   79   \n",
       "4      ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  1/1   58   11   11   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36647  ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  130  186  186   \n",
       "36648  ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....  1/1  255   54   54   \n",
       "36649  ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...  1/1  182   32   32   \n",
       "36650  ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...  1/1  164   29   29   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR  \n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9  \n",
       "1        0  114    100%   4.392E-68   0   0  98  16  \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2  \n",
       "3       54   25  31.65%   3.1101E-9  43  11  23   2  \n",
       "4        0   11    100%   1.4176E-6   0   0  10   1  \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  \n",
       "36647  146   40  21.51%  8.6616E-14  77  69  17  23  \n",
       "36648    0   54    100%  4.0229E-32   0   0  35  19  \n",
       "36649    0   32    100%  5.4567E-19   0   0  28   4  \n",
       "36650    0   29    100%  3.3259E-17   0   0  24   5  \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8  \n",
       "\n",
       "[36652 rows x 20 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "5b176468",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686266</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>54</td>\n",
       "      <td>25</td>\n",
       "      <td>31.65%</td>\n",
       "      <td>3.1101E-9</td>\n",
       "      <td>43</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36647</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9528488</td>\n",
       "      <td>rs1179775265</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>130</td>\n",
       "      <td>186</td>\n",
       "      <td>186</td>\n",
       "      <td>146</td>\n",
       "      <td>40</td>\n",
       "      <td>21.51%</td>\n",
       "      <td>8.6616E-14</td>\n",
       "      <td>77</td>\n",
       "      <td>69</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36648</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13360045</td>\n",
       "      <td>rs2032674</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>54</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.0229E-32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36649</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14941891</td>\n",
       "      <td>rs17269816</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>182</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4567E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36650</th>\n",
       "      <td>chrY</td>\n",
       "      <td>15174113</td>\n",
       "      <td>rs17307398</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>164</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.3259E-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36652 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270   rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511     rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290             .   A   C    .   PASS   \n",
       "3      chr1    686266             .   A   C    .   PASS   \n",
       "4      chr1    924533   rs112703963   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "36647  chrY   9528488  rs1179775265   T   G    .   PASS   \n",
       "36648  chrY  13360045     rs2032674   T   C    .   PASS   \n",
       "36649  chrY  14941891    rs17269816   C   T    .   PASS   \n",
       "36650  chrY  15174113    rs17307398   T   C    .   PASS   \n",
       "36651  chrY  20577481     rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "3      ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   85   79   79   \n",
       "4      ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  1/1   58   11   11   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36647  ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  130  186  186   \n",
       "36648  ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....  1/1  255   54   54   \n",
       "36649  ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...  1/1  182   32   32   \n",
       "36650  ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...  1/1  164   29   29   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous  \n",
       "1        0  114    100%   4.392E-68   0   0  98  16   0   1    Homozygous  \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous  \n",
       "3       54   25  31.65%   3.1101E-9  43  11  23   2   1   0  Heterozygous  \n",
       "4        0   11    100%   1.4176E-6   0   0  10   1   0   1    Homozygous  \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...  \n",
       "36647  146   40  21.51%  8.6616E-14  77  69  17  23   1   0  Heterozygous  \n",
       "36648    0   54    100%  4.0229E-32   0   0  35  19   0   1    Homozygous  \n",
       "36649    0   32    100%  5.4567E-19   0   0  28   4   0   1    Homozygous  \n",
       "36650    0   29    100%  3.3259E-17   0   0  24   5   0   1    Homozygous  \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous  \n",
       "\n",
       "[36652 rows x 23 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "7043eef8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686266</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>54</td>\n",
       "      <td>25</td>\n",
       "      <td>31.65%</td>\n",
       "      <td>3.1101E-9</td>\n",
       "      <td>43</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>LOC107985728:107985728|SAMD11:148398</td>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36647</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9528488</td>\n",
       "      <td>rs1179775265</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>130</td>\n",
       "      <td>186</td>\n",
       "      <td>186</td>\n",
       "      <td>146</td>\n",
       "      <td>40</td>\n",
       "      <td>21.51%</td>\n",
       "      <td>8.6616E-14</td>\n",
       "      <td>77</td>\n",
       "      <td>69</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY10:100289087</td>\n",
       "      <td>TSPY10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36648</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13360045</td>\n",
       "      <td>rs2032674</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>54</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.0229E-32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>UTY:7404</td>\n",
       "      <td>UTY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36649</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14941891</td>\n",
       "      <td>rs17269816</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>182</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4567E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36650</th>\n",
       "      <td>chrY</td>\n",
       "      <td>15174113</td>\n",
       "      <td>rs17307398</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>164</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.3259E-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36652 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270   rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511     rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290             .   A   C    .   PASS   \n",
       "3      chr1    686266             .   A   C    .   PASS   \n",
       "4      chr1    924533   rs112703963   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "36647  chrY   9528488  rs1179775265   T   G    .   PASS   \n",
       "36648  chrY  13360045     rs2032674   T   C    .   PASS   \n",
       "36649  chrY  14941891    rs17269816   C   T    .   PASS   \n",
       "36650  chrY  15174113    rs17307398   T   C    .   PASS   \n",
       "36651  chrY  20577481     rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "3      ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   85   79   79   \n",
       "4      ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  1/1   58   11   11   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36647  ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  130  186  186   \n",
       "36648  ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....  1/1  255   54   54   \n",
       "36649  ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...  1/1  182   32   32   \n",
       "36650  ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...  1/1  164   29   29   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "1        0  114    100%   4.392E-68   0   0  98  16   0   1    Homozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "3       54   25  31.65%   3.1101E-9  43  11  23   2   1   0  Heterozygous   \n",
       "4        0   11    100%   1.4176E-6   0   0  10   1   0   1    Homozygous   \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "36647  146   40  21.51%  8.6616E-14  77  69  17  23   1   0  Heterozygous   \n",
       "36648    0   54    100%  4.0229E-32   0   0  35  19   0   1    Homozygous   \n",
       "36649    0   32    100%  5.4567E-19   0   0  28   4   0   1    Homozygous   \n",
       "36650    0   29    100%  3.3259E-17   0   0  24   5   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "\n",
       "                                  Gene_Name            Gene Name  \n",
       "0                               OR4F5:79501                OR4F5  \n",
       "1                               OR4F5:79501                OR4F5  \n",
       "2                                       NaN                       \n",
       "3                                       NaN                       \n",
       "4      LOC107985728:107985728|SAMD11:148398  LOC107985728,SAMD11  \n",
       "...                                     ...                  ...  \n",
       "36647                      TSPY10:100289087               TSPY10  \n",
       "36648                              UTY:7404                  UTY  \n",
       "36649                                   NaN                       \n",
       "36650                                   NaN                       \n",
       "36651                           EIF1AY:9086               EIF1AY  \n",
       "\n",
       "[36652 rows x 25 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"Gene_Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "39a93136",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "1        0  114    100%   4.392E-68   0   0  98  16   0   1    Homozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "\n",
       "         Gene_Name Gene Name  \\\n",
       "0      OR4F5:79501     OR4F5   \n",
       "0      OR4F5:79501     OR4F5   \n",
       "1      OR4F5:79501     OR4F5   \n",
       "2              NaN             \n",
       "2              NaN             \n",
       "...            ...       ...   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "...                                                  ...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "\n",
       "                                                     csq  \n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...  \n",
       "0      G|regulatory_region_variant|MODIFIER|||Regulat...  \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...  \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...  \n",
       "2      C|intron_variant&non_coding_transcript_variant...  \n",
       "...                                                  ...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "\n",
       "[328334 rows x 27 columns]"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "3045ed68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chr1                       29303\n",
       "chr19                      25115\n",
       "chr17                      23572\n",
       "chr11                      22962\n",
       "chr2                       22900\n",
       "chr16                      20186\n",
       "chr3                       17731\n",
       "chr7                       17335\n",
       "chr12                      16069\n",
       "chr9                       13364\n",
       "chr5                       12909\n",
       "chr6                       12758\n",
       "chr8                       12545\n",
       "chr15                      12444\n",
       "chr10                      12436\n",
       "chr14                      11737\n",
       "chr4                       10735\n",
       "chr22                       9572\n",
       "chr20                       7483\n",
       "chr18                       4874\n",
       "chr13                       4688\n",
       "chrX                        3826\n",
       "chr21                       3676\n",
       "chrY                          65\n",
       "chr16_KI270728v1_random       11\n",
       "chr19_GL949746v1_alt           6\n",
       "chr19_KI270938v1_alt           6\n",
       "chr16_KI270853v1_alt           4\n",
       "chr12_KI270904v1_alt           4\n",
       "chr12_GL877876v1_alt           3\n",
       "chr6_GL000254v2_alt            2\n",
       "chr4_GL000257v2_alt            2\n",
       "chr9_KI270719v1_random         2\n",
       "chr6_GL000251v2_alt            1\n",
       "chr9_KI270823v1_alt            1\n",
       "chr17_KI270857v1_alt           1\n",
       "chr6_GL000252v2_alt            1\n",
       "chr17_KI270861v1_alt           1\n",
       "chr19_GL949752v1_alt           1\n",
       "chr19_GL949753v2_alt           1\n",
       "chr16_GL383556v1_alt           1\n",
       "chr11_KI270903v1_alt           1\n",
       "Name: CHROM, dtype: int64"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.CHROM.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c57b15af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000456328.2:n.932C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH7P|ENSG...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.2746T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.2950T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.1404T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>661656 rows × 45 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER  \\\n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "\n",
       "                                                    INFO   GT  GQ  SDP  DP  \\\n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "...                                                  ...  ...  ..  ...  ..   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "\n",
       "       RD  AD    FREQ      PVAL  RDF  RDR  ADF  ADR HET HOM      Zygosity  \\\n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "...    ..  ..     ...       ...  ...  ...  ...  ...  ..  ..           ...   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "\n",
       "               Gene_Name Gene Name  \\\n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "...                  ...       ...   \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...                                                  ...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...                 \n",
       "0      T|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "0      T|downstream_gene_variant|MODIFIER|WASH7P|ENSG...                 \n",
       "0      T|downstream_gene_variant|MODIFIER|MIR6859-1|E...                 \n",
       "0      T|regulatory_region_variant|MODIFIER|||Regulat...                 \n",
       "...                                                  ...           ...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "83759  C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...                 \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "83759  C|downstream_gene_variant|MODIFIER|DDX11L16|EN...                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar                        HGVSC HGVSP  \\\n",
       "0                                                                              \n",
       "0                                           ENST00000456328.2:n.932C>T         \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "...        ...                ...     ...                          ...   ...   \n",
       "83759                                      ENST00000484415.6:n.2746T>C         \n",
       "83759                                      ENST00000492963.6:n.2950T>C         \n",
       "83759                                                                          \n",
       "83759                                      ENST00000496301.6:n.1404T>C         \n",
       "83759                                                                          \n",
       "\n",
       "      PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0               transcribed_unprocessed_pseudogene               \n",
       "0                                           lncRNA  3/3          \n",
       "0                           unprocessed_pseudogene               \n",
       "0                                            miRNA               \n",
       "0                                         enhancer               \n",
       "...        ...                                 ...  ...    ...   \n",
       "83759                              retained_intron  5/5          \n",
       "83759                              retained_intron  6/6          \n",
       "83759                              retained_intron               \n",
       "83759                              retained_intron  2/2          \n",
       "83759                       unprocessed_pseudogene               \n",
       "\n",
       "      Protein_position Amino_acids Codons STRAND PUBMED  \\\n",
       "0                                              1          \n",
       "0                                              1          \n",
       "0                                             -1          \n",
       "0                                             -1          \n",
       "0                                                         \n",
       "...                ...         ...    ...    ...    ...   \n",
       "83759                                          1          \n",
       "83759                                          1          \n",
       "83759                                          1          \n",
       "83759                                          1          \n",
       "83759                                         -1          \n",
       "\n",
       "                              Consequence    IMPACT SIFT  \n",
       "0                 downstream_gene_variant  MODIFIER       \n",
       "0      non_coding_transcript_exon_variant  MODIFIER       \n",
       "0                 downstream_gene_variant  MODIFIER       \n",
       "0                 downstream_gene_variant  MODIFIER       \n",
       "0               regulatory_region_variant  MODIFIER       \n",
       "...                                   ...       ...  ...  \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER       \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER       \n",
       "83759             downstream_gene_variant  MODIFIER       \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER       \n",
       "83759             downstream_gene_variant  MODIFIER       \n",
       "\n",
       "[661656 rows x 45 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#vcf['SYMBOL / Gene Name'] = vcf['csq'].str.split('|').str[3]\n",
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "932993c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000456328.2:n.932C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH7P|ENSG...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.2746T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.2950T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.1404T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>661656 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER  \\\n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "\n",
       "                                                    INFO   GT  GQ  SDP  DP  \\\n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "...                                                  ...  ...  ..  ...  ..   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "\n",
       "       RD  AD    FREQ      PVAL  RDF  RDR  ADF  ADR HET HOM      Zygosity  \\\n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "...    ..  ..     ...       ...  ...  ...  ...  ...  ..  ..           ...   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "\n",
       "               Gene_Name Gene Name  \\\n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "...                  ...       ...   \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...                                                  ...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...                 \n",
       "0      T|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "0      T|downstream_gene_variant|MODIFIER|WASH7P|ENSG...                 \n",
       "0      T|downstream_gene_variant|MODIFIER|MIR6859-1|E...                 \n",
       "0      T|regulatory_region_variant|MODIFIER|||Regulat...                 \n",
       "...                                                  ...           ...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "83759  C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...                 \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "83759  C|downstream_gene_variant|MODIFIER|DDX11L16|EN...                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar                        HGVSC HGVSP  \\\n",
       "0                                                                              \n",
       "0                                           ENST00000456328.2:n.932C>T         \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "...        ...                ...     ...                          ...   ...   \n",
       "83759                                      ENST00000484415.6:n.2746T>C         \n",
       "83759                                      ENST00000492963.6:n.2950T>C         \n",
       "83759                                                                          \n",
       "83759                                      ENST00000496301.6:n.1404T>C         \n",
       "83759                                                                          \n",
       "\n",
       "      PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0               transcribed_unprocessed_pseudogene               \n",
       "0                                           lncRNA  3/3          \n",
       "0                           unprocessed_pseudogene               \n",
       "0                                            miRNA               \n",
       "0                                         enhancer               \n",
       "...        ...                                 ...  ...    ...   \n",
       "83759                              retained_intron  5/5          \n",
       "83759                              retained_intron  6/6          \n",
       "83759                              retained_intron               \n",
       "83759                              retained_intron  2/2          \n",
       "83759                       unprocessed_pseudogene               \n",
       "\n",
       "      Protein_position Amino_acids Codons STRAND PUBMED  \\\n",
       "0                                              1          \n",
       "0                                              1          \n",
       "0                                             -1          \n",
       "0                                             -1          \n",
       "0                                                         \n",
       "...                ...         ...    ...    ...    ...   \n",
       "83759                                          1          \n",
       "83759                                          1          \n",
       "83759                                          1          \n",
       "83759                                          1          \n",
       "83759                                         -1          \n",
       "\n",
       "                              Consequence    IMPACT SIFT  \\\n",
       "0                 downstream_gene_variant  MODIFIER        \n",
       "0      non_coding_transcript_exon_variant  MODIFIER        \n",
       "0                 downstream_gene_variant  MODIFIER        \n",
       "0                 downstream_gene_variant  MODIFIER        \n",
       "0               regulatory_region_variant  MODIFIER        \n",
       "...                                   ...       ...  ...   \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER        \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER        \n",
       "83759             downstream_gene_variant  MODIFIER        \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER        \n",
       "83759             downstream_gene_variant  MODIFIER        \n",
       "\n",
       "      Protein Position and Amino Acid  \n",
       "0                                 NaN  \n",
       "0                                 NaN  \n",
       "0                                 NaN  \n",
       "0                                 NaN  \n",
       "0                                 NaN  \n",
       "...                               ...  \n",
       "83759                             NaN  \n",
       "83759                             NaN  \n",
       "83759                             NaN  \n",
       "83759                             NaN  \n",
       "83759                             NaN  \n",
       "\n",
       "[661656 rows x 46 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c3aa97f0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX11L1:100287102;GNO;RS=71260404;RSPOS=13684;RV;SAO=0;SLO;SSR=0;TOPMED=0.72495380988786952,0.27504619011213047;VC=SNV;VP=0x050100000005000102000100;WGT=1;dbSNPBuildID=130;CSQ=T|downstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000450305|transcribed_unprocessed_pseudogene||||||||||rs71260404|14|1||SNV|HGNC|HGNC:37102|YES|||||||||||||||||||||||||||||||||0.1495|0.02914|0.2292|0.1784|0.2692|0.1122|0.1975|0.18|0.2406|0.149|0.1845|0.2692|gnomADg_ASJ|||||||||||||,T|non_coding_transcript_exon_variant|MODIFIER|DDX11L2|ENSG00000290825|Transcript|ENST00000456328|lncRNA|3/3||ENST00000456328.2:n.932C>T||932|||||rs71260404||1||SNV|EntrezGene||YES|||1||||||||||||||||||||||||||||||0.1495|0.02914|0.2292|0.1784|0.2692|0.1122|0.1975|0.18|0.2406|0.149|0.1845|0.2692|gnomADg_ASJ|||||||||||||,T|downstream_gene_variant|MODIFIER|WASH7P|ENSG00000227232|Transcript|ENST00000488147|unprocessed_pseudogene||||||||||rs71260404|720|-1||SNV|HGNC|HGNC:38034|YES|||||||||||||||||||||||||||||||||0.1495|0.02914|0.2292|0.1784|0.2692|0.1122|0.1975|0.18|0.2406|0.149|0.1845|0.2692|gnomADg_ASJ|||||||||||||,T|downstream_gene_variant|MODIFIER|MIR6859-1|ENSG00000278267|Transcript|ENST00000619216|miRNA||||||||||rs71260404|3685|-1||SNV|HGNC|HGNC:50039|YES|||||||||||||||||||||||||||||||||0.1495|0.02914|0.2292|0.1784|0.2692|0.1122|0.1975|0.18|0.2406|0.149|0.1845|0.2692|gnomADg_ASJ|||||||||||||,T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00001164745|enhancer||||||||||rs71260404||||SNV||||||||||||||||||||||||||||||||||||0.1495|0.02914|0.2292|0.1784|0.2692|0.1122|0.1975|0.18|0.2406|0.149|0.1845|0.2692|gnomADg_ASJ|||||||||||||'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf.INFO.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8620ac30",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000456328.2:n.932C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000456328.2</td>\n",
       "      <td>n.932C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|WASH7P|ENSG...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|MIR6859-1|E...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>DDX11L1:100287102</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...</td>\n",
       "      <td>T|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6:n.2746T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000484415.6</td>\n",
       "      <td>n.2746T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6:n.2950T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000492963.6</td>\n",
       "      <td>n.2950T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6:n.1404T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000496301.6</td>\n",
       "      <td>n.1404T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>C|downstream_gene_variant|MODIFIER|DDX11L16|EN...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>661656 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          ID REF ALT QUAL FILTER  \\\n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "0      chr1     13684  rs71260404   C   T    .   PASS   \n",
       "...     ...       ...         ...  ..  ..  ...    ...   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "83759  chrY  57211949           .   T   C    .   PASS   \n",
       "\n",
       "                                                    INFO   GT  GQ  SDP  DP  \\\n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "0      ADP=30;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=DDX1...  0/1  22   30  30   \n",
       "...                                                  ...  ...  ..  ...  ..   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "83759  ADP=15;WT=0;HET=1;HOM=0;NC=0;CSQ=C|non_coding_...  0/1  29   15  15   \n",
       "\n",
       "       RD  AD    FREQ      PVAL  RDF  RDR  ADF  ADR HET HOM      Zygosity  \\\n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "0      23   7  23.33%  0.005271    5   18    4    3   1   0  Heterozygous   \n",
       "...    ..  ..     ...       ...  ...  ...  ...  ...  ..  ..           ...   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "83759   7   8  53.33%  0.001099    4    3    2    6   1   0  Heterozygous   \n",
       "\n",
       "               Gene_Name Gene Name  \\\n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "0      DDX11L1:100287102   DDX11L1   \n",
       "...                  ...       ...   \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "83759                NaN             \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...   \n",
       "...                                                  ...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      T|downstream_gene_variant|MODIFIER|DDX11L1|ENS...                 \n",
       "0      T|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "0      T|downstream_gene_variant|MODIFIER|WASH7P|ENSG...                 \n",
       "0      T|downstream_gene_variant|MODIFIER|MIR6859-1|E...                 \n",
       "0      T|regulatory_region_variant|MODIFIER|||Regulat...                 \n",
       "...                                                  ...           ...   \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "83759  C|downstream_gene_variant|MODIFIER|WASH6P|ENSG...                 \n",
       "83759  C|non_coding_transcript_exon_variant|MODIFIER|...                 \n",
       "83759  C|downstream_gene_variant|MODIFIER|DDX11L16|EN...                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar                        HGVSC HGVSP  \\\n",
       "0                                                                              \n",
       "0                                           ENST00000456328.2:n.932C>T         \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "0                                                                              \n",
       "...        ...                ...     ...                          ...   ...   \n",
       "83759                                      ENST00000484415.6:n.2746T>C         \n",
       "83759                                      ENST00000492963.6:n.2950T>C         \n",
       "83759                                                                          \n",
       "83759                                      ENST00000496301.6:n.1404T>C         \n",
       "83759                                                                          \n",
       "\n",
       "      PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0               transcribed_unprocessed_pseudogene               \n",
       "0                                           lncRNA  3/3          \n",
       "0                           unprocessed_pseudogene               \n",
       "0                                            miRNA               \n",
       "0                                         enhancer               \n",
       "...        ...                                 ...  ...    ...   \n",
       "83759                              retained_intron  5/5          \n",
       "83759                              retained_intron  6/6          \n",
       "83759                              retained_intron               \n",
       "83759                              retained_intron  2/2          \n",
       "83759                       unprocessed_pseudogene               \n",
       "\n",
       "      Protein_position Amino_acids Codons STRAND PUBMED  \\\n",
       "0                                              1          \n",
       "0                                              1          \n",
       "0                                             -1          \n",
       "0                                             -1          \n",
       "0                                                         \n",
       "...                ...         ...    ...    ...    ...   \n",
       "83759                                          1          \n",
       "83759                                          1          \n",
       "83759                                          1          \n",
       "83759                                          1          \n",
       "83759                                         -1          \n",
       "\n",
       "                              Consequence    IMPACT SIFT  \\\n",
       "0                 downstream_gene_variant  MODIFIER        \n",
       "0      non_coding_transcript_exon_variant  MODIFIER        \n",
       "0                 downstream_gene_variant  MODIFIER        \n",
       "0                 downstream_gene_variant  MODIFIER        \n",
       "0               regulatory_region_variant  MODIFIER        \n",
       "...                                   ...       ...  ...   \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER        \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER        \n",
       "83759             downstream_gene_variant  MODIFIER        \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER        \n",
       "83759             downstream_gene_variant  MODIFIER        \n",
       "\n",
       "      Protein Position and Amino Acid              HGVSc HGVSc (Transcript)  \\\n",
       "0                                 NaN                                  None   \n",
       "0                                 NaN  ENST00000456328.2           n.932C>T   \n",
       "0                                 NaN                                  None   \n",
       "0                                 NaN                                  None   \n",
       "0                                 NaN                                  None   \n",
       "...                               ...                ...                ...   \n",
       "83759                             NaN  ENST00000484415.6          n.2746T>C   \n",
       "83759                             NaN  ENST00000492963.6          n.2950T>C   \n",
       "83759                             NaN                                  None   \n",
       "83759                             NaN  ENST00000496301.6          n.1404T>C   \n",
       "83759                             NaN                                  None   \n",
       "\n",
       "      HGVSp HGVSp (Transcript)  \n",
       "0                         None  \n",
       "0                         None  \n",
       "0                         None  \n",
       "0                         None  \n",
       "0                         None  \n",
       "...     ...                ...  \n",
       "83759                     None  \n",
       "83759                     None  \n",
       "83759                     None  \n",
       "83759                     None  \n",
       "83759                     None  \n",
       "\n",
       "[661656 rows x 50 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "6a2fa041",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000456328.2</td>\n",
       "      <td>n.932C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6</td>\n",
       "      <td>n.2746T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6</td>\n",
       "      <td>n.2950T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6</td>\n",
       "      <td>n.1404T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>661656 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name          ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "...         ...         ...   ...       ...  ..  ..           ...   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "\n",
       "                              Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0                 downstream_gene_variant  MODIFIER                          \n",
       "0      non_coding_transcript_exon_variant  MODIFIER                          \n",
       "0                 downstream_gene_variant  MODIFIER                          \n",
       "0                 downstream_gene_variant  MODIFIER                          \n",
       "0               regulatory_region_variant  MODIFIER                          \n",
       "...                                   ...       ...           ...      ...   \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER                          \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER                          \n",
       "83759             downstream_gene_variant  MODIFIER                          \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER                          \n",
       "83759             downstream_gene_variant  MODIFIER                          \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript) HGVSp  \\\n",
       "0                                                                 None         \n",
       "0                                 ENST00000456328.2           n.932C>T         \n",
       "0                                                                 None         \n",
       "0                                                                 None         \n",
       "0                                                                 None         \n",
       "...                  ...     ...                ...                ...   ...   \n",
       "83759                             ENST00000484415.6          n.2746T>C         \n",
       "83759                             ENST00000492963.6          n.2950T>C         \n",
       "83759                                                             None         \n",
       "83759                             ENST00000496301.6          n.1404T>C         \n",
       "83759                                                             None         \n",
       "\n",
       "      HGVSp (Transcript)   GT  GQ  SDP  DP  RD  AD    FREQ      PVAL  RDF  \\\n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "...                  ...  ...  ..  ...  ..  ..  ..     ...       ...  ...   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "\n",
       "       RDR  ADF  ADR SIFT PolyPhen                             BIOTYPE EXON  \\\n",
       "0       18    4    3                transcribed_unprocessed_pseudogene        \n",
       "0       18    4    3                                            lncRNA  3/3   \n",
       "0       18    4    3                            unprocessed_pseudogene        \n",
       "0       18    4    3                                             miRNA        \n",
       "0       18    4    3                                          enhancer        \n",
       "...    ...  ...  ...  ...      ...                                 ...  ...   \n",
       "83759    3    2    6                                   retained_intron  5/5   \n",
       "83759    3    2    6                                   retained_intron  6/6   \n",
       "83759    3    2    6                                   retained_intron        \n",
       "83759    3    2    6                                   retained_intron  2/2   \n",
       "83759    3    2    6                            unprocessed_pseudogene        \n",
       "\n",
       "      INTRON Protein Position and Amino Acid Codons STRAND PUBMED  \n",
       "0                                        NaN             1         \n",
       "0                                        NaN             1         \n",
       "0                                        NaN            -1         \n",
       "0                                        NaN            -1         \n",
       "0                                        NaN                       \n",
       "...      ...                             ...    ...    ...    ...  \n",
       "83759                                    NaN             1         \n",
       "83759                                    NaN             1         \n",
       "83759                                    NaN             1         \n",
       "83759                                    NaN             1         \n",
       "83759                                    NaN            -1         \n",
       "\n",
       "[661656 rows x 38 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf[['Gene Name', 'ID','CHROM', 'POS', 'REF', 'ALT', 'Zygosity', 'Consequence', 'IMPACT',\n",
    "          'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "          'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "          'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b70b45f5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                                                                                                                                                                                                                                                               476553\n",
       "not_provided                                                                                                                                                                                                                                                    76180\n",
       "not_specified&not_provided                                                                                                                                                                                                                                       2856\n",
       "not_specified                                                                                                                                                                                                                                                    2582\n",
       "not_provided&not_specified                                                                                                                                                                                                                                       1532\n",
       "                                                                                                                                                                                                                                                                ...  \n",
       "Achromatopsia&Cone_dystrophy_4&not_specified&not_provided                                                                                                                                                                                                           1\n",
       "Bile_acid_malabsorption&_primary&_1                                                                                                                                                                                                                                 1\n",
       "Achromatopsia&not_specified&not_provided&Cone_dystrophy_4                                                                                                                                                                                                           1\n",
       "Cardiovascular_phenotype&not_specified&not_provided&Cardiomyopathy&Hypertrophic_cardiomyopathy_1&MYH7-related_skeletal_myopathy&Dilated_cardiomyopathy_1S&Myosin_storage_myopathy&Hypertrophic_cardiomyopathy&Left_ventricular_noncompaction_cardiomyopathy         1\n",
       "Amyotrophic_lateral_sclerosis_type_4&Spinocerebellar_ataxia&_autosomal_recessive&_with_axonal_neuropathy_2&not_provided&not_specified                                                                                                                               1\n",
       "Name: ClinVar_CLNDN, Length: 5572, dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final.ClinVar_CLNDN.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "75cb37d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed_unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000456328.2</td>\n",
       "      <td>n.932C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.005271</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6</td>\n",
       "      <td>n.2746T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6</td>\n",
       "      <td>n.2950T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6</td>\n",
       "      <td>n.1404T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.001099</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>661656 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name          ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "...         ...         ...   ...       ...  ..  ..           ...   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "\n",
       "                              Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0                 downstream_gene_variant  MODIFIER                          \n",
       "0      non_coding_transcript_exon_variant  MODIFIER                          \n",
       "0                 downstream_gene_variant  MODIFIER                          \n",
       "0                 downstream_gene_variant  MODIFIER                          \n",
       "0               regulatory_region_variant  MODIFIER                          \n",
       "...                                   ...       ...           ...      ...   \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER                          \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER                          \n",
       "83759             downstream_gene_variant  MODIFIER                          \n",
       "83759  non_coding_transcript_exon_variant  MODIFIER                          \n",
       "83759             downstream_gene_variant  MODIFIER                          \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript) HGVSp  \\\n",
       "0                                                                 None         \n",
       "0                                 ENST00000456328.2           n.932C>T         \n",
       "0                                                                 None         \n",
       "0                                                                 None         \n",
       "0                                                                 None         \n",
       "...                  ...     ...                ...                ...   ...   \n",
       "83759                             ENST00000484415.6          n.2746T>C         \n",
       "83759                             ENST00000492963.6          n.2950T>C         \n",
       "83759                                                             None         \n",
       "83759                             ENST00000496301.6          n.1404T>C         \n",
       "83759                                                             None         \n",
       "\n",
       "      HGVSp (Transcript)   GT  GQ  SDP  DP  RD  AD    FREQ      PVAL  RDF  \\\n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "0                   None  0/1  22   30  30  23   7  23.33%  0.005271    5   \n",
       "...                  ...  ...  ..  ...  ..  ..  ..     ...       ...  ...   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "83759               None  0/1  29   15  15   7   8  53.33%  0.001099    4   \n",
       "\n",
       "       RDR  ADF  ADR SIFT PolyPhen                             BIOTYPE EXON  \\\n",
       "0       18    4    3                transcribed_unprocessed_pseudogene        \n",
       "0       18    4    3                                            lncRNA  3/3   \n",
       "0       18    4    3                            unprocessed_pseudogene        \n",
       "0       18    4    3                                             miRNA        \n",
       "0       18    4    3                                          enhancer        \n",
       "...    ...  ...  ...  ...      ...                                 ...  ...   \n",
       "83759    3    2    6                                   retained_intron  5/5   \n",
       "83759    3    2    6                                   retained_intron  6/6   \n",
       "83759    3    2    6                                   retained_intron        \n",
       "83759    3    2    6                                   retained_intron  2/2   \n",
       "83759    3    2    6                            unprocessed_pseudogene        \n",
       "\n",
       "      INTRON Protein Position and Amino Acid Codons STRAND PUBMED  \n",
       "0                                        NaN             1         \n",
       "0                                        NaN             1         \n",
       "0                                        NaN            -1         \n",
       "0                                        NaN            -1         \n",
       "0                                        NaN                       \n",
       "...      ...                             ...    ...    ...    ...  \n",
       "83759                                    NaN             1         \n",
       "83759                                    NaN             1         \n",
       "83759                                    NaN             1         \n",
       "83759                                    NaN             1         \n",
       "83759                                    NaN            -1         \n",
       "\n",
       "[661656 rows x 38 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "# Print the modified DataFrame\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "704c84c6",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000456328.2</td>\n",
       "      <td>n.932C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6</td>\n",
       "      <td>n.2746T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6</td>\n",
       "      <td>n.2950T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6</td>\n",
       "      <td>n.1404T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>661656 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name          ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "...         ...         ...   ...       ...  ..  ..           ...   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "\n",
       "                              Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0                 downstream gene variant  MODIFIER                          \n",
       "0      non coding transcript exon variant  MODIFIER                          \n",
       "0                 downstream gene variant  MODIFIER                          \n",
       "0                 downstream gene variant  MODIFIER                          \n",
       "0               regulatory region variant  MODIFIER                          \n",
       "...                                   ...       ...           ...      ...   \n",
       "83759  non coding transcript exon variant  MODIFIER                          \n",
       "83759  non coding transcript exon variant  MODIFIER                          \n",
       "83759             downstream gene variant  MODIFIER                          \n",
       "83759  non coding transcript exon variant  MODIFIER                          \n",
       "83759             downstream gene variant  MODIFIER                          \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript) HGVSp  \\\n",
       "0                                                                 None         \n",
       "0                                 ENST00000456328.2           n.932C>T         \n",
       "0                                                                 None         \n",
       "0                                                                 None         \n",
       "0                                                                 None         \n",
       "...                  ...     ...                ...                ...   ...   \n",
       "83759                             ENST00000484415.6          n.2746T>C         \n",
       "83759                             ENST00000492963.6          n.2950T>C         \n",
       "83759                                                             None         \n",
       "83759                             ENST00000496301.6          n.1404T>C         \n",
       "83759                                                             None         \n",
       "\n",
       "      HGVSp (Transcript)   GT  GQ SDP  DP  RD AD    FREQ       PVAL RDF RDR  \\\n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "...                  ...  ...  ..  ..  ..  .. ..     ...        ...  ..  ..   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "\n",
       "      ADF ADR SIFT PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0       4   3                transcribed unprocessed pseudogene               \n",
       "0       4   3                                            lncRNA  3/3          \n",
       "0       4   3                            unprocessed pseudogene               \n",
       "0       4   3                                             miRNA               \n",
       "0       4   3                                          enhancer               \n",
       "...    ..  ..  ...      ...                                 ...  ...    ...   \n",
       "83759   2   6                                   retained intron  5/5          \n",
       "83759   2   6                                   retained intron  6/6          \n",
       "83759   2   6                                   retained intron               \n",
       "83759   2   6                                   retained intron  2/2          \n",
       "83759   2   6                            unprocessed pseudogene               \n",
       "\n",
       "      Protein Position and Amino Acid Codons STRAND PUBMED  \n",
       "0                                 nan             1         \n",
       "0                                 nan             1         \n",
       "0                                 nan            -1         \n",
       "0                                 nan            -1         \n",
       "0                                 nan                       \n",
       "...                               ...    ...    ...    ...  \n",
       "83759                             nan             1         \n",
       "83759                             nan             1         \n",
       "83759                             nan             1         \n",
       "83759                             nan             1         \n",
       "83759                             nan            -1         \n",
       "\n",
       "[661656 rows x 38 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "6aaaafab",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000456328.2</td>\n",
       "      <td>n.932C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>miRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>rs71260404</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>22</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>23</td>\n",
       "      <td>7</td>\n",
       "      <td>23.33%</td>\n",
       "      <td>0.0052713</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>enhancer</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484415.6</td>\n",
       "      <td>n.2746T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>5/5</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000492963.6</td>\n",
       "      <td>n.2950T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000496301.6</td>\n",
       "      <td>n.1404T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83759</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chrY</td>\n",
       "      <td>57211949</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>0.0010995</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>661656 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name          ID CHROM       POS REF ALT      Zygosity  \\\n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "0       DDX11L1  rs71260404  chr1     13684   C   T  Heterozygous   \n",
       "...         ...         ...   ...       ...  ..  ..           ...   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "83759                     .  chrY  57211949   T   C  Heterozygous   \n",
       "\n",
       "                              Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0                 downstream gene variant  MODIFIER                          \n",
       "0      non coding transcript exon variant  MODIFIER                          \n",
       "0                 downstream gene variant  MODIFIER                          \n",
       "0                 downstream gene variant  MODIFIER                          \n",
       "0               regulatory region variant  MODIFIER                          \n",
       "...                                   ...       ...           ...      ...   \n",
       "83759  non coding transcript exon variant  MODIFIER                          \n",
       "83759  non coding transcript exon variant  MODIFIER                          \n",
       "83759             downstream gene variant  MODIFIER                          \n",
       "83759  non coding transcript exon variant  MODIFIER                          \n",
       "83759             downstream gene variant  MODIFIER                          \n",
       "\n",
       "      ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript) HGVSp  \\\n",
       "0                                                                 None         \n",
       "0                                 ENST00000456328.2           n.932C>T         \n",
       "0                                                                 None         \n",
       "0                                                                 None         \n",
       "0                                                                 None         \n",
       "...                  ...     ...                ...                ...   ...   \n",
       "83759                             ENST00000484415.6          n.2746T>C         \n",
       "83759                             ENST00000492963.6          n.2950T>C         \n",
       "83759                                                             None         \n",
       "83759                             ENST00000496301.6          n.1404T>C         \n",
       "83759                                                             None         \n",
       "\n",
       "      HGVSp (Transcript)   GT  GQ SDP  DP  RD AD    FREQ       PVAL RDF RDR  \\\n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "0                   None  0/1  22  30  30  23  7  23.33%  0.0052713   5  18   \n",
       "...                  ...  ...  ..  ..  ..  .. ..     ...        ...  ..  ..   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "83759               None  0/1  29  15  15   7  8  53.33%  0.0010995   4   3   \n",
       "\n",
       "      ADF ADR SIFT PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0       4   3                transcribed unprocessed pseudogene               \n",
       "0       4   3                                            lncRNA  3/3          \n",
       "0       4   3                            unprocessed pseudogene               \n",
       "0       4   3                                             miRNA               \n",
       "0       4   3                                          enhancer               \n",
       "...    ..  ..  ...      ...                                 ...  ...    ...   \n",
       "83759   2   6                                   retained intron  5/5          \n",
       "83759   2   6                                   retained intron  6/6          \n",
       "83759   2   6                                   retained intron               \n",
       "83759   2   6                                   retained intron  2/2          \n",
       "83759   2   6                            unprocessed pseudogene               \n",
       "\n",
       "      Protein Position and Amino Acid Codons STRAND PUBMED  \n",
       "0                                 nan             1         \n",
       "0                                 nan             1         \n",
       "0                                 nan            -1         \n",
       "0                                 nan            -1         \n",
       "0                                 nan                       \n",
       "...                               ...    ...    ...    ...  \n",
       "83759                             nan             1         \n",
       "83759                             nan             1         \n",
       "83759                             nan             1         \n",
       "83759                             nan             1         \n",
       "83759                             nan            -1         \n",
       "\n",
       "[661656 rows x 38 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final['POS'] = vcf_final['POS'].astype('int64')\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "e30d730c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene Name                          0\n",
       "ID                                 0\n",
       "CHROM                              0\n",
       "POS                                0\n",
       "REF                                0\n",
       "ALT                                0\n",
       "Zygosity                           0\n",
       "Consequence                        0\n",
       "IMPACT                             0\n",
       "ClinVar_CLNDN                      0\n",
       "CLIN_SIG                           0\n",
       "ClinVar_CLNREVSTAT                 0\n",
       "ClinVar                            0\n",
       "HGVSc                              0\n",
       "HGVSc (Transcript)                 0\n",
       "HGVSp                              0\n",
       "HGVSp (Transcript)                 0\n",
       "GT                                 0\n",
       "GQ                                 0\n",
       "SDP                                0\n",
       "DP                                 0\n",
       "RD                                 0\n",
       "AD                                 0\n",
       "FREQ                               0\n",
       "PVAL                               0\n",
       "RDF                                0\n",
       "RDR                                0\n",
       "ADF                                0\n",
       "ADR                                0\n",
       "SIFT                               0\n",
       "PolyPhen                           0\n",
       "BIOTYPE                            0\n",
       "EXON                               0\n",
       "INTRON                             0\n",
       "Protein Position and Amino Acid    0\n",
       "Codons                             0\n",
       "STRAND                             0\n",
       "PUBMED                             0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "aece4e6b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                                                                            477203\n",
       "benign                                                                      168597\n",
       "benign,likely benign                                                          4708\n",
       "likely benign                                                                 2996\n",
       "benign,benign/likely benign                                                   1331\n",
       "                                                                             ...  \n",
       "benign,benign/likely benign,conflicting interpretations of pathogenicity         3\n",
       "uncertain significance,risk factor                                               2\n",
       "association,likely benign                                                        2\n",
       "likely benign,drug response,other                                                2\n",
       "conflicting interpretations of pathogenicity,risk factor,benign                  2\n",
       "Name: CLIN_SIG, Length: 109, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final.CLIN_SIG.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "457f12d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>69270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>69511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>69735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>69897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>451229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38110</th>\n",
       "      <td>2789135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38111</th>\n",
       "      <td>3019783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38112</th>\n",
       "      <td>6885470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38113</th>\n",
       "      <td>9466925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38114</th>\n",
       "      <td>9467257</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38115 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           POS\n",
       "0        69270\n",
       "1        69511\n",
       "2        69735\n",
       "3        69897\n",
       "4       451229\n",
       "...        ...\n",
       "38110  2789135\n",
       "38111  3019783\n",
       "38112  6885470\n",
       "38113  9466925\n",
       "38114  9467257\n",
       "\n",
       "[38115 rows x 1 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHGLBS535/POS.xlsx')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "07d2dd1b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM      POS REF ALT      Zygosity  \\\n",
       "0          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "1          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "2          OR4F5     rs2691305  chr1    69511   A   G    Homozygous   \n",
       "3          OR4F5  rs1245478362  chr1    69735   A   G  Heterozygous   \n",
       "4          OR4F5   rs200676709  chr1    69897   T   C  Heterozygous   \n",
       "...          ...           ...   ...      ...  ..  ..           ...   \n",
       "333893     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333894     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333895     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333896     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333897     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "\n",
       "                      Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0              synonymous variant       LOW                          \n",
       "1       regulatory region variant  MODIFIER                          \n",
       "2                missense variant  MODERATE                          \n",
       "3              synonymous variant       LOW                          \n",
       "4              synonymous variant       LOW                          \n",
       "...                           ...       ...           ...      ...   \n",
       "333893      upstream gene variant  MODIFIER                          \n",
       "333894           missense variant  MODERATE                          \n",
       "333895           missense variant  MODERATE                          \n",
       "333896      upstream gene variant  MODIFIER                          \n",
       "333897      upstream gene variant  MODIFIER                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                  ENST00000641515.2           c.243A>G   \n",
       "1                                                                  None   \n",
       "2                                  ENST00000641515.2           c.484A>G   \n",
       "3                                  ENST00000641515.2           c.708A>G   \n",
       "4                                  ENST00000641515.2           c.870T>C   \n",
       "...                   ...     ...                ...                ...   \n",
       "333893                                                             None   \n",
       "333894                             ENST00000423647.6           c.257G>A   \n",
       "333895                             ENST00000451548.6           c.257G>A   \n",
       "333896                                                             None   \n",
       "333897                                                             None   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD  \\\n",
       "0       ENSP00000493376.2         p.Ser81%3D  0/1  120  101  101   66   35   \n",
       "1                                       None  0/1  120  101  101   66   35   \n",
       "2       ENSP00000493376.2        p.Thr162Ala  1/1  255  143  143    0  143   \n",
       "3       ENSP00000493376.2        p.Leu236%3D  0/1  163  180  180  131   49   \n",
       "4       ENSP00000493376.2        p.Ser290%3D  0/1   75   96   96   73   23   \n",
       "...                   ...                ...  ...  ...  ...  ...  ...  ...   \n",
       "333893                                  None  0/1  255  383  383  273  110   \n",
       "333894  ENSP00000389324.3         p.Arg86Gln  0/1  255  383  383  273  110   \n",
       "333895  ENSP00000403304.1         p.Arg86Gln  0/1  255  383  383  273  110   \n",
       "333896                                  None  0/1  255  383  383  273  110   \n",
       "333897                                  None  0/1  255  383  383  273  110   \n",
       "\n",
       "          FREQ                   PVAL  RDF  RDR  ADF ADR             SIFT  \\\n",
       "0       34.65%             8.1311e-13   48   18   24  11                    \n",
       "1       34.65%             8.1311e-13   48   18   24  11                    \n",
       "2         100%             1.7063e-85    0    0  100  43  tolerated(0.92)   \n",
       "3       27.22%             4.0111e-17   84   47   39  10                    \n",
       "4       23.96%             2.6678e-08   30   43   13  10                    \n",
       "...        ...                    ...  ...  ...  ...  ..              ...   \n",
       "333893  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "333894  28.72%  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333895  28.72%  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333896  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "333897  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "\n",
       "             PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0                                          protein coding  3/3          \n",
       "1                                         TF binding site               \n",
       "2           benign(0)                      protein coding  3/3          \n",
       "3                                          protein coding  3/3          \n",
       "4                                          protein coding  3/3          \n",
       "...               ...                                 ...  ...    ...   \n",
       "333893                 transcribed unprocessed pseudogene               \n",
       "333894  benign(0.375)                      protein coding  1/6          \n",
       "333895  benign(0.101)                      protein coding  1/6          \n",
       "333896                                             lncRNA               \n",
       "333897                                             lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  S81  tcA/tcG      1         \n",
       "1                                  nan                         \n",
       "2                                T162A  Aca/Gca      1         \n",
       "3                                 L236  ctA/ctG      1         \n",
       "4                                 S290  tcT/tcC      1         \n",
       "...                                ...      ...    ...    ...  \n",
       "333893                             nan              -1         \n",
       "333894                            R86Q  cGg/cAg      1         \n",
       "333895                            R86Q  cGg/cAg      1         \n",
       "333896                             nan              -1         \n",
       "333897                             nan              -1         \n",
       "\n",
       "[333898 rows x 38 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_vcf = pd.merge(vcf_final, df, on = 'POS', how = 'inner')\n",
    "merged_vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "cac9bbbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "354"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "336150-335796"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "4a401a1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#merged_vcf.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHHSPTGPCSP6_processed_vcf.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e50eb726",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "636bb5be",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHHSPTGPCSP6_processed_vcf.xlsx')\n",
    "#df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "1831f41b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "downstream gene variant                                     58852\n",
       "intron variant                                              46047\n",
       "upstream gene variant                                       43542\n",
       "synonymous variant                                          35881\n",
       "missense variant                                            28788\n",
       "                                                            ...  \n",
       "stop gained,splice region variant                               4\n",
       "coding sequence variant                                         4\n",
       "start lost,splice region variant,NMD transcript variant         3\n",
       "stop gained,splice region variant,NMD transcript variant        3\n",
       "stop lost,splice region variant,NMD transcript variant          1\n",
       "Name: Consequence, Length: 63, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_vcf.Consequence.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "ac6adf7f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM      POS REF ALT      Zygosity  \\\n",
       "0          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "1          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "2          OR4F5     rs2691305  chr1    69511   A   G    Homozygous   \n",
       "3          OR4F5  rs1245478362  chr1    69735   A   G  Heterozygous   \n",
       "4          OR4F5   rs200676709  chr1    69897   T   C  Heterozygous   \n",
       "...          ...           ...   ...      ...  ..  ..           ...   \n",
       "333893     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333894     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333895     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333896     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333897     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "\n",
       "                      Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0              synonymous variant       LOW                          \n",
       "1       regulatory region variant  MODIFIER                          \n",
       "2                missense variant  MODERATE                          \n",
       "3              synonymous variant       LOW                          \n",
       "4              synonymous variant       LOW                          \n",
       "...                           ...       ...           ...      ...   \n",
       "333893      upstream gene variant  MODIFIER                          \n",
       "333894           missense variant  MODERATE                          \n",
       "333895           missense variant  MODERATE                          \n",
       "333896      upstream gene variant  MODIFIER                          \n",
       "333897      upstream gene variant  MODIFIER                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                  ENST00000641515.2           c.243A>G   \n",
       "1                                                                  None   \n",
       "2                                  ENST00000641515.2           c.484A>G   \n",
       "3                                  ENST00000641515.2           c.708A>G   \n",
       "4                                  ENST00000641515.2           c.870T>C   \n",
       "...                   ...     ...                ...                ...   \n",
       "333893                                                             None   \n",
       "333894                             ENST00000423647.6           c.257G>A   \n",
       "333895                             ENST00000451548.6           c.257G>A   \n",
       "333896                                                             None   \n",
       "333897                                                             None   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD  \\\n",
       "0       ENSP00000493376.2         p.Ser81%3D  0/1  120  101  101   66   35   \n",
       "1                                       None  0/1  120  101  101   66   35   \n",
       "2       ENSP00000493376.2        p.Thr162Ala  1/1  255  143  143    0  143   \n",
       "3       ENSP00000493376.2        p.Leu236%3D  0/1  163  180  180  131   49   \n",
       "4       ENSP00000493376.2        p.Ser290%3D  0/1   75   96   96   73   23   \n",
       "...                   ...                ...  ...  ...  ...  ...  ...  ...   \n",
       "333893                                  None  0/1  255  383  383  273  110   \n",
       "333894  ENSP00000389324.3         p.Arg86Gln  0/1  255  383  383  273  110   \n",
       "333895  ENSP00000403304.1         p.Arg86Gln  0/1  255  383  383  273  110   \n",
       "333896                                  None  0/1  255  383  383  273  110   \n",
       "333897                                  None  0/1  255  383  383  273  110   \n",
       "\n",
       "          FREQ                   PVAL  RDF  RDR  ADF ADR             SIFT  \\\n",
       "0       34.65%             8.1311e-13   48   18   24  11                    \n",
       "1       34.65%             8.1311e-13   48   18   24  11                    \n",
       "2         100%             1.7063e-85    0    0  100  43  tolerated(0.92)   \n",
       "3       27.22%             4.0111e-17   84   47   39  10                    \n",
       "4       23.96%             2.6678e-08   30   43   13  10                    \n",
       "...        ...                    ...  ...  ...  ...  ..              ...   \n",
       "333893  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "333894  28.72%  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333895  28.72%  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333896  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "333897  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "\n",
       "             PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0                                          protein coding  3/3          \n",
       "1                                         TF binding site               \n",
       "2           benign(0)                      protein coding  3/3          \n",
       "3                                          protein coding  3/3          \n",
       "4                                          protein coding  3/3          \n",
       "...               ...                                 ...  ...    ...   \n",
       "333893                 transcribed unprocessed pseudogene               \n",
       "333894  benign(0.375)                      protein coding  1/6          \n",
       "333895  benign(0.101)                      protein coding  1/6          \n",
       "333896                                             lncRNA               \n",
       "333897                                             lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                  S81  tcA/tcG      1          \n",
       "1                                  nan                          \n",
       "2                                T162A  Aca/Gca      1          \n",
       "3                                 L236  ctA/ctG      1          \n",
       "4                                 S290  tcT/tcC      1          \n",
       "...                                ...      ...    ...    ...   \n",
       "333893                             nan              -1          \n",
       "333894                            R86Q  cGg/cAg      1          \n",
       "333895                            R86Q  cGg/cAg      1          \n",
       "333896                             nan              -1          \n",
       "333897                             nan              -1          \n",
       "\n",
       "                      consequence  \n",
       "0              synonymous variant  \n",
       "1       regulatory region variant  \n",
       "2                missense variant  \n",
       "3              synonymous variant  \n",
       "4              synonymous variant  \n",
       "...                           ...  \n",
       "333893      upstream gene variant  \n",
       "333894           missense variant  \n",
       "333895           missense variant  \n",
       "333896      upstream gene variant  \n",
       "333897      upstream gene variant  \n",
       "\n",
       "[333898 rows x 39 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_vcf['consequence'] = merged_vcf['Consequence'].str.split(',').str[0]\n",
    "merged_vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "947b57f5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.751T&gt;C</td>\n",
       "      <td>ENSP00000349216.4</td>\n",
       "      <td>p.Trp251Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/12</td>\n",
       "      <td></td>\n",
       "      <td>W251R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>c.1027T&gt;C</td>\n",
       "      <td>ENSP00000342313.3</td>\n",
       "      <td>p.Trp343Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W343R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000455979.1</td>\n",
       "      <td>c.508T&gt;C</td>\n",
       "      <td>ENSP00000412228.1</td>\n",
       "      <td>p.Trp170Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>4/7</td>\n",
       "      <td></td>\n",
       "      <td>W170R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000464948.1</td>\n",
       "      <td>n.286T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>1/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000466827.1</td>\n",
       "      <td>n.191T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000474461.1</td>\n",
       "      <td>n.389T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>3/4</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000616016.5</td>\n",
       "      <td>c.1516T&gt;C</td>\n",
       "      <td>ENSP00000478421.2</td>\n",
       "      <td>p.Trp506Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W506R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000616125.5</td>\n",
       "      <td>c.703T&gt;C</td>\n",
       "      <td>ENSP00000484643.1</td>\n",
       "      <td>p.Trp235Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/11</td>\n",
       "      <td></td>\n",
       "      <td>W235R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000617307.5</td>\n",
       "      <td>c.1030T&gt;C</td>\n",
       "      <td>ENSP00000482090.2</td>\n",
       "      <td>p.Trp344Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>W344R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618181.5</td>\n",
       "      <td>c.652T&gt;C</td>\n",
       "      <td>ENSP00000480870.1</td>\n",
       "      <td>p.Trp218Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/10</td>\n",
       "      <td></td>\n",
       "      <td>W218R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618323.5</td>\n",
       "      <td>c.1519T&gt;C</td>\n",
       "      <td>ENSP00000480678.2</td>\n",
       "      <td>p.Trp507Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W507R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618779.5</td>\n",
       "      <td>c.841T&gt;C</td>\n",
       "      <td>ENSP00000484256.1</td>\n",
       "      <td>p.Trp281Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/12</td>\n",
       "      <td></td>\n",
       "      <td>W281R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000622503.5</td>\n",
       "      <td>c.1030T&gt;C</td>\n",
       "      <td>ENSP00000482138.1</td>\n",
       "      <td>p.Trp344Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>W344R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name         ID CHROM     POS REF ALT    Zygosity  \\\n",
       "46    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "47    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "48    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "49    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "50    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "51    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "52    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "53    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "54    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "55    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "56    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "57    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "58    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "59    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "60    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "61    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "62    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "63    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "64    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "\n",
       "                           Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "46             downstream gene variant  MODIFIER  not provided   benign   \n",
       "47                    missense variant  MODERATE  not provided   benign   \n",
       "48                    missense variant  MODERATE  not provided   benign   \n",
       "49                    missense variant  MODERATE  not provided   benign   \n",
       "50  non coding transcript exon variant  MODIFIER  not provided   benign   \n",
       "51  non coding transcript exon variant  MODIFIER  not provided   benign   \n",
       "52  non coding transcript exon variant  MODIFIER  not provided   benign   \n",
       "53             downstream gene variant  MODIFIER  not provided   benign   \n",
       "54             downstream gene variant  MODIFIER  not provided   benign   \n",
       "55             downstream gene variant  MODIFIER  not provided   benign   \n",
       "56             downstream gene variant  MODIFIER  not provided   benign   \n",
       "57                    missense variant  MODERATE  not provided   benign   \n",
       "58                    missense variant  MODERATE  not provided   benign   \n",
       "59                    missense variant  MODERATE  not provided   benign   \n",
       "60                    missense variant  MODERATE  not provided   benign   \n",
       "61                    missense variant  MODERATE  not provided   benign   \n",
       "62                    missense variant  MODERATE  not provided   benign   \n",
       "63                    missense variant  MODERATE  not provided   benign   \n",
       "64           regulatory region variant  MODIFIER  not provided   benign   \n",
       "\n",
       "                     ClinVar_CLNREVSTAT  ClinVar              HGVSc  \\\n",
       "46  criteria provided, single submitter  1166513                      \n",
       "47  criteria provided, single submitter  1166513  ENST00000341065.8   \n",
       "48  criteria provided, single submitter  1166513  ENST00000342066.8   \n",
       "49  criteria provided, single submitter  1166513  ENST00000455979.1   \n",
       "50  criteria provided, single submitter  1166513  ENST00000464948.1   \n",
       "51  criteria provided, single submitter  1166513  ENST00000466827.1   \n",
       "52  criteria provided, single submitter  1166513  ENST00000474461.1   \n",
       "53  criteria provided, single submitter  1166513                      \n",
       "54  criteria provided, single submitter  1166513                      \n",
       "55  criteria provided, single submitter  1166513                      \n",
       "56  criteria provided, single submitter  1166513                      \n",
       "57  criteria provided, single submitter  1166513  ENST00000616016.5   \n",
       "58  criteria provided, single submitter  1166513  ENST00000616125.5   \n",
       "59  criteria provided, single submitter  1166513  ENST00000617307.5   \n",
       "60  criteria provided, single submitter  1166513  ENST00000618181.5   \n",
       "61  criteria provided, single submitter  1166513  ENST00000618323.5   \n",
       "62  criteria provided, single submitter  1166513  ENST00000618779.5   \n",
       "63  criteria provided, single submitter  1166513  ENST00000622503.5   \n",
       "64  criteria provided, single submitter  1166513                      \n",
       "\n",
       "   HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  \\\n",
       "46               None                                  None  1/1  255  79  79   \n",
       "47           c.751T>C  ENSP00000349216.4        p.Trp251Arg  1/1  255  79  79   \n",
       "48          c.1027T>C  ENSP00000342313.3        p.Trp343Arg  1/1  255  79  79   \n",
       "49           c.508T>C  ENSP00000412228.1        p.Trp170Arg  1/1  255  79  79   \n",
       "50           n.286T>C                                  None  1/1  255  79  79   \n",
       "51           n.191T>C                                  None  1/1  255  79  79   \n",
       "52           n.389T>C                                  None  1/1  255  79  79   \n",
       "53               None                                  None  1/1  255  79  79   \n",
       "54               None                                  None  1/1  255  79  79   \n",
       "55               None                                  None  1/1  255  79  79   \n",
       "56               None                                  None  1/1  255  79  79   \n",
       "57          c.1516T>C  ENSP00000478421.2        p.Trp506Arg  1/1  255  79  79   \n",
       "58           c.703T>C  ENSP00000484643.1        p.Trp235Arg  1/1  255  79  79   \n",
       "59          c.1030T>C  ENSP00000482090.2        p.Trp344Arg  1/1  255  79  79   \n",
       "60           c.652T>C  ENSP00000480870.1        p.Trp218Arg  1/1  255  79  79   \n",
       "61          c.1519T>C  ENSP00000480678.2        p.Trp507Arg  1/1  255  79  79   \n",
       "62           c.841T>C  ENSP00000484256.1        p.Trp281Arg  1/1  255  79  79   \n",
       "63          c.1030T>C  ENSP00000482138.1        p.Trp344Arg  1/1  255  79  79   \n",
       "64               None                                  None  1/1  255  79  79   \n",
       "\n",
       "   RD  AD  FREQ        PVAL RDF RDR ADF ADR                         SIFT  \\\n",
       "46  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "47  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "48  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "49  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "50  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "51  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "52  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "53  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "54  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "55  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "56  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "57  0  79  100%  4.3185e-47   0   0  38  41  tolerated low confidence(1)   \n",
       "58  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "59  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "60  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "61  0  79  100%  4.3185e-47   0   0  38  41  tolerated low confidence(1)   \n",
       "62  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "63  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "64  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "\n",
       "         PolyPhen                         BIOTYPE   EXON INTRON  \\\n",
       "46                                 protein coding                 \n",
       "47      benign(0)                  protein coding   8/12          \n",
       "48      benign(0)                  protein coding  10/14          \n",
       "49      benign(0)                  protein coding    4/7          \n",
       "50                                retained intron    1/2          \n",
       "51                                retained intron    2/2          \n",
       "52                                retained intron    3/4          \n",
       "53                                retained intron                 \n",
       "54                 protein coding CDS not defined                 \n",
       "55                                retained intron                 \n",
       "56                 protein coding CDS not defined                 \n",
       "57      benign(0)                  protein coding  10/14          \n",
       "58  benign(0.001)                  protein coding   7/11          \n",
       "59      benign(0)                  protein coding   9/13          \n",
       "60      benign(0)                  protein coding   6/10          \n",
       "61      benign(0)                  protein coding  10/14          \n",
       "62      benign(0)                  protein coding   8/12          \n",
       "63      benign(0)                  protein coding   9/13          \n",
       "64                                       promoter                 \n",
       "\n",
       "   Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "46                             nan              -1          \n",
       "47                           W251R  Tgg/Cgg      1          \n",
       "48                           W343R  Tgg/Cgg      1          \n",
       "49                           W170R  Tgg/Cgg      1          \n",
       "50                             nan               1          \n",
       "51                             nan               1          \n",
       "52                             nan               1          \n",
       "53                             nan              -1          \n",
       "54                             nan               1          \n",
       "55                             nan              -1          \n",
       "56                             nan              -1          \n",
       "57                           W506R  Tgg/Cgg      1          \n",
       "58                           W235R  Tgg/Cgg      1          \n",
       "59                           W344R  Tgg/Cgg      1          \n",
       "60                           W218R  Tgg/Cgg      1          \n",
       "61                           W507R  Tgg/Cgg      1          \n",
       "62                           W281R  Tgg/Cgg      1          \n",
       "63                           W344R  Tgg/Cgg      1          \n",
       "64                             nan                          \n",
       "\n",
       "                           consequence  \n",
       "46             downstream gene variant  \n",
       "47                    missense variant  \n",
       "48                    missense variant  \n",
       "49                    missense variant  \n",
       "50  non coding transcript exon variant  \n",
       "51  non coding transcript exon variant  \n",
       "52  non coding transcript exon variant  \n",
       "53             downstream gene variant  \n",
       "54             downstream gene variant  \n",
       "55             downstream gene variant  \n",
       "56             downstream gene variant  \n",
       "57                    missense variant  \n",
       "58                    missense variant  \n",
       "59                    missense variant  \n",
       "60                    missense variant  \n",
       "61                    missense variant  \n",
       "62                    missense variant  \n",
       "63                    missense variant  \n",
       "64           regulatory region variant  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_vcf[merged_vcf['ID'] == 'rs6672356']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "e3b3866a",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript ablation</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice acceptor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>stop lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>transcript amplification</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inframe insertion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>inframe deletion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein altering variant</td>\n",
       "      <td>5/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>splice region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>splice donor 5th base variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>splice donor region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>splice polypyrimidine tract variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>incomplete terminal codon variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>start retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>stop retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>coding sequence variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>mature miRNA variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TFBS ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TFBS amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>regulatory region ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>regulatory region amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>feature elongation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>feature truncation</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            consequence Consequence_score\n",
       "0                   transcript ablation             10/10\n",
       "1               splice acceptor variant              8/10\n",
       "2                  splice donor variant              8/10\n",
       "3                           stop gained             10/10\n",
       "4                    frameshift variant             10/10\n",
       "5                             stop lost              9/10\n",
       "6                            start lost              9/10\n",
       "7              transcript amplification              8/10\n",
       "8                     inframe insertion              6/10\n",
       "9                      inframe deletion              6/10\n",
       "10                     missense variant              7/10\n",
       "11             protein altering variant              5/10\n",
       "12                splice region variant              4/10\n",
       "13        splice donor 5th base variant              4/10\n",
       "14          splice donor region variant              4/10\n",
       "15  splice polypyrimidine tract variant              4/10\n",
       "16    incomplete terminal codon variant              3/10\n",
       "17               start retained variant              3/10\n",
       "18                stop retained variant              3/10\n",
       "19                   synonymous variant              3/10\n",
       "20              coding sequence variant              3/10\n",
       "21                 mature miRNA variant              3/10\n",
       "22                  5 prime UTR variant              3/10\n",
       "23                  3 prime UTR variant              3/10\n",
       "24   non coding transcript exon variant              2/10\n",
       "25                       intron variant              2/10\n",
       "26               NMD transcript variant              2/10\n",
       "27        non coding transcript variant              2/10\n",
       "28                upstream gene variant              2/10\n",
       "29              downstream gene variant              2/10\n",
       "30                        TFBS ablation              2/10\n",
       "31                   TFBS amplification              2/10\n",
       "32              TF binding site variant              2/10\n",
       "33           regulatory region ablation              2/10\n",
       "34      regulatory region amplification              2/10\n",
       "35                   feature elongation              2/10\n",
       "36            regulatory region variant              2/10\n",
       "37                   feature truncation              1/10\n",
       "38                   intergenic variant              1/10"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "1538e3e0",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM      POS REF ALT      Zygosity  \\\n",
       "0          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "1          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "2          OR4F5     rs2691305  chr1    69511   A   G    Homozygous   \n",
       "3          OR4F5  rs1245478362  chr1    69735   A   G  Heterozygous   \n",
       "4          OR4F5   rs200676709  chr1    69897   T   C  Heterozygous   \n",
       "...          ...           ...   ...      ...  ..  ..           ...   \n",
       "333893     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333894     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333895     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333896     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333897     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "\n",
       "                      Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0              synonymous variant       LOW                          \n",
       "1       regulatory region variant  MODIFIER                          \n",
       "2                missense variant  MODERATE                          \n",
       "3              synonymous variant       LOW                          \n",
       "4              synonymous variant       LOW                          \n",
       "...                           ...       ...           ...      ...   \n",
       "333893      upstream gene variant  MODIFIER                          \n",
       "333894           missense variant  MODERATE                          \n",
       "333895           missense variant  MODERATE                          \n",
       "333896      upstream gene variant  MODIFIER                          \n",
       "333897      upstream gene variant  MODIFIER                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                  ENST00000641515.2           c.243A>G   \n",
       "1                                                                  None   \n",
       "2                                  ENST00000641515.2           c.484A>G   \n",
       "3                                  ENST00000641515.2           c.708A>G   \n",
       "4                                  ENST00000641515.2           c.870T>C   \n",
       "...                   ...     ...                ...                ...   \n",
       "333893                                                             None   \n",
       "333894                             ENST00000423647.6           c.257G>A   \n",
       "333895                             ENST00000451548.6           c.257G>A   \n",
       "333896                                                             None   \n",
       "333897                                                             None   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD  \\\n",
       "0       ENSP00000493376.2         p.Ser81%3D  0/1  120  101  101   66   35   \n",
       "1                                       None  0/1  120  101  101   66   35   \n",
       "2       ENSP00000493376.2        p.Thr162Ala  1/1  255  143  143    0  143   \n",
       "3       ENSP00000493376.2        p.Leu236%3D  0/1  163  180  180  131   49   \n",
       "4       ENSP00000493376.2        p.Ser290%3D  0/1   75   96   96   73   23   \n",
       "...                   ...                ...  ...  ...  ...  ...  ...  ...   \n",
       "333893                                  None  0/1  255  383  383  273  110   \n",
       "333894  ENSP00000389324.3         p.Arg86Gln  0/1  255  383  383  273  110   \n",
       "333895  ENSP00000403304.1         p.Arg86Gln  0/1  255  383  383  273  110   \n",
       "333896                                  None  0/1  255  383  383  273  110   \n",
       "333897                                  None  0/1  255  383  383  273  110   \n",
       "\n",
       "          FREQ                   PVAL  RDF  RDR  ADF ADR             SIFT  \\\n",
       "0       34.65%             8.1311e-13   48   18   24  11                    \n",
       "1       34.65%             8.1311e-13   48   18   24  11                    \n",
       "2         100%             1.7063e-85    0    0  100  43  tolerated(0.92)   \n",
       "3       27.22%             4.0111e-17   84   47   39  10                    \n",
       "4       23.96%             2.6678e-08   30   43   13  10                    \n",
       "...        ...                    ...  ...  ...  ...  ..              ...   \n",
       "333893  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "333894  28.72%  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333895  28.72%  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333896  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "333897  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "\n",
       "             PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0                                          protein coding  3/3          \n",
       "1                                         TF binding site               \n",
       "2           benign(0)                      protein coding  3/3          \n",
       "3                                          protein coding  3/3          \n",
       "4                                          protein coding  3/3          \n",
       "...               ...                                 ...  ...    ...   \n",
       "333893                 transcribed unprocessed pseudogene               \n",
       "333894  benign(0.375)                      protein coding  1/6          \n",
       "333895  benign(0.101)                      protein coding  1/6          \n",
       "333896                                             lncRNA               \n",
       "333897                                             lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                  S81  tcA/tcG      1          \n",
       "1                                  nan                          \n",
       "2                                T162A  Aca/Gca      1          \n",
       "3                                 L236  ctA/ctG      1          \n",
       "4                                 S290  tcT/tcC      1          \n",
       "...                                ...      ...    ...    ...   \n",
       "333893                             nan              -1          \n",
       "333894                            R86Q  cGg/cAg      1          \n",
       "333895                            R86Q  cGg/cAg      1          \n",
       "333896                             nan              -1          \n",
       "333897                             nan              -1          \n",
       "\n",
       "                      consequence Consequence_score  \n",
       "0              synonymous variant              3/10  \n",
       "1       regulatory region variant              2/10  \n",
       "2                missense variant              7/10  \n",
       "3              synonymous variant              3/10  \n",
       "4              synonymous variant              3/10  \n",
       "...                           ...               ...  \n",
       "333893      upstream gene variant              2/10  \n",
       "333894           missense variant              7/10  \n",
       "333895           missense variant              7/10  \n",
       "333896      upstream gene variant              2/10  \n",
       "333897      upstream gene variant              2/10  \n",
       "\n",
       "[333898 rows x 40 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.merge(merged_vcf, df_1, on='consequence', how='left', sort=False)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "e35b54be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.751T&gt;C</td>\n",
       "      <td>ENSP00000349216.4</td>\n",
       "      <td>p.Trp251Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/12</td>\n",
       "      <td></td>\n",
       "      <td>W251R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>c.1027T&gt;C</td>\n",
       "      <td>ENSP00000342313.3</td>\n",
       "      <td>p.Trp343Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W343R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000455979.1</td>\n",
       "      <td>c.508T&gt;C</td>\n",
       "      <td>ENSP00000412228.1</td>\n",
       "      <td>p.Trp170Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>4/7</td>\n",
       "      <td></td>\n",
       "      <td>W170R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000464948.1</td>\n",
       "      <td>n.286T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>1/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000466827.1</td>\n",
       "      <td>n.191T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000474461.1</td>\n",
       "      <td>n.389T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>3/4</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000616016.5</td>\n",
       "      <td>c.1516T&gt;C</td>\n",
       "      <td>ENSP00000478421.2</td>\n",
       "      <td>p.Trp506Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W506R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000616125.5</td>\n",
       "      <td>c.703T&gt;C</td>\n",
       "      <td>ENSP00000484643.1</td>\n",
       "      <td>p.Trp235Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/11</td>\n",
       "      <td></td>\n",
       "      <td>W235R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000617307.5</td>\n",
       "      <td>c.1030T&gt;C</td>\n",
       "      <td>ENSP00000482090.2</td>\n",
       "      <td>p.Trp344Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>W344R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618181.5</td>\n",
       "      <td>c.652T&gt;C</td>\n",
       "      <td>ENSP00000480870.1</td>\n",
       "      <td>p.Trp218Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/10</td>\n",
       "      <td></td>\n",
       "      <td>W218R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618323.5</td>\n",
       "      <td>c.1519T&gt;C</td>\n",
       "      <td>ENSP00000480678.2</td>\n",
       "      <td>p.Trp507Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W507R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618779.5</td>\n",
       "      <td>c.841T&gt;C</td>\n",
       "      <td>ENSP00000484256.1</td>\n",
       "      <td>p.Trp281Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/12</td>\n",
       "      <td></td>\n",
       "      <td>W281R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000622503.5</td>\n",
       "      <td>c.1030T&gt;C</td>\n",
       "      <td>ENSP00000482138.1</td>\n",
       "      <td>p.Trp344Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>W344R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name         ID CHROM     POS REF ALT    Zygosity  \\\n",
       "46    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "47    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "48    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "49    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "50    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "51    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "52    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "53    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "54    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "55    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "56    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "57    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "58    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "59    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "60    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "61    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "62    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "63    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "64    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "\n",
       "                           Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "46             downstream gene variant  MODIFIER  not provided   benign   \n",
       "47                    missense variant  MODERATE  not provided   benign   \n",
       "48                    missense variant  MODERATE  not provided   benign   \n",
       "49                    missense variant  MODERATE  not provided   benign   \n",
       "50  non coding transcript exon variant  MODIFIER  not provided   benign   \n",
       "51  non coding transcript exon variant  MODIFIER  not provided   benign   \n",
       "52  non coding transcript exon variant  MODIFIER  not provided   benign   \n",
       "53             downstream gene variant  MODIFIER  not provided   benign   \n",
       "54             downstream gene variant  MODIFIER  not provided   benign   \n",
       "55             downstream gene variant  MODIFIER  not provided   benign   \n",
       "56             downstream gene variant  MODIFIER  not provided   benign   \n",
       "57                    missense variant  MODERATE  not provided   benign   \n",
       "58                    missense variant  MODERATE  not provided   benign   \n",
       "59                    missense variant  MODERATE  not provided   benign   \n",
       "60                    missense variant  MODERATE  not provided   benign   \n",
       "61                    missense variant  MODERATE  not provided   benign   \n",
       "62                    missense variant  MODERATE  not provided   benign   \n",
       "63                    missense variant  MODERATE  not provided   benign   \n",
       "64           regulatory region variant  MODIFIER  not provided   benign   \n",
       "\n",
       "                     ClinVar_CLNREVSTAT  ClinVar              HGVSc  \\\n",
       "46  criteria provided, single submitter  1166513                      \n",
       "47  criteria provided, single submitter  1166513  ENST00000341065.8   \n",
       "48  criteria provided, single submitter  1166513  ENST00000342066.8   \n",
       "49  criteria provided, single submitter  1166513  ENST00000455979.1   \n",
       "50  criteria provided, single submitter  1166513  ENST00000464948.1   \n",
       "51  criteria provided, single submitter  1166513  ENST00000466827.1   \n",
       "52  criteria provided, single submitter  1166513  ENST00000474461.1   \n",
       "53  criteria provided, single submitter  1166513                      \n",
       "54  criteria provided, single submitter  1166513                      \n",
       "55  criteria provided, single submitter  1166513                      \n",
       "56  criteria provided, single submitter  1166513                      \n",
       "57  criteria provided, single submitter  1166513  ENST00000616016.5   \n",
       "58  criteria provided, single submitter  1166513  ENST00000616125.5   \n",
       "59  criteria provided, single submitter  1166513  ENST00000617307.5   \n",
       "60  criteria provided, single submitter  1166513  ENST00000618181.5   \n",
       "61  criteria provided, single submitter  1166513  ENST00000618323.5   \n",
       "62  criteria provided, single submitter  1166513  ENST00000618779.5   \n",
       "63  criteria provided, single submitter  1166513  ENST00000622503.5   \n",
       "64  criteria provided, single submitter  1166513                      \n",
       "\n",
       "   HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  \\\n",
       "46               None                                  None  1/1  255  79  79   \n",
       "47           c.751T>C  ENSP00000349216.4        p.Trp251Arg  1/1  255  79  79   \n",
       "48          c.1027T>C  ENSP00000342313.3        p.Trp343Arg  1/1  255  79  79   \n",
       "49           c.508T>C  ENSP00000412228.1        p.Trp170Arg  1/1  255  79  79   \n",
       "50           n.286T>C                                  None  1/1  255  79  79   \n",
       "51           n.191T>C                                  None  1/1  255  79  79   \n",
       "52           n.389T>C                                  None  1/1  255  79  79   \n",
       "53               None                                  None  1/1  255  79  79   \n",
       "54               None                                  None  1/1  255  79  79   \n",
       "55               None                                  None  1/1  255  79  79   \n",
       "56               None                                  None  1/1  255  79  79   \n",
       "57          c.1516T>C  ENSP00000478421.2        p.Trp506Arg  1/1  255  79  79   \n",
       "58           c.703T>C  ENSP00000484643.1        p.Trp235Arg  1/1  255  79  79   \n",
       "59          c.1030T>C  ENSP00000482090.2        p.Trp344Arg  1/1  255  79  79   \n",
       "60           c.652T>C  ENSP00000480870.1        p.Trp218Arg  1/1  255  79  79   \n",
       "61          c.1519T>C  ENSP00000480678.2        p.Trp507Arg  1/1  255  79  79   \n",
       "62           c.841T>C  ENSP00000484256.1        p.Trp281Arg  1/1  255  79  79   \n",
       "63          c.1030T>C  ENSP00000482138.1        p.Trp344Arg  1/1  255  79  79   \n",
       "64               None                                  None  1/1  255  79  79   \n",
       "\n",
       "   RD  AD  FREQ        PVAL RDF RDR ADF ADR                         SIFT  \\\n",
       "46  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "47  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "48  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "49  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "50  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "51  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "52  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "53  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "54  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "55  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "56  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "57  0  79  100%  4.3185e-47   0   0  38  41  tolerated low confidence(1)   \n",
       "58  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "59  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "60  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "61  0  79  100%  4.3185e-47   0   0  38  41  tolerated low confidence(1)   \n",
       "62  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "63  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "64  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "\n",
       "         PolyPhen                         BIOTYPE   EXON INTRON  \\\n",
       "46                                 protein coding                 \n",
       "47      benign(0)                  protein coding   8/12          \n",
       "48      benign(0)                  protein coding  10/14          \n",
       "49      benign(0)                  protein coding    4/7          \n",
       "50                                retained intron    1/2          \n",
       "51                                retained intron    2/2          \n",
       "52                                retained intron    3/4          \n",
       "53                                retained intron                 \n",
       "54                 protein coding CDS not defined                 \n",
       "55                                retained intron                 \n",
       "56                 protein coding CDS not defined                 \n",
       "57      benign(0)                  protein coding  10/14          \n",
       "58  benign(0.001)                  protein coding   7/11          \n",
       "59      benign(0)                  protein coding   9/13          \n",
       "60      benign(0)                  protein coding   6/10          \n",
       "61      benign(0)                  protein coding  10/14          \n",
       "62      benign(0)                  protein coding   8/12          \n",
       "63      benign(0)                  protein coding   9/13          \n",
       "64                                       promoter                 \n",
       "\n",
       "   Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "46                             nan              -1          \n",
       "47                           W251R  Tgg/Cgg      1          \n",
       "48                           W343R  Tgg/Cgg      1          \n",
       "49                           W170R  Tgg/Cgg      1          \n",
       "50                             nan               1          \n",
       "51                             nan               1          \n",
       "52                             nan               1          \n",
       "53                             nan              -1          \n",
       "54                             nan               1          \n",
       "55                             nan              -1          \n",
       "56                             nan              -1          \n",
       "57                           W506R  Tgg/Cgg      1          \n",
       "58                           W235R  Tgg/Cgg      1          \n",
       "59                           W344R  Tgg/Cgg      1          \n",
       "60                           W218R  Tgg/Cgg      1          \n",
       "61                           W507R  Tgg/Cgg      1          \n",
       "62                           W281R  Tgg/Cgg      1          \n",
       "63                           W344R  Tgg/Cgg      1          \n",
       "64                             nan                          \n",
       "\n",
       "                           consequence Consequence_score  \n",
       "46             downstream gene variant              2/10  \n",
       "47                    missense variant              7/10  \n",
       "48                    missense variant              7/10  \n",
       "49                    missense variant              7/10  \n",
       "50  non coding transcript exon variant              2/10  \n",
       "51  non coding transcript exon variant              2/10  \n",
       "52  non coding transcript exon variant              2/10  \n",
       "53             downstream gene variant              2/10  \n",
       "54             downstream gene variant              2/10  \n",
       "55             downstream gene variant              2/10  \n",
       "56             downstream gene variant              2/10  \n",
       "57                    missense variant              7/10  \n",
       "58                    missense variant              7/10  \n",
       "59                    missense variant              7/10  \n",
       "60                    missense variant              7/10  \n",
       "61                    missense variant              7/10  \n",
       "62                    missense variant              7/10  \n",
       "63                    missense variant              7/10  \n",
       "64           regulatory region variant              2/10  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1[merged_1['ID'] == 'rs6672356']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "1543e770",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HIGH</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     IMPACT  IMPACT_score\n",
       "0      HIGH          10.0\n",
       "1  MODERATE           5.0\n",
       "2       LOW           2.5\n",
       "3  MODIFIER           1.5"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "2e96b6cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM      POS REF ALT      Zygosity  \\\n",
       "0          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "1          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "2          OR4F5     rs2691305  chr1    69511   A   G    Homozygous   \n",
       "3          OR4F5  rs1245478362  chr1    69735   A   G  Heterozygous   \n",
       "4          OR4F5   rs200676709  chr1    69897   T   C  Heterozygous   \n",
       "...          ...           ...   ...      ...  ..  ..           ...   \n",
       "333893     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333894     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333895     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333896     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333897     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "\n",
       "                      Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "0              synonymous variant       LOW                          \n",
       "1       regulatory region variant  MODIFIER                          \n",
       "2                missense variant  MODERATE                          \n",
       "3              synonymous variant       LOW                          \n",
       "4              synonymous variant       LOW                          \n",
       "...                           ...       ...           ...      ...   \n",
       "333893      upstream gene variant  MODIFIER                          \n",
       "333894           missense variant  MODERATE                          \n",
       "333895           missense variant  MODERATE                          \n",
       "333896      upstream gene variant  MODIFIER                          \n",
       "333897      upstream gene variant  MODIFIER                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                  ENST00000641515.2           c.243A>G   \n",
       "1                                                                  None   \n",
       "2                                  ENST00000641515.2           c.484A>G   \n",
       "3                                  ENST00000641515.2           c.708A>G   \n",
       "4                                  ENST00000641515.2           c.870T>C   \n",
       "...                   ...     ...                ...                ...   \n",
       "333893                                                             None   \n",
       "333894                             ENST00000423647.6           c.257G>A   \n",
       "333895                             ENST00000451548.6           c.257G>A   \n",
       "333896                                                             None   \n",
       "333897                                                             None   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD  \\\n",
       "0       ENSP00000493376.2         p.Ser81%3D  0/1  120  101  101   66   35   \n",
       "1                                       None  0/1  120  101  101   66   35   \n",
       "2       ENSP00000493376.2        p.Thr162Ala  1/1  255  143  143    0  143   \n",
       "3       ENSP00000493376.2        p.Leu236%3D  0/1  163  180  180  131   49   \n",
       "4       ENSP00000493376.2        p.Ser290%3D  0/1   75   96   96   73   23   \n",
       "...                   ...                ...  ...  ...  ...  ...  ...  ...   \n",
       "333893                                  None  0/1  255  383  383  273  110   \n",
       "333894  ENSP00000389324.3         p.Arg86Gln  0/1  255  383  383  273  110   \n",
       "333895  ENSP00000403304.1         p.Arg86Gln  0/1  255  383  383  273  110   \n",
       "333896                                  None  0/1  255  383  383  273  110   \n",
       "333897                                  None  0/1  255  383  383  273  110   \n",
       "\n",
       "          FREQ                   PVAL  RDF  RDR  ADF ADR             SIFT  \\\n",
       "0       34.65%             8.1311e-13   48   18   24  11                    \n",
       "1       34.65%             8.1311e-13   48   18   24  11                    \n",
       "2         100%             1.7063e-85    0    0  100  43  tolerated(0.92)   \n",
       "3       27.22%             4.0111e-17   84   47   39  10                    \n",
       "4       23.96%             2.6678e-08   30   43   13  10                    \n",
       "...        ...                    ...  ...  ...  ...  ..              ...   \n",
       "333893  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "333894  28.72%  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333895  28.72%  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333896  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "333897  28.72%  7.984299999999999e-38  136  137   83  27                    \n",
       "\n",
       "             PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0                                          protein coding  3/3          \n",
       "1                                         TF binding site               \n",
       "2           benign(0)                      protein coding  3/3          \n",
       "3                                          protein coding  3/3          \n",
       "4                                          protein coding  3/3          \n",
       "...               ...                                 ...  ...    ...   \n",
       "333893                 transcribed unprocessed pseudogene               \n",
       "333894  benign(0.375)                      protein coding  1/6          \n",
       "333895  benign(0.101)                      protein coding  1/6          \n",
       "333896                                             lncRNA               \n",
       "333897                                             lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0                                  S81  tcA/tcG      1          \n",
       "1                                  nan                          \n",
       "2                                T162A  Aca/Gca      1          \n",
       "3                                 L236  ctA/ctG      1          \n",
       "4                                 S290  tcT/tcC      1          \n",
       "...                                ...      ...    ...    ...   \n",
       "333893                             nan              -1          \n",
       "333894                            R86Q  cGg/cAg      1          \n",
       "333895                            R86Q  cGg/cAg      1          \n",
       "333896                             nan              -1          \n",
       "333897                             nan              -1          \n",
       "\n",
       "                      consequence Consequence_score  IMPACT_score  \n",
       "0              synonymous variant              3/10           2.5  \n",
       "1       regulatory region variant              2/10           1.5  \n",
       "2                missense variant              7/10           5.0  \n",
       "3              synonymous variant              3/10           2.5  \n",
       "4              synonymous variant              3/10           2.5  \n",
       "...                           ...               ...           ...  \n",
       "333893      upstream gene variant              2/10           1.5  \n",
       "333894           missense variant              7/10           5.0  \n",
       "333895           missense variant              7/10           5.0  \n",
       "333896      upstream gene variant              2/10           1.5  \n",
       "333897      upstream gene variant              2/10           1.5  \n",
       "\n",
       "[333898 rows x 41 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "771c61be",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.751T&gt;C</td>\n",
       "      <td>ENSP00000349216.4</td>\n",
       "      <td>p.Trp251Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/12</td>\n",
       "      <td></td>\n",
       "      <td>W251R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>c.1027T&gt;C</td>\n",
       "      <td>ENSP00000342313.3</td>\n",
       "      <td>p.Trp343Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W343R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000455979.1</td>\n",
       "      <td>c.508T&gt;C</td>\n",
       "      <td>ENSP00000412228.1</td>\n",
       "      <td>p.Trp170Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>4/7</td>\n",
       "      <td></td>\n",
       "      <td>W170R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000464948.1</td>\n",
       "      <td>n.286T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>1/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000466827.1</td>\n",
       "      <td>n.191T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000474461.1</td>\n",
       "      <td>n.389T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>3/4</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000616016.5</td>\n",
       "      <td>c.1516T&gt;C</td>\n",
       "      <td>ENSP00000478421.2</td>\n",
       "      <td>p.Trp506Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W506R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000616125.5</td>\n",
       "      <td>c.703T&gt;C</td>\n",
       "      <td>ENSP00000484643.1</td>\n",
       "      <td>p.Trp235Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/11</td>\n",
       "      <td></td>\n",
       "      <td>W235R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000617307.5</td>\n",
       "      <td>c.1030T&gt;C</td>\n",
       "      <td>ENSP00000482090.2</td>\n",
       "      <td>p.Trp344Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>W344R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618181.5</td>\n",
       "      <td>c.652T&gt;C</td>\n",
       "      <td>ENSP00000480870.1</td>\n",
       "      <td>p.Trp218Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/10</td>\n",
       "      <td></td>\n",
       "      <td>W218R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618323.5</td>\n",
       "      <td>c.1519T&gt;C</td>\n",
       "      <td>ENSP00000480678.2</td>\n",
       "      <td>p.Trp507Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W507R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618779.5</td>\n",
       "      <td>c.841T&gt;C</td>\n",
       "      <td>ENSP00000484256.1</td>\n",
       "      <td>p.Trp281Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/12</td>\n",
       "      <td></td>\n",
       "      <td>W281R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000622503.5</td>\n",
       "      <td>c.1030T&gt;C</td>\n",
       "      <td>ENSP00000482138.1</td>\n",
       "      <td>p.Trp344Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>W344R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name         ID CHROM     POS REF ALT    Zygosity  \\\n",
       "46    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "47    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "48    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "49    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "50    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "51    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "52    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "53    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "54    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "55    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "56    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "57    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "58    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "59    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "60    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "61    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "62    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "63    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "64    SAMD11  rs6672356  chr1  942451   T   C  Homozygous   \n",
       "\n",
       "                           Consequence    IMPACT ClinVar_CLNDN CLIN_SIG  \\\n",
       "46             downstream gene variant  MODIFIER  not provided   benign   \n",
       "47                    missense variant  MODERATE  not provided   benign   \n",
       "48                    missense variant  MODERATE  not provided   benign   \n",
       "49                    missense variant  MODERATE  not provided   benign   \n",
       "50  non coding transcript exon variant  MODIFIER  not provided   benign   \n",
       "51  non coding transcript exon variant  MODIFIER  not provided   benign   \n",
       "52  non coding transcript exon variant  MODIFIER  not provided   benign   \n",
       "53             downstream gene variant  MODIFIER  not provided   benign   \n",
       "54             downstream gene variant  MODIFIER  not provided   benign   \n",
       "55             downstream gene variant  MODIFIER  not provided   benign   \n",
       "56             downstream gene variant  MODIFIER  not provided   benign   \n",
       "57                    missense variant  MODERATE  not provided   benign   \n",
       "58                    missense variant  MODERATE  not provided   benign   \n",
       "59                    missense variant  MODERATE  not provided   benign   \n",
       "60                    missense variant  MODERATE  not provided   benign   \n",
       "61                    missense variant  MODERATE  not provided   benign   \n",
       "62                    missense variant  MODERATE  not provided   benign   \n",
       "63                    missense variant  MODERATE  not provided   benign   \n",
       "64           regulatory region variant  MODIFIER  not provided   benign   \n",
       "\n",
       "                     ClinVar_CLNREVSTAT  ClinVar              HGVSc  \\\n",
       "46  criteria provided, single submitter  1166513                      \n",
       "47  criteria provided, single submitter  1166513  ENST00000341065.8   \n",
       "48  criteria provided, single submitter  1166513  ENST00000342066.8   \n",
       "49  criteria provided, single submitter  1166513  ENST00000455979.1   \n",
       "50  criteria provided, single submitter  1166513  ENST00000464948.1   \n",
       "51  criteria provided, single submitter  1166513  ENST00000466827.1   \n",
       "52  criteria provided, single submitter  1166513  ENST00000474461.1   \n",
       "53  criteria provided, single submitter  1166513                      \n",
       "54  criteria provided, single submitter  1166513                      \n",
       "55  criteria provided, single submitter  1166513                      \n",
       "56  criteria provided, single submitter  1166513                      \n",
       "57  criteria provided, single submitter  1166513  ENST00000616016.5   \n",
       "58  criteria provided, single submitter  1166513  ENST00000616125.5   \n",
       "59  criteria provided, single submitter  1166513  ENST00000617307.5   \n",
       "60  criteria provided, single submitter  1166513  ENST00000618181.5   \n",
       "61  criteria provided, single submitter  1166513  ENST00000618323.5   \n",
       "62  criteria provided, single submitter  1166513  ENST00000618779.5   \n",
       "63  criteria provided, single submitter  1166513  ENST00000622503.5   \n",
       "64  criteria provided, single submitter  1166513                      \n",
       "\n",
       "   HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  \\\n",
       "46               None                                  None  1/1  255  79  79   \n",
       "47           c.751T>C  ENSP00000349216.4        p.Trp251Arg  1/1  255  79  79   \n",
       "48          c.1027T>C  ENSP00000342313.3        p.Trp343Arg  1/1  255  79  79   \n",
       "49           c.508T>C  ENSP00000412228.1        p.Trp170Arg  1/1  255  79  79   \n",
       "50           n.286T>C                                  None  1/1  255  79  79   \n",
       "51           n.191T>C                                  None  1/1  255  79  79   \n",
       "52           n.389T>C                                  None  1/1  255  79  79   \n",
       "53               None                                  None  1/1  255  79  79   \n",
       "54               None                                  None  1/1  255  79  79   \n",
       "55               None                                  None  1/1  255  79  79   \n",
       "56               None                                  None  1/1  255  79  79   \n",
       "57          c.1516T>C  ENSP00000478421.2        p.Trp506Arg  1/1  255  79  79   \n",
       "58           c.703T>C  ENSP00000484643.1        p.Trp235Arg  1/1  255  79  79   \n",
       "59          c.1030T>C  ENSP00000482090.2        p.Trp344Arg  1/1  255  79  79   \n",
       "60           c.652T>C  ENSP00000480870.1        p.Trp218Arg  1/1  255  79  79   \n",
       "61          c.1519T>C  ENSP00000480678.2        p.Trp507Arg  1/1  255  79  79   \n",
       "62           c.841T>C  ENSP00000484256.1        p.Trp281Arg  1/1  255  79  79   \n",
       "63          c.1030T>C  ENSP00000482138.1        p.Trp344Arg  1/1  255  79  79   \n",
       "64               None                                  None  1/1  255  79  79   \n",
       "\n",
       "   RD  AD  FREQ        PVAL RDF RDR ADF ADR                         SIFT  \\\n",
       "46  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "47  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "48  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "49  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "50  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "51  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "52  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "53  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "54  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "55  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "56  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "57  0  79  100%  4.3185e-47   0   0  38  41  tolerated low confidence(1)   \n",
       "58  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "59  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "60  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "61  0  79  100%  4.3185e-47   0   0  38  41  tolerated low confidence(1)   \n",
       "62  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "63  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "64  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "\n",
       "         PolyPhen                         BIOTYPE   EXON INTRON  \\\n",
       "46                                 protein coding                 \n",
       "47      benign(0)                  protein coding   8/12          \n",
       "48      benign(0)                  protein coding  10/14          \n",
       "49      benign(0)                  protein coding    4/7          \n",
       "50                                retained intron    1/2          \n",
       "51                                retained intron    2/2          \n",
       "52                                retained intron    3/4          \n",
       "53                                retained intron                 \n",
       "54                 protein coding CDS not defined                 \n",
       "55                                retained intron                 \n",
       "56                 protein coding CDS not defined                 \n",
       "57      benign(0)                  protein coding  10/14          \n",
       "58  benign(0.001)                  protein coding   7/11          \n",
       "59      benign(0)                  protein coding   9/13          \n",
       "60      benign(0)                  protein coding   6/10          \n",
       "61      benign(0)                  protein coding  10/14          \n",
       "62      benign(0)                  protein coding   8/12          \n",
       "63      benign(0)                  protein coding   9/13          \n",
       "64                                       promoter                 \n",
       "\n",
       "   Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "46                             nan              -1          \n",
       "47                           W251R  Tgg/Cgg      1          \n",
       "48                           W343R  Tgg/Cgg      1          \n",
       "49                           W170R  Tgg/Cgg      1          \n",
       "50                             nan               1          \n",
       "51                             nan               1          \n",
       "52                             nan               1          \n",
       "53                             nan              -1          \n",
       "54                             nan               1          \n",
       "55                             nan              -1          \n",
       "56                             nan              -1          \n",
       "57                           W506R  Tgg/Cgg      1          \n",
       "58                           W235R  Tgg/Cgg      1          \n",
       "59                           W344R  Tgg/Cgg      1          \n",
       "60                           W218R  Tgg/Cgg      1          \n",
       "61                           W507R  Tgg/Cgg      1          \n",
       "62                           W281R  Tgg/Cgg      1          \n",
       "63                           W344R  Tgg/Cgg      1          \n",
       "64                             nan                          \n",
       "\n",
       "                           consequence Consequence_score  IMPACT_score  \n",
       "46             downstream gene variant              2/10           1.5  \n",
       "47                    missense variant              7/10           5.0  \n",
       "48                    missense variant              7/10           5.0  \n",
       "49                    missense variant              7/10           5.0  \n",
       "50  non coding transcript exon variant              2/10           1.5  \n",
       "51  non coding transcript exon variant              2/10           1.5  \n",
       "52  non coding transcript exon variant              2/10           1.5  \n",
       "53             downstream gene variant              2/10           1.5  \n",
       "54             downstream gene variant              2/10           1.5  \n",
       "55             downstream gene variant              2/10           1.5  \n",
       "56             downstream gene variant              2/10           1.5  \n",
       "57                    missense variant              7/10           5.0  \n",
       "58                    missense variant              7/10           5.0  \n",
       "59                    missense variant              7/10           5.0  \n",
       "60                    missense variant              7/10           5.0  \n",
       "61                    missense variant              7/10           5.0  \n",
       "62                    missense variant              7/10           5.0  \n",
       "63                    missense variant              7/10           5.0  \n",
       "64           regulatory region variant              2/10           1.5  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2[merged_2['ID'] == 'rs6672356']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "f903ba0f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM      POS REF ALT      Zygosity  \\\n",
       "0          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "1          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "2          OR4F5     rs2691305  chr1    69511   A   G    Homozygous   \n",
       "3          OR4F5  rs1245478362  chr1    69735   A   G  Heterozygous   \n",
       "4          OR4F5   rs200676709  chr1    69897   T   C  Heterozygous   \n",
       "...          ...           ...   ...      ...  ..  ..           ...   \n",
       "333893     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333894     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333895     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333896     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333897     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "\n",
       "                      Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0              synonymous variant              3/10       LOW           2.5   \n",
       "1       regulatory region variant              2/10  MODIFIER           1.5   \n",
       "2                missense variant              7/10  MODERATE           5.0   \n",
       "3              synonymous variant              3/10       LOW           2.5   \n",
       "4              synonymous variant              3/10       LOW           2.5   \n",
       "...                           ...               ...       ...           ...   \n",
       "333893      upstream gene variant              2/10  MODIFIER           1.5   \n",
       "333894           missense variant              7/10  MODERATE           5.0   \n",
       "333895           missense variant              7/10  MODERATE           5.0   \n",
       "333896      upstream gene variant              2/10  MODIFIER           1.5   \n",
       "333897      upstream gene variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                                         ENST00000641515.2   \n",
       "1                                                                             \n",
       "2                                                         ENST00000641515.2   \n",
       "3                                                         ENST00000641515.2   \n",
       "4                                                         ENST00000641515.2   \n",
       "...              ...      ...                ...     ...                ...   \n",
       "333893                                                                        \n",
       "333894                                                    ENST00000423647.6   \n",
       "333895                                                    ENST00000451548.6   \n",
       "333896                                                                        \n",
       "333897                                                                        \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  120   \n",
       "1                    None                                  None  0/1  120   \n",
       "2                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "3                c.708A>G  ENSP00000493376.2        p.Leu236%3D  0/1  163   \n",
       "4                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   75   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "333893               None                                  None  0/1  255   \n",
       "333894           c.257G>A  ENSP00000389324.3         p.Arg86Gln  0/1  255   \n",
       "333895           c.257G>A  ENSP00000403304.1         p.Arg86Gln  0/1  255   \n",
       "333896               None                                  None  0/1  255   \n",
       "333897               None                                  None  0/1  255   \n",
       "\n",
       "        SDP   DP   RD   AD    FREQ                   PVAL  RDF  RDR  ADF ADR  \\\n",
       "0       101  101   66   35  34.65%             8.1311e-13   48   18   24  11   \n",
       "1       101  101   66   35  34.65%             8.1311e-13   48   18   24  11   \n",
       "2       143  143    0  143    100%             1.7063e-85    0    0  100  43   \n",
       "3       180  180  131   49  27.22%             4.0111e-17   84   47   39  10   \n",
       "4        96   96   73   23  23.96%             2.6678e-08   30   43   13  10   \n",
       "...     ...  ...  ...  ...     ...                    ...  ...  ...  ...  ..   \n",
       "333893  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "333894  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "333895  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "333896  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "333897  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "\n",
       "                   SIFT       PolyPhen                             BIOTYPE  \\\n",
       "0                                                           protein coding   \n",
       "1                                                          TF binding site   \n",
       "2       tolerated(0.92)      benign(0)                      protein coding   \n",
       "3                                                           protein coding   \n",
       "4                                                           protein coding   \n",
       "...                 ...            ...                                 ...   \n",
       "333893                                  transcribed unprocessed pseudogene   \n",
       "333894     tolerated(1)  benign(0.375)                      protein coding   \n",
       "333895     tolerated(1)  benign(0.101)                      protein coding   \n",
       "333896                                                              lncRNA   \n",
       "333897                                                              lncRNA   \n",
       "\n",
       "       EXON INTRON Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0       3/3                                    S81  tcA/tcG      1         \n",
       "1                                              nan                         \n",
       "2       3/3                                  T162A  Aca/Gca      1         \n",
       "3       3/3                                   L236  ctA/ctG      1         \n",
       "4       3/3                                   S290  tcT/tcC      1         \n",
       "...     ...    ...                             ...      ...    ...    ...  \n",
       "333893                                         nan              -1         \n",
       "333894  1/6                                   R86Q  cGg/cAg      1         \n",
       "333895  1/6                                   R86Q  cGg/cAg      1         \n",
       "333896                                         nan              -1         \n",
       "333897                                         nan              -1         \n",
       "\n",
       "[333898 rows x 40 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'ID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "b560ec50",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HLADR4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HLADR7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HLADQ7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HLADR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HLAB13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HLADQW3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HLACW6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HLAA30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AIS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>FOXD3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NALP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CXCL10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HLAA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PTPN22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>LPP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name\n",
       "0     HLADR4\n",
       "1     HLADR7\n",
       "2     HLADQ7\n",
       "3     HLADR1\n",
       "4     HLAB13\n",
       "5    HLADQW3\n",
       "6     HLACW6\n",
       "7     HLAA30\n",
       "8       AIS1\n",
       "9      FOXD3\n",
       "10     NALP1\n",
       "11    CXCL10\n",
       "12     HLAA2\n",
       "13    PTPN22\n",
       "14       LPP"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Vitiligo_genes.xlsx')\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "a4a2bc08",
   "metadata": {},
   "outputs": [],
   "source": [
    "#merged_2 = pd.merge(merged_2, df_gene, on = 'Gene Name', how='left', sort=False)\n",
    "#merged_2['Gene_Match_Condition'] = merged_2['Gene_Match_Condition'].fillna('No')\n",
    "#merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "704518f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name            ID CHROM      POS REF ALT      Zygosity  \\\n",
       "0          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "1          OR4F5   rs201219564  chr1    69270   A   G  Heterozygous   \n",
       "2          OR4F5     rs2691305  chr1    69511   A   G    Homozygous   \n",
       "3          OR4F5  rs1245478362  chr1    69735   A   G  Heterozygous   \n",
       "4          OR4F5   rs200676709  chr1    69897   T   C  Heterozygous   \n",
       "...          ...           ...   ...      ...  ..  ..           ...   \n",
       "333893     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333894     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333895     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333896     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "333897     TSPY1   rs777840135  chrY  9467257   G   A  Heterozygous   \n",
       "\n",
       "                      Consequence Consequence_score    IMPACT  IMPACT_score  \\\n",
       "0              synonymous variant              3/10       LOW           2.5   \n",
       "1       regulatory region variant              2/10  MODIFIER           1.5   \n",
       "2                missense variant              7/10  MODERATE           5.0   \n",
       "3              synonymous variant              3/10       LOW           2.5   \n",
       "4              synonymous variant              3/10       LOW           2.5   \n",
       "...                           ...               ...       ...           ...   \n",
       "333893      upstream gene variant              2/10  MODIFIER           1.5   \n",
       "333894           missense variant              7/10  MODERATE           5.0   \n",
       "333895           missense variant              7/10  MODERATE           5.0   \n",
       "333896      upstream gene variant              2/10  MODIFIER           1.5   \n",
       "333897      upstream gene variant              2/10  MODIFIER           1.5   \n",
       "\n",
       "       ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                                         ENST00000641515.2   \n",
       "1                                                                             \n",
       "2                                                         ENST00000641515.2   \n",
       "3                                                         ENST00000641515.2   \n",
       "4                                                         ENST00000641515.2   \n",
       "...              ...      ...                ...     ...                ...   \n",
       "333893                                                                        \n",
       "333894                                                    ENST00000423647.6   \n",
       "333895                                                    ENST00000451548.6   \n",
       "333896                                                                        \n",
       "333897                                                                        \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  120   \n",
       "1                    None                                  None  0/1  120   \n",
       "2                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "3                c.708A>G  ENSP00000493376.2        p.Leu236%3D  0/1  163   \n",
       "4                c.870T>C  ENSP00000493376.2        p.Ser290%3D  0/1   75   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "333893               None                                  None  0/1  255   \n",
       "333894           c.257G>A  ENSP00000389324.3         p.Arg86Gln  0/1  255   \n",
       "333895           c.257G>A  ENSP00000403304.1         p.Arg86Gln  0/1  255   \n",
       "333896               None                                  None  0/1  255   \n",
       "333897               None                                  None  0/1  255   \n",
       "\n",
       "        SDP   DP   RD   AD    FREQ                   PVAL  RDF  RDR  ADF ADR  \\\n",
       "0       101  101   66   35  34.65%             8.1311e-13   48   18   24  11   \n",
       "1       101  101   66   35  34.65%             8.1311e-13   48   18   24  11   \n",
       "2       143  143    0  143    100%             1.7063e-85    0    0  100  43   \n",
       "3       180  180  131   49  27.22%             4.0111e-17   84   47   39  10   \n",
       "4        96   96   73   23  23.96%             2.6678e-08   30   43   13  10   \n",
       "...     ...  ...  ...  ...     ...                    ...  ...  ...  ...  ..   \n",
       "333893  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "333894  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "333895  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "333896  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "333897  383  383  273  110  28.72%  7.984299999999999e-38  136  137   83  27   \n",
       "\n",
       "                   SIFT       PolyPhen                             BIOTYPE  \\\n",
       "0                                                           protein coding   \n",
       "1                                                          TF binding site   \n",
       "2       tolerated(0.92)      benign(0)                      protein coding   \n",
       "3                                                           protein coding   \n",
       "4                                                           protein coding   \n",
       "...                 ...            ...                                 ...   \n",
       "333893                                  transcribed unprocessed pseudogene   \n",
       "333894     tolerated(1)  benign(0.375)                      protein coding   \n",
       "333895     tolerated(1)  benign(0.101)                      protein coding   \n",
       "333896                                                              lncRNA   \n",
       "333897                                                              lncRNA   \n",
       "\n",
       "       EXON INTRON Protein Position and Amino Acid   Codons STRAND PUBMED  \\\n",
       "0       3/3                                    S81  tcA/tcG      1          \n",
       "1                                              nan                          \n",
       "2       3/3                                  T162A  Aca/Gca      1          \n",
       "3       3/3                                   L236  ctA/ctG      1          \n",
       "4       3/3                                   S290  tcT/tcC      1          \n",
       "...     ...    ...                             ...      ...    ...    ...   \n",
       "333893                                         nan              -1          \n",
       "333894  1/6                                   R86Q  cGg/cAg      1          \n",
       "333895  1/6                                   R86Q  cGg/cAg      1          \n",
       "333896                                         nan              -1          \n",
       "333897                                         nan              -1          \n",
       "\n",
       "       Gene_Match  \n",
       "0              No  \n",
       "1              No  \n",
       "2              No  \n",
       "3              No  \n",
       "4              No  \n",
       "...           ...  \n",
       "333893         No  \n",
       "333894         No  \n",
       "333895         No  \n",
       "333896         No  \n",
       "333897         No  \n",
       "\n",
       "[333898 rows x 41 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in merged_2['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            merged_2.loc[merged_2['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "02cefd69",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "44a65ce5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     333870\n",
       "Yes        28\n",
       "Name: Gene_Match, dtype: int64"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.Gene_Match.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "a9d239a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene Name', 'ID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
       "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
       "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
       "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
       "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
       "       'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
       "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED',\n",
       "       'Gene_Match'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "d4cc627d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match            ID CHROM      POS REF ALT  \\\n",
       "0          OR4F5         No   rs201219564  chr1    69270   A   G   \n",
       "1          OR4F5         No   rs201219564  chr1    69270   A   G   \n",
       "2          OR4F5         No     rs2691305  chr1    69511   A   G   \n",
       "3          OR4F5         No  rs1245478362  chr1    69735   A   G   \n",
       "4          OR4F5         No   rs200676709  chr1    69897   T   C   \n",
       "...          ...        ...           ...   ...      ...  ..  ..   \n",
       "333893     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333894     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333895     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333896     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333897     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "\n",
       "            Zygosity                Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous         synonymous variant              3/10       LOW   \n",
       "1       Heterozygous  regulatory region variant              2/10  MODIFIER   \n",
       "2         Homozygous           missense variant              7/10  MODERATE   \n",
       "3       Heterozygous         synonymous variant              3/10       LOW   \n",
       "4       Heterozygous         synonymous variant              3/10       LOW   \n",
       "...              ...                        ...               ...       ...   \n",
       "333893  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "333894  Heterozygous           missense variant              7/10  MODERATE   \n",
       "333895  Heterozygous           missense variant              7/10  MODERATE   \n",
       "333896  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "333897  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                2.5                                                     \n",
       "1                1.5                                                     \n",
       "2                5.0                                                     \n",
       "3                2.5                                                     \n",
       "4                2.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "333893           1.5                                                     \n",
       "333894           5.0                                                     \n",
       "333895           5.0                                                     \n",
       "333896           1.5                                                     \n",
       "333897           1.5                                                     \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "1                                       None                      \n",
       "2       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "3       ENST00000641515.2           c.708A>G  ENSP00000493376.2   \n",
       "4       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "...                   ...                ...                ...   \n",
       "333893                                  None                      \n",
       "333894  ENST00000423647.6           c.257G>A  ENSP00000389324.3   \n",
       "333895  ENST00000451548.6           c.257G>A  ENSP00000403304.1   \n",
       "333896                                  None                      \n",
       "333897                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD    FREQ  \\\n",
       "0              p.Ser81%3D  0/1  120  101  101   66   35  34.65%   \n",
       "1                    None  0/1  120  101  101   66   35  34.65%   \n",
       "2             p.Thr162Ala  1/1  255  143  143    0  143    100%   \n",
       "3             p.Leu236%3D  0/1  163  180  180  131   49  27.22%   \n",
       "4             p.Ser290%3D  0/1   75   96   96   73   23  23.96%   \n",
       "...                   ...  ...  ...  ...  ...  ...  ...     ...   \n",
       "333893               None  0/1  255  383  383  273  110  28.72%   \n",
       "333894         p.Arg86Gln  0/1  255  383  383  273  110  28.72%   \n",
       "333895         p.Arg86Gln  0/1  255  383  383  273  110  28.72%   \n",
       "333896               None  0/1  255  383  383  273  110  28.72%   \n",
       "333897               None  0/1  255  383  383  273  110  28.72%   \n",
       "\n",
       "                         PVAL  RDF  RDR  ADF ADR             SIFT  \\\n",
       "0                  8.1311e-13   48   18   24  11                    \n",
       "1                  8.1311e-13   48   18   24  11                    \n",
       "2                  1.7063e-85    0    0  100  43  tolerated(0.92)   \n",
       "3                  4.0111e-17   84   47   39  10                    \n",
       "4                  2.6678e-08   30   43   13  10                    \n",
       "...                       ...  ...  ...  ...  ..              ...   \n",
       "333893  7.984299999999999e-38  136  137   83  27                    \n",
       "333894  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333895  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333896  7.984299999999999e-38  136  137   83  27                    \n",
       "333897  7.984299999999999e-38  136  137   83  27                    \n",
       "\n",
       "             PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0                                          protein coding  3/3          \n",
       "1                                         TF binding site               \n",
       "2           benign(0)                      protein coding  3/3          \n",
       "3                                          protein coding  3/3          \n",
       "4                                          protein coding  3/3          \n",
       "...               ...                                 ...  ...    ...   \n",
       "333893                 transcribed unprocessed pseudogene               \n",
       "333894  benign(0.375)                      protein coding  1/6          \n",
       "333895  benign(0.101)                      protein coding  1/6          \n",
       "333896                                             lncRNA               \n",
       "333897                                             lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  S81  tcA/tcG      1         \n",
       "1                                  nan                         \n",
       "2                                T162A  Aca/Gca      1         \n",
       "3                                 L236  ctA/ctG      1         \n",
       "4                                 S290  tcT/tcC      1         \n",
       "...                                ...      ...    ...    ...  \n",
       "333893                             nan              -1         \n",
       "333894                            R86Q  cGg/cAg      1         \n",
       "333895                            R86Q  cGg/cAg      1         \n",
       "333896                             nan              -1         \n",
       "333897                             nan              -1         \n",
       "\n",
       "[333898 rows x 41 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'Gene_Match', 'ID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 376,
   "id": "cd3dd564",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "269"
      ]
     },
     "execution_count": 376,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "330219 - 329950"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4210c984",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "28ca064f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>113834946</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>188371466</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>188394766</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>188396801</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>188406566</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         POS Literature\n",
       "0  113834946        Yes\n",
       "1  188371466        Yes\n",
       "2  188394766        Yes\n",
       "3  188396801        Yes\n",
       "4  188406566        Yes"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Condition_pos/Vitiligo_pos.xlsx')\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "c2d42df2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match            ID CHROM      POS REF ALT  \\\n",
       "0          OR4F5         No   rs201219564  chr1    69270   A   G   \n",
       "1          OR4F5         No   rs201219564  chr1    69270   A   G   \n",
       "2          OR4F5         No     rs2691305  chr1    69511   A   G   \n",
       "3          OR4F5         No  rs1245478362  chr1    69735   A   G   \n",
       "4          OR4F5         No   rs200676709  chr1    69897   T   C   \n",
       "...          ...        ...           ...   ...      ...  ..  ..   \n",
       "333893     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333894     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333895     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333896     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333897     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "\n",
       "            Zygosity                Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous         synonymous variant              3/10       LOW   \n",
       "1       Heterozygous  regulatory region variant              2/10  MODIFIER   \n",
       "2         Homozygous           missense variant              7/10  MODERATE   \n",
       "3       Heterozygous         synonymous variant              3/10       LOW   \n",
       "4       Heterozygous         synonymous variant              3/10       LOW   \n",
       "...              ...                        ...               ...       ...   \n",
       "333893  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "333894  Heterozygous           missense variant              7/10  MODERATE   \n",
       "333895  Heterozygous           missense variant              7/10  MODERATE   \n",
       "333896  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "333897  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                2.5                                                     \n",
       "1                1.5                                                     \n",
       "2                5.0                                                     \n",
       "3                2.5                                                     \n",
       "4                2.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "333893           1.5                                                     \n",
       "333894           5.0                                                     \n",
       "333895           5.0                                                     \n",
       "333896           1.5                                                     \n",
       "333897           1.5                                                     \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "1                                       None                      \n",
       "2       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "3       ENST00000641515.2           c.708A>G  ENSP00000493376.2   \n",
       "4       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "...                   ...                ...                ...   \n",
       "333893                                  None                      \n",
       "333894  ENST00000423647.6           c.257G>A  ENSP00000389324.3   \n",
       "333895  ENST00000451548.6           c.257G>A  ENSP00000403304.1   \n",
       "333896                                  None                      \n",
       "333897                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD    FREQ  \\\n",
       "0              p.Ser81%3D  0/1  120  101  101   66   35  34.65%   \n",
       "1                    None  0/1  120  101  101   66   35  34.65%   \n",
       "2             p.Thr162Ala  1/1  255  143  143    0  143    100%   \n",
       "3             p.Leu236%3D  0/1  163  180  180  131   49  27.22%   \n",
       "4             p.Ser290%3D  0/1   75   96   96   73   23  23.96%   \n",
       "...                   ...  ...  ...  ...  ...  ...  ...     ...   \n",
       "333893               None  0/1  255  383  383  273  110  28.72%   \n",
       "333894         p.Arg86Gln  0/1  255  383  383  273  110  28.72%   \n",
       "333895         p.Arg86Gln  0/1  255  383  383  273  110  28.72%   \n",
       "333896               None  0/1  255  383  383  273  110  28.72%   \n",
       "333897               None  0/1  255  383  383  273  110  28.72%   \n",
       "\n",
       "                         PVAL  RDF  RDR  ADF ADR             SIFT  \\\n",
       "0                  8.1311e-13   48   18   24  11                    \n",
       "1                  8.1311e-13   48   18   24  11                    \n",
       "2                  1.7063e-85    0    0  100  43  tolerated(0.92)   \n",
       "3                  4.0111e-17   84   47   39  10                    \n",
       "4                  2.6678e-08   30   43   13  10                    \n",
       "...                       ...  ...  ...  ...  ..              ...   \n",
       "333893  7.984299999999999e-38  136  137   83  27                    \n",
       "333894  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333895  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333896  7.984299999999999e-38  136  137   83  27                    \n",
       "333897  7.984299999999999e-38  136  137   83  27                    \n",
       "\n",
       "             PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0                                          protein coding  3/3          \n",
       "1                                         TF binding site               \n",
       "2           benign(0)                      protein coding  3/3          \n",
       "3                                          protein coding  3/3          \n",
       "4                                          protein coding  3/3          \n",
       "...               ...                                 ...  ...    ...   \n",
       "333893                 transcribed unprocessed pseudogene               \n",
       "333894  benign(0.375)                      protein coding  1/6          \n",
       "333895  benign(0.101)                      protein coding  1/6          \n",
       "333896                                             lncRNA               \n",
       "333897                                             lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Literature  \n",
       "0                                  S81  tcA/tcG      1               NaN  \n",
       "1                                  nan                               NaN  \n",
       "2                                T162A  Aca/Gca      1               NaN  \n",
       "3                                 L236  ctA/ctG      1               NaN  \n",
       "4                                 S290  tcT/tcC      1               NaN  \n",
       "...                                ...      ...    ...    ...        ...  \n",
       "333893                             nan              -1               NaN  \n",
       "333894                            R86Q  cGg/cAg      1               NaN  \n",
       "333895                            R86Q  cGg/cAg      1               NaN  \n",
       "333896                             nan              -1               NaN  \n",
       "333897                             nan              -1               NaN  \n",
       "\n",
       "[333898 rows x 42 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = pd.merge(merged_2, df_3, on = 'POS', how = 'left', sort=False)\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "d4ddb796",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match            ID CHROM      POS REF ALT  \\\n",
       "0          OR4F5         No   rs201219564  chr1    69270   A   G   \n",
       "1          OR4F5         No   rs201219564  chr1    69270   A   G   \n",
       "2          OR4F5         No     rs2691305  chr1    69511   A   G   \n",
       "3          OR4F5         No  rs1245478362  chr1    69735   A   G   \n",
       "4          OR4F5         No   rs200676709  chr1    69897   T   C   \n",
       "...          ...        ...           ...   ...      ...  ..  ..   \n",
       "333893     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333894     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333895     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333896     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "333897     TSPY1         No   rs777840135  chrY  9467257   G   A   \n",
       "\n",
       "            Zygosity                Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous         synonymous variant              3/10       LOW   \n",
       "1       Heterozygous  regulatory region variant              2/10  MODIFIER   \n",
       "2         Homozygous           missense variant              7/10  MODERATE   \n",
       "3       Heterozygous         synonymous variant              3/10       LOW   \n",
       "4       Heterozygous         synonymous variant              3/10       LOW   \n",
       "...              ...                        ...               ...       ...   \n",
       "333893  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "333894  Heterozygous           missense variant              7/10  MODERATE   \n",
       "333895  Heterozygous           missense variant              7/10  MODERATE   \n",
       "333896  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "333897  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                2.5                                                     \n",
       "1                1.5                                                     \n",
       "2                5.0                                                     \n",
       "3                2.5                                                     \n",
       "4                2.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "333893           1.5                                                     \n",
       "333894           5.0                                                     \n",
       "333895           5.0                                                     \n",
       "333896           1.5                                                     \n",
       "333897           1.5                                                     \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "1                                       None                      \n",
       "2       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "3       ENST00000641515.2           c.708A>G  ENSP00000493376.2   \n",
       "4       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "...                   ...                ...                ...   \n",
       "333893                                  None                      \n",
       "333894  ENST00000423647.6           c.257G>A  ENSP00000389324.3   \n",
       "333895  ENST00000451548.6           c.257G>A  ENSP00000403304.1   \n",
       "333896                                  None                      \n",
       "333897                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD    FREQ  \\\n",
       "0              p.Ser81%3D  0/1  120  101  101   66   35  34.65%   \n",
       "1                    None  0/1  120  101  101   66   35  34.65%   \n",
       "2             p.Thr162Ala  1/1  255  143  143    0  143    100%   \n",
       "3             p.Leu236%3D  0/1  163  180  180  131   49  27.22%   \n",
       "4             p.Ser290%3D  0/1   75   96   96   73   23  23.96%   \n",
       "...                   ...  ...  ...  ...  ...  ...  ...     ...   \n",
       "333893               None  0/1  255  383  383  273  110  28.72%   \n",
       "333894         p.Arg86Gln  0/1  255  383  383  273  110  28.72%   \n",
       "333895         p.Arg86Gln  0/1  255  383  383  273  110  28.72%   \n",
       "333896               None  0/1  255  383  383  273  110  28.72%   \n",
       "333897               None  0/1  255  383  383  273  110  28.72%   \n",
       "\n",
       "                         PVAL  RDF  RDR  ADF ADR             SIFT  \\\n",
       "0                  8.1311e-13   48   18   24  11                    \n",
       "1                  8.1311e-13   48   18   24  11                    \n",
       "2                  1.7063e-85    0    0  100  43  tolerated(0.92)   \n",
       "3                  4.0111e-17   84   47   39  10                    \n",
       "4                  2.6678e-08   30   43   13  10                    \n",
       "...                       ...  ...  ...  ...  ..              ...   \n",
       "333893  7.984299999999999e-38  136  137   83  27                    \n",
       "333894  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333895  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333896  7.984299999999999e-38  136  137   83  27                    \n",
       "333897  7.984299999999999e-38  136  137   83  27                    \n",
       "\n",
       "             PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0                                          protein coding  3/3          \n",
       "1                                         TF binding site               \n",
       "2           benign(0)                      protein coding  3/3          \n",
       "3                                          protein coding  3/3          \n",
       "4                                          protein coding  3/3          \n",
       "...               ...                                 ...  ...    ...   \n",
       "333893                 transcribed unprocessed pseudogene               \n",
       "333894  benign(0.375)                      protein coding  1/6          \n",
       "333895  benign(0.101)                      protein coding  1/6          \n",
       "333896                                             lncRNA               \n",
       "333897                                             lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED Literature  \n",
       "0                                  S81  tcA/tcG      1                No  \n",
       "1                                  nan                                No  \n",
       "2                                T162A  Aca/Gca      1                No  \n",
       "3                                 L236  ctA/ctG      1                No  \n",
       "4                                 S290  tcT/tcC      1                No  \n",
       "...                                ...      ...    ...    ...        ...  \n",
       "333893                             nan              -1                No  \n",
       "333894                            R86Q  cGg/cAg      1                No  \n",
       "333895                            R86Q  cGg/cAg      1                No  \n",
       "333896                             nan              -1                No  \n",
       "333897                             nan              -1                No  \n",
       "\n",
       "[333898 rows x 42 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "d5cbadf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     333889\n",
       "Yes         9\n",
       "Name: Literature, dtype: int64"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Literature.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "a7a1ecf8",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16704</th>\n",
       "      <td>PTPN22,AP4B1-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2476601</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113834946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td>benign,risk factor</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000359785.10</td>\n",
       "      <td>c.1858T&gt;C</td>\n",
       "      <td>ENSP00000352833.5</td>\n",
       "      <td>p.Trp620Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>14/21</td>\n",
       "      <td></td>\n",
       "      <td>W620R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>-1</td>\n",
       "      <td>18248681,30409984,20941391,27417569,20444268,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16705</th>\n",
       "      <td>PTPN22,AP4B1-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2476601</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113834946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td>benign,risk factor</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000420377.6</td>\n",
       "      <td>c.1858T&gt;C</td>\n",
       "      <td>ENSP00000388229.2</td>\n",
       "      <td>p.Trp620Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>14/20</td>\n",
       "      <td></td>\n",
       "      <td>W620R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>-1</td>\n",
       "      <td>18248681,30409984,20941391,27417569,20444268,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16706</th>\n",
       "      <td>PTPN22,AP4B1-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2476601</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113834946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign,risk factor</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000460620.5</td>\n",
       "      <td>c.469-15292T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td>6/7</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>18248681,30409984,20941391,27417569,20444268,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16707</th>\n",
       "      <td>PTPN22,AP4B1-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2476601</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113834946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign,risk factor</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000484147.5</td>\n",
       "      <td>n.1899T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>14/16</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>18248681,30409984,20941391,27417569,20444268,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16708</th>\n",
       "      <td>PTPN22,AP4B1-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2476601</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113834946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td>benign,risk factor</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000525799.1</td>\n",
       "      <td>c.1477T&gt;C</td>\n",
       "      <td>ENSP00000432674.1</td>\n",
       "      <td>p.Trp493Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/15</td>\n",
       "      <td></td>\n",
       "      <td>W493R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>-1</td>\n",
       "      <td>18248681,30409984,20941391,27417569,20444268,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16709</th>\n",
       "      <td>PTPN22,AP4B1-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2476601</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113834946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td>benign,risk factor</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000528414.5</td>\n",
       "      <td>c.1693T&gt;C</td>\n",
       "      <td>ENSP00000435176.1</td>\n",
       "      <td>p.Trp565Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>12/19</td>\n",
       "      <td></td>\n",
       "      <td>W565R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>-1</td>\n",
       "      <td>18248681,30409984,20941391,27417569,20444268,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16710</th>\n",
       "      <td>PTPN22,AP4B1-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2476601</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113834946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>3 prime UTR variant,NMD transcript variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign,risk factor</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000532224.5</td>\n",
       "      <td>c.*1136T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nonsense mediated decay</td>\n",
       "      <td>10/17</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>18248681,30409984,20941391,27417569,20444268,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16711</th>\n",
       "      <td>PTPN22,AP4B1-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2476601</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113834946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td>benign,risk factor</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000538253.5</td>\n",
       "      <td>c.1786T&gt;C</td>\n",
       "      <td>ENSP00000439372.2</td>\n",
       "      <td>p.Trp596Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>13/20</td>\n",
       "      <td></td>\n",
       "      <td>W596R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>-1</td>\n",
       "      <td>18248681,30409984,20941391,27417569,20444268,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16712</th>\n",
       "      <td>PTPN22,AP4B1-AS1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs2476601</td>\n",
       "      <td>chr1</td>\n",
       "      <td>113834946</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td>benign,risk factor</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000664434.1</td>\n",
       "      <td>n.470+3133A&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>70</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>100%</td>\n",
       "      <td>9.6148e-08</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>4/5</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>18248681,30409984,20941391,27417569,20444268,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene Name Gene_Match         ID CHROM        POS REF ALT  \\\n",
       "16704  PTPN22,AP4B1-AS1        Yes  rs2476601  chr1  113834946   A   G   \n",
       "16705  PTPN22,AP4B1-AS1        Yes  rs2476601  chr1  113834946   A   G   \n",
       "16706  PTPN22,AP4B1-AS1        Yes  rs2476601  chr1  113834946   A   G   \n",
       "16707  PTPN22,AP4B1-AS1        Yes  rs2476601  chr1  113834946   A   G   \n",
       "16708  PTPN22,AP4B1-AS1        Yes  rs2476601  chr1  113834946   A   G   \n",
       "16709  PTPN22,AP4B1-AS1        Yes  rs2476601  chr1  113834946   A   G   \n",
       "16710  PTPN22,AP4B1-AS1        Yes  rs2476601  chr1  113834946   A   G   \n",
       "16711  PTPN22,AP4B1-AS1        Yes  rs2476601  chr1  113834946   A   G   \n",
       "16712  PTPN22,AP4B1-AS1        Yes  rs2476601  chr1  113834946   A   G   \n",
       "\n",
       "         Zygosity                                   Consequence  \\\n",
       "16704  Homozygous                              missense variant   \n",
       "16705  Homozygous                              missense variant   \n",
       "16706  Homozygous                                intron variant   \n",
       "16707  Homozygous            non coding transcript exon variant   \n",
       "16708  Homozygous                              missense variant   \n",
       "16709  Homozygous                              missense variant   \n",
       "16710  Homozygous    3 prime UTR variant,NMD transcript variant   \n",
       "16711  Homozygous                              missense variant   \n",
       "16712  Homozygous  intron variant,non coding transcript variant   \n",
       "\n",
       "      Consequence_score    IMPACT  IMPACT_score ClinVar_CLNDN  \\\n",
       "16704              7/10  MODERATE           5.0                 \n",
       "16705              7/10  MODERATE           5.0                 \n",
       "16706              2/10  MODIFIER           1.5                 \n",
       "16707              2/10  MODIFIER           1.5                 \n",
       "16708              7/10  MODERATE           5.0                 \n",
       "16709              7/10  MODERATE           5.0                 \n",
       "16710              3/10  MODIFIER           1.5                 \n",
       "16711              7/10  MODERATE           5.0                 \n",
       "16712              2/10  MODIFIER           1.5                 \n",
       "\n",
       "                 CLIN_SIG ClinVar_CLNREVSTAT ClinVar               HGVSc  \\\n",
       "16704  benign,risk factor                             ENST00000359785.10   \n",
       "16705  benign,risk factor                              ENST00000420377.6   \n",
       "16706  benign,risk factor                              ENST00000460620.5   \n",
       "16707  benign,risk factor                              ENST00000484147.5   \n",
       "16708  benign,risk factor                              ENST00000525799.1   \n",
       "16709  benign,risk factor                              ENST00000528414.5   \n",
       "16710  benign,risk factor                              ENST00000532224.5   \n",
       "16711  benign,risk factor                              ENST00000538253.5   \n",
       "16712  benign,risk factor                              ENST00000664434.1   \n",
       "\n",
       "      HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT  GQ SDP  \\\n",
       "16704          c.1858T>C  ENSP00000352833.5        p.Trp620Arg  1/1  70  13   \n",
       "16705          c.1858T>C  ENSP00000388229.2        p.Trp620Arg  1/1  70  13   \n",
       "16706     c.469-15292T>C                                  None  1/1  70  13   \n",
       "16707          n.1899T>C                                  None  1/1  70  13   \n",
       "16708          c.1477T>C  ENSP00000432674.1        p.Trp493Arg  1/1  70  13   \n",
       "16709          c.1693T>C  ENSP00000435176.1        p.Trp565Arg  1/1  70  13   \n",
       "16710         c.*1136T>C                                  None  1/1  70  13   \n",
       "16711          c.1786T>C  ENSP00000439372.2        p.Trp596Arg  1/1  70  13   \n",
       "16712      n.470+3133A>G                                  None  1/1  70  13   \n",
       "\n",
       "       DP RD  AD  FREQ        PVAL RDF RDR ADF ADR          SIFT   PolyPhen  \\\n",
       "16704  13  0  13  100%  9.6148e-08   0   0   9   4  tolerated(1)  benign(0)   \n",
       "16705  13  0  13  100%  9.6148e-08   0   0   9   4  tolerated(1)  benign(0)   \n",
       "16706  13  0  13  100%  9.6148e-08   0   0   9   4                            \n",
       "16707  13  0  13  100%  9.6148e-08   0   0   9   4                            \n",
       "16708  13  0  13  100%  9.6148e-08   0   0   9   4  tolerated(1)  benign(0)   \n",
       "16709  13  0  13  100%  9.6148e-08   0   0   9   4  tolerated(1)  benign(0)   \n",
       "16710  13  0  13  100%  9.6148e-08   0   0   9   4                            \n",
       "16711  13  0  13  100%  9.6148e-08   0   0   9   4  tolerated(1)  benign(0)   \n",
       "16712  13  0  13  100%  9.6148e-08   0   0   9   4                            \n",
       "\n",
       "                       BIOTYPE   EXON INTRON Protein Position and Amino Acid  \\\n",
       "16704           protein coding  14/21                                  W620R   \n",
       "16705           protein coding  14/20                                  W620R   \n",
       "16706           protein coding           6/7                             nan   \n",
       "16707          retained intron  14/16                                    nan   \n",
       "16708           protein coding   9/15                                  W493R   \n",
       "16709           protein coding  12/19                                  W565R   \n",
       "16710  nonsense mediated decay  10/17                                    nan   \n",
       "16711           protein coding  13/20                                  W596R   \n",
       "16712                   lncRNA           4/5                             nan   \n",
       "\n",
       "        Codons STRAND                                             PUBMED  \\\n",
       "16704  Tgg/Cgg     -1  18248681,30409984,20941391,27417569,20444268,2...   \n",
       "16705  Tgg/Cgg     -1  18248681,30409984,20941391,27417569,20444268,2...   \n",
       "16706              -1  18248681,30409984,20941391,27417569,20444268,2...   \n",
       "16707              -1  18248681,30409984,20941391,27417569,20444268,2...   \n",
       "16708  Tgg/Cgg     -1  18248681,30409984,20941391,27417569,20444268,2...   \n",
       "16709  Tgg/Cgg     -1  18248681,30409984,20941391,27417569,20444268,2...   \n",
       "16710              -1  18248681,30409984,20941391,27417569,20444268,2...   \n",
       "16711  Tgg/Cgg     -1  18248681,30409984,20941391,27417569,20444268,2...   \n",
       "16712               1  18248681,30409984,20941391,27417569,20444268,2...   \n",
       "\n",
       "      Literature  \n",
       "16704        Yes  \n",
       "16705        Yes  \n",
       "16706        Yes  \n",
       "16707        Yes  \n",
       "16708        Yes  \n",
       "16709        Yes  \n",
       "16710        Yes  \n",
       "16711        Yes  \n",
       "16712        Yes  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3[merged_3['Literature'] == 'Yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "503a4907",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Gene Name', 'Gene_Match', 'ID', 'CHROM', 'POS', 'REF', 'ALT',\n",
       "       'Zygosity', 'Consequence', 'Consequence_score', 'IMPACT',\n",
       "       'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
       "       'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)',\n",
       "       'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR',\n",
       "       'ADF', 'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
       "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED',\n",
       "       'Literature'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "2a2ca42b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>120</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>66</td>\n",
       "      <td>35</td>\n",
       "      <td>34.65%</td>\n",
       "      <td>8.1311e-13</td>\n",
       "      <td>48</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>143</td>\n",
       "      <td>143</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.7063e-85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>43</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs1245478362</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69735</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.708A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Leu236%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>163</td>\n",
       "      <td>180</td>\n",
       "      <td>180</td>\n",
       "      <td>131</td>\n",
       "      <td>49</td>\n",
       "      <td>27.22%</td>\n",
       "      <td>4.0111e-17</td>\n",
       "      <td>84</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>L236</td>\n",
       "      <td>ctA/ctG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.870T&gt;C</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser290%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>75</td>\n",
       "      <td>96</td>\n",
       "      <td>96</td>\n",
       "      <td>73</td>\n",
       "      <td>23</td>\n",
       "      <td>23.96%</td>\n",
       "      <td>2.6678e-08</td>\n",
       "      <td>30</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S290</td>\n",
       "      <td>tcT/tcC</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333893</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>No</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>transcribed unprocessed pseudogene</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333894</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>No</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000423647.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000389324.3</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.375)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333895</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>No</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000451548.6</td>\n",
       "      <td>c.257G&gt;A</td>\n",
       "      <td>ENSP00000403304.1</td>\n",
       "      <td>p.Arg86Gln</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.101)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/6</td>\n",
       "      <td></td>\n",
       "      <td>R86Q</td>\n",
       "      <td>cGg/cAg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333896</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>No</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333897</th>\n",
       "      <td>TSPY1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs777840135</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9467257</td>\n",
       "      <td>No</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>383</td>\n",
       "      <td>273</td>\n",
       "      <td>110</td>\n",
       "      <td>28.72%</td>\n",
       "      <td>7.984299999999999e-38</td>\n",
       "      <td>136</td>\n",
       "      <td>137</td>\n",
       "      <td>83</td>\n",
       "      <td>27</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>333898 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match            ID CHROM      POS Literature REF ALT  \\\n",
       "0          OR4F5         No   rs201219564  chr1    69270         No   A   G   \n",
       "1          OR4F5         No   rs201219564  chr1    69270         No   A   G   \n",
       "2          OR4F5         No     rs2691305  chr1    69511         No   A   G   \n",
       "3          OR4F5         No  rs1245478362  chr1    69735         No   A   G   \n",
       "4          OR4F5         No   rs200676709  chr1    69897         No   T   C   \n",
       "...          ...        ...           ...   ...      ...        ...  ..  ..   \n",
       "333893     TSPY1         No   rs777840135  chrY  9467257         No   G   A   \n",
       "333894     TSPY1         No   rs777840135  chrY  9467257         No   G   A   \n",
       "333895     TSPY1         No   rs777840135  chrY  9467257         No   G   A   \n",
       "333896     TSPY1         No   rs777840135  chrY  9467257         No   G   A   \n",
       "333897     TSPY1         No   rs777840135  chrY  9467257         No   G   A   \n",
       "\n",
       "            Zygosity                Consequence Consequence_score    IMPACT  \\\n",
       "0       Heterozygous         synonymous variant              3/10       LOW   \n",
       "1       Heterozygous  regulatory region variant              2/10  MODIFIER   \n",
       "2         Homozygous           missense variant              7/10  MODERATE   \n",
       "3       Heterozygous         synonymous variant              3/10       LOW   \n",
       "4       Heterozygous         synonymous variant              3/10       LOW   \n",
       "...              ...                        ...               ...       ...   \n",
       "333893  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "333894  Heterozygous           missense variant              7/10  MODERATE   \n",
       "333895  Heterozygous           missense variant              7/10  MODERATE   \n",
       "333896  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "333897  Heterozygous      upstream gene variant              2/10  MODIFIER   \n",
       "\n",
       "        IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT ClinVar  \\\n",
       "0                2.5                                                     \n",
       "1                1.5                                                     \n",
       "2                5.0                                                     \n",
       "3                2.5                                                     \n",
       "4                2.5                                                     \n",
       "...              ...           ...      ...                ...     ...   \n",
       "333893           1.5                                                     \n",
       "333894           5.0                                                     \n",
       "333895           5.0                                                     \n",
       "333896           1.5                                                     \n",
       "333897           1.5                                                     \n",
       "\n",
       "                    HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0       ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "1                                       None                      \n",
       "2       ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "3       ENST00000641515.2           c.708A>G  ENSP00000493376.2   \n",
       "4       ENST00000641515.2           c.870T>C  ENSP00000493376.2   \n",
       "...                   ...                ...                ...   \n",
       "333893                                  None                      \n",
       "333894  ENST00000423647.6           c.257G>A  ENSP00000389324.3   \n",
       "333895  ENST00000451548.6           c.257G>A  ENSP00000403304.1   \n",
       "333896                                  None                      \n",
       "333897                                  None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD    FREQ  \\\n",
       "0              p.Ser81%3D  0/1  120  101  101   66   35  34.65%   \n",
       "1                    None  0/1  120  101  101   66   35  34.65%   \n",
       "2             p.Thr162Ala  1/1  255  143  143    0  143    100%   \n",
       "3             p.Leu236%3D  0/1  163  180  180  131   49  27.22%   \n",
       "4             p.Ser290%3D  0/1   75   96   96   73   23  23.96%   \n",
       "...                   ...  ...  ...  ...  ...  ...  ...     ...   \n",
       "333893               None  0/1  255  383  383  273  110  28.72%   \n",
       "333894         p.Arg86Gln  0/1  255  383  383  273  110  28.72%   \n",
       "333895         p.Arg86Gln  0/1  255  383  383  273  110  28.72%   \n",
       "333896               None  0/1  255  383  383  273  110  28.72%   \n",
       "333897               None  0/1  255  383  383  273  110  28.72%   \n",
       "\n",
       "                         PVAL  RDF  RDR  ADF ADR             SIFT  \\\n",
       "0                  8.1311e-13   48   18   24  11                    \n",
       "1                  8.1311e-13   48   18   24  11                    \n",
       "2                  1.7063e-85    0    0  100  43  tolerated(0.92)   \n",
       "3                  4.0111e-17   84   47   39  10                    \n",
       "4                  2.6678e-08   30   43   13  10                    \n",
       "...                       ...  ...  ...  ...  ..              ...   \n",
       "333893  7.984299999999999e-38  136  137   83  27                    \n",
       "333894  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333895  7.984299999999999e-38  136  137   83  27     tolerated(1)   \n",
       "333896  7.984299999999999e-38  136  137   83  27                    \n",
       "333897  7.984299999999999e-38  136  137   83  27                    \n",
       "\n",
       "             PolyPhen                             BIOTYPE EXON INTRON  \\\n",
       "0                                          protein coding  3/3          \n",
       "1                                         TF binding site               \n",
       "2           benign(0)                      protein coding  3/3          \n",
       "3                                          protein coding  3/3          \n",
       "4                                          protein coding  3/3          \n",
       "...               ...                                 ...  ...    ...   \n",
       "333893                 transcribed unprocessed pseudogene               \n",
       "333894  benign(0.375)                      protein coding  1/6          \n",
       "333895  benign(0.101)                      protein coding  1/6          \n",
       "333896                                             lncRNA               \n",
       "333897                                             lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "0                                  S81  tcA/tcG      1         \n",
       "1                                  nan                         \n",
       "2                                T162A  Aca/Gca      1         \n",
       "3                                 L236  ctA/ctG      1         \n",
       "4                                 S290  tcT/tcC      1         \n",
       "...                                ...      ...    ...    ...  \n",
       "333893                             nan              -1         \n",
       "333894                            R86Q  cGg/cAg      1         \n",
       "333895                            R86Q  cGg/cAg      1         \n",
       "333896                             nan              -1         \n",
       "333897                             nan              -1         \n",
       "\n",
       "[333898 rows x 42 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = merged_3[['Gene Name', 'Gene_Match', 'ID', 'CHROM', 'POS', 'Literature', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "961dbac3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>c.751T&gt;C</td>\n",
       "      <td>ENSP00000349216.4</td>\n",
       "      <td>p.Trp251Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/12</td>\n",
       "      <td></td>\n",
       "      <td>W251R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>c.1027T&gt;C</td>\n",
       "      <td>ENSP00000342313.3</td>\n",
       "      <td>p.Trp343Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W343R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000455979.1</td>\n",
       "      <td>c.508T&gt;C</td>\n",
       "      <td>ENSP00000412228.1</td>\n",
       "      <td>p.Trp170Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>4/7</td>\n",
       "      <td></td>\n",
       "      <td>W170R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000464948.1</td>\n",
       "      <td>n.286T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>1/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000466827.1</td>\n",
       "      <td>n.191T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>2/2</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000474461.1</td>\n",
       "      <td>n.389T&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>3/4</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding CDS not defined</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000616016.5</td>\n",
       "      <td>c.1516T&gt;C</td>\n",
       "      <td>ENSP00000478421.2</td>\n",
       "      <td>p.Trp506Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W506R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000616125.5</td>\n",
       "      <td>c.703T&gt;C</td>\n",
       "      <td>ENSP00000484643.1</td>\n",
       "      <td>p.Trp235Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>7/11</td>\n",
       "      <td></td>\n",
       "      <td>W235R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000617307.5</td>\n",
       "      <td>c.1030T&gt;C</td>\n",
       "      <td>ENSP00000482090.2</td>\n",
       "      <td>p.Trp344Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>W344R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618181.5</td>\n",
       "      <td>c.652T&gt;C</td>\n",
       "      <td>ENSP00000480870.1</td>\n",
       "      <td>p.Trp218Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>6/10</td>\n",
       "      <td></td>\n",
       "      <td>W218R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618323.5</td>\n",
       "      <td>c.1519T&gt;C</td>\n",
       "      <td>ENSP00000480678.2</td>\n",
       "      <td>p.Trp507Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated low confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>10/14</td>\n",
       "      <td></td>\n",
       "      <td>W507R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000618779.5</td>\n",
       "      <td>c.841T&gt;C</td>\n",
       "      <td>ENSP00000484256.1</td>\n",
       "      <td>p.Trp281Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>8/12</td>\n",
       "      <td></td>\n",
       "      <td>W281R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td>ENST00000622503.5</td>\n",
       "      <td>c.1030T&gt;C</td>\n",
       "      <td>ENSP00000482138.1</td>\n",
       "      <td>p.Trp344Arg</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>W344R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SAMD11</td>\n",
       "      <td>No</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>chr1</td>\n",
       "      <td>942451</td>\n",
       "      <td>No</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>not provided</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>1166513</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>0</td>\n",
       "      <td>79</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.3185e-47</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name Gene_Match         ID CHROM     POS Literature REF ALT  \\\n",
       "46    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "47    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "48    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "49    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "50    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "51    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "52    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "53    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "54    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "55    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "56    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "57    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "58    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "59    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "60    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "61    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "62    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "63    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "64    SAMD11         No  rs6672356  chr1  942451         No   T   C   \n",
       "\n",
       "      Zygosity                         Consequence Consequence_score  \\\n",
       "46  Homozygous             downstream gene variant              2/10   \n",
       "47  Homozygous                    missense variant              7/10   \n",
       "48  Homozygous                    missense variant              7/10   \n",
       "49  Homozygous                    missense variant              7/10   \n",
       "50  Homozygous  non coding transcript exon variant              2/10   \n",
       "51  Homozygous  non coding transcript exon variant              2/10   \n",
       "52  Homozygous  non coding transcript exon variant              2/10   \n",
       "53  Homozygous             downstream gene variant              2/10   \n",
       "54  Homozygous             downstream gene variant              2/10   \n",
       "55  Homozygous             downstream gene variant              2/10   \n",
       "56  Homozygous             downstream gene variant              2/10   \n",
       "57  Homozygous                    missense variant              7/10   \n",
       "58  Homozygous                    missense variant              7/10   \n",
       "59  Homozygous                    missense variant              7/10   \n",
       "60  Homozygous                    missense variant              7/10   \n",
       "61  Homozygous                    missense variant              7/10   \n",
       "62  Homozygous                    missense variant              7/10   \n",
       "63  Homozygous                    missense variant              7/10   \n",
       "64  Homozygous           regulatory region variant              2/10   \n",
       "\n",
       "      IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "46  MODIFIER           1.5  not provided   benign   \n",
       "47  MODERATE           5.0  not provided   benign   \n",
       "48  MODERATE           5.0  not provided   benign   \n",
       "49  MODERATE           5.0  not provided   benign   \n",
       "50  MODIFIER           1.5  not provided   benign   \n",
       "51  MODIFIER           1.5  not provided   benign   \n",
       "52  MODIFIER           1.5  not provided   benign   \n",
       "53  MODIFIER           1.5  not provided   benign   \n",
       "54  MODIFIER           1.5  not provided   benign   \n",
       "55  MODIFIER           1.5  not provided   benign   \n",
       "56  MODIFIER           1.5  not provided   benign   \n",
       "57  MODERATE           5.0  not provided   benign   \n",
       "58  MODERATE           5.0  not provided   benign   \n",
       "59  MODERATE           5.0  not provided   benign   \n",
       "60  MODERATE           5.0  not provided   benign   \n",
       "61  MODERATE           5.0  not provided   benign   \n",
       "62  MODERATE           5.0  not provided   benign   \n",
       "63  MODERATE           5.0  not provided   benign   \n",
       "64  MODIFIER           1.5  not provided   benign   \n",
       "\n",
       "                     ClinVar_CLNREVSTAT  ClinVar              HGVSc  \\\n",
       "46  criteria provided, single submitter  1166513                      \n",
       "47  criteria provided, single submitter  1166513  ENST00000341065.8   \n",
       "48  criteria provided, single submitter  1166513  ENST00000342066.8   \n",
       "49  criteria provided, single submitter  1166513  ENST00000455979.1   \n",
       "50  criteria provided, single submitter  1166513  ENST00000464948.1   \n",
       "51  criteria provided, single submitter  1166513  ENST00000466827.1   \n",
       "52  criteria provided, single submitter  1166513  ENST00000474461.1   \n",
       "53  criteria provided, single submitter  1166513                      \n",
       "54  criteria provided, single submitter  1166513                      \n",
       "55  criteria provided, single submitter  1166513                      \n",
       "56  criteria provided, single submitter  1166513                      \n",
       "57  criteria provided, single submitter  1166513  ENST00000616016.5   \n",
       "58  criteria provided, single submitter  1166513  ENST00000616125.5   \n",
       "59  criteria provided, single submitter  1166513  ENST00000617307.5   \n",
       "60  criteria provided, single submitter  1166513  ENST00000618181.5   \n",
       "61  criteria provided, single submitter  1166513  ENST00000618323.5   \n",
       "62  criteria provided, single submitter  1166513  ENST00000618779.5   \n",
       "63  criteria provided, single submitter  1166513  ENST00000622503.5   \n",
       "64  criteria provided, single submitter  1166513                      \n",
       "\n",
       "   HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ SDP  DP  \\\n",
       "46               None                                  None  1/1  255  79  79   \n",
       "47           c.751T>C  ENSP00000349216.4        p.Trp251Arg  1/1  255  79  79   \n",
       "48          c.1027T>C  ENSP00000342313.3        p.Trp343Arg  1/1  255  79  79   \n",
       "49           c.508T>C  ENSP00000412228.1        p.Trp170Arg  1/1  255  79  79   \n",
       "50           n.286T>C                                  None  1/1  255  79  79   \n",
       "51           n.191T>C                                  None  1/1  255  79  79   \n",
       "52           n.389T>C                                  None  1/1  255  79  79   \n",
       "53               None                                  None  1/1  255  79  79   \n",
       "54               None                                  None  1/1  255  79  79   \n",
       "55               None                                  None  1/1  255  79  79   \n",
       "56               None                                  None  1/1  255  79  79   \n",
       "57          c.1516T>C  ENSP00000478421.2        p.Trp506Arg  1/1  255  79  79   \n",
       "58           c.703T>C  ENSP00000484643.1        p.Trp235Arg  1/1  255  79  79   \n",
       "59          c.1030T>C  ENSP00000482090.2        p.Trp344Arg  1/1  255  79  79   \n",
       "60           c.652T>C  ENSP00000480870.1        p.Trp218Arg  1/1  255  79  79   \n",
       "61          c.1519T>C  ENSP00000480678.2        p.Trp507Arg  1/1  255  79  79   \n",
       "62           c.841T>C  ENSP00000484256.1        p.Trp281Arg  1/1  255  79  79   \n",
       "63          c.1030T>C  ENSP00000482138.1        p.Trp344Arg  1/1  255  79  79   \n",
       "64               None                                  None  1/1  255  79  79   \n",
       "\n",
       "   RD  AD  FREQ        PVAL RDF RDR ADF ADR                         SIFT  \\\n",
       "46  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "47  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "48  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "49  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "50  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "51  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "52  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "53  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "54  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "55  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "56  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "57  0  79  100%  4.3185e-47   0   0  38  41  tolerated low confidence(1)   \n",
       "58  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "59  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "60  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "61  0  79  100%  4.3185e-47   0   0  38  41  tolerated low confidence(1)   \n",
       "62  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "63  0  79  100%  4.3185e-47   0   0  38  41                 tolerated(1)   \n",
       "64  0  79  100%  4.3185e-47   0   0  38  41                                \n",
       "\n",
       "         PolyPhen                         BIOTYPE   EXON INTRON  \\\n",
       "46                                 protein coding                 \n",
       "47      benign(0)                  protein coding   8/12          \n",
       "48      benign(0)                  protein coding  10/14          \n",
       "49      benign(0)                  protein coding    4/7          \n",
       "50                                retained intron    1/2          \n",
       "51                                retained intron    2/2          \n",
       "52                                retained intron    3/4          \n",
       "53                                retained intron                 \n",
       "54                 protein coding CDS not defined                 \n",
       "55                                retained intron                 \n",
       "56                 protein coding CDS not defined                 \n",
       "57      benign(0)                  protein coding  10/14          \n",
       "58  benign(0.001)                  protein coding   7/11          \n",
       "59      benign(0)                  protein coding   9/13          \n",
       "60      benign(0)                  protein coding   6/10          \n",
       "61      benign(0)                  protein coding  10/14          \n",
       "62      benign(0)                  protein coding   8/12          \n",
       "63      benign(0)                  protein coding   9/13          \n",
       "64                                       promoter                 \n",
       "\n",
       "   Protein Position and Amino Acid   Codons STRAND PUBMED  \n",
       "46                             nan              -1         \n",
       "47                           W251R  Tgg/Cgg      1         \n",
       "48                           W343R  Tgg/Cgg      1         \n",
       "49                           W170R  Tgg/Cgg      1         \n",
       "50                             nan               1         \n",
       "51                             nan               1         \n",
       "52                             nan               1         \n",
       "53                             nan              -1         \n",
       "54                             nan               1         \n",
       "55                             nan              -1         \n",
       "56                             nan              -1         \n",
       "57                           W506R  Tgg/Cgg      1         \n",
       "58                           W235R  Tgg/Cgg      1         \n",
       "59                           W344R  Tgg/Cgg      1         \n",
       "60                           W218R  Tgg/Cgg      1         \n",
       "61                           W507R  Tgg/Cgg      1         \n",
       "62                           W281R  Tgg/Cgg      1         \n",
       "63                           W344R  Tgg/Cgg      1         \n",
       "64                             nan                         "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3[merged_3['ID'] == 'rs6672356']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c10d71b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHGLBS535_depth_vcf_processed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2cc6a6fb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cd7de0a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7dde714a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "4bd5502d",
   "metadata": {},
   "source": [
    "# InDel Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1eebd294",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['ATCC', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KIF1A</td>\n",
       "      <td>rs10594016</td>\n",
       "      <td>inframe_insertion', 'SO:0001627</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Hereditary_spastic_paraplegia_30|not_specified...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['ATCC', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KIF1A</td>\n",
       "      <td>rs10594016</td>\n",
       "      <td>inframe_deletion', 'SO:0001627</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>['AACTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>PLCB1</td>\n",
       "      <td>rs78940282</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Early_Infantile_Epileptic_Encephalopathy', '_A...</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>['GT', G]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CD2AP</td>\n",
       "      <td>rs797004904</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Focal_segmental_glomerulosclerosis</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>['TA', T]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs8176144</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Hereditary_breast_ovarian_cancer_syndrome|Brea...</td>\n",
       "      <td>reviewed_by_expert_panel</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>583 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             allele      zygocity Gene Name         rsid  \\\n",
       "0    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "1    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "2      ['ACCTT', A]  Heterozygous  ADAMTS17   rs10549565   \n",
       "3       ['ATCC', A]  Heterozygous     KIF1A   rs10594016   \n",
       "4       ['ATCC', A]  Heterozygous     KIF1A   rs10594016   \n",
       "..              ...           ...       ...          ...   \n",
       "578    ['AACTT', A]  Heterozygous     PLCB1   rs78940282   \n",
       "579       ['AT', A]  Heterozygous     ATCAY  rs796133913   \n",
       "580       ['AT', A]  Heterozygous     ATCAY  rs796133913   \n",
       "581       ['GT', G]  Heterozygous     CD2AP  rs797004904   \n",
       "582       ['TA', T]  Heterozygous     BRCA1    rs8176144   \n",
       "\n",
       "                         consequence  \\\n",
       "0                5_prime_UTR_variant   \n",
       "1                5_prime_UTR_variant   \n",
       "2                3_prime_UTR_variant   \n",
       "3    inframe_insertion', 'SO:0001627   \n",
       "4     inframe_deletion', 'SO:0001627   \n",
       "..                               ...   \n",
       "578              3_prime_UTR_variant   \n",
       "579              3_prime_UTR_variant   \n",
       "580              3_prime_UTR_variant   \n",
       "581              3_prime_UTR_variant   \n",
       "582                   intron_variant   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                          Uncertain_significance   \n",
       "1                          Uncertain_significance   \n",
       "2                                          Benign   \n",
       "3    Conflicting_interpretations_of_pathogenicity   \n",
       "4                                          Benign   \n",
       "..                                            ...   \n",
       "578                                 Likely_benign   \n",
       "579                                        Benign   \n",
       "580                        Uncertain_significance   \n",
       "581                                        Benign   \n",
       "582                                        Benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0                       Trichorhinophalangeal_syndrome   \n",
       "1                       Trichorhinophalangeal_syndrome   \n",
       "2    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "3    Hereditary_spastic_paraplegia_30|not_specified...   \n",
       "4                           not_specified|not_provided   \n",
       "..                                                 ...   \n",
       "578  Early_Infantile_Epileptic_Encephalopathy', '_A...   \n",
       "579                      Cayman_type_cerebellar_ataxia   \n",
       "580                      Cayman_type_cerebellar_ataxia   \n",
       "581                 Focal_segmental_glomerulosclerosis   \n",
       "582  Hereditary_breast_ovarian_cancer_syndrome|Brea...   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0               criteria_provided', '_single_submitter  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3    criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "4    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "578             criteria_provided', '_single_submitter  germline        indel   \n",
       "579             criteria_provided', '_single_submitter  germline        indel   \n",
       "580             criteria_provided', '_single_submitter  germline        indel   \n",
       "581             criteria_provided', '_single_submitter  germline        indel   \n",
       "582                           reviewed_by_expert_panel  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \n",
       "0         Insertion           MayBe No          No                  No  \n",
       "1          Deletion           MayBe No          No                  No  \n",
       "2          Deletion           MayBe No          No                  No  \n",
       "3         Insertion           MayBe No          No                 Yes  \n",
       "4          Deletion           MayBe No          No                 Yes  \n",
       "..              ...                ...         ...                 ...  \n",
       "578        Deletion           MayBe No          No                  No  \n",
       "579        Deletion           MayBe No          No                  No  \n",
       "580        Deletion           MayBe No          No                  No  \n",
       "581        Deletion           MayBe No          No                  No  \n",
       "582        Deletion           MayBe No          No                 Yes  \n",
       "\n",
       "[583 rows x 14 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indel = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/cdpr_and_csps_all_files/KHCDPRGPTTL5/KHCDPRGPTTL5_indel_clinical_significance.xlsx')\n",
    "indel = indel.rename({'gene_name' : 'Gene Name'}, axis=1)\n",
    "indel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "67f0962a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LDLR-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NOS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRKAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LRP6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>TRDN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>LDLRAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>PCSK9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>APOB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>STAP1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name\n",
       "0       LDLR\n",
       "1   LDLR-AS1\n",
       "2       NOS3\n",
       "3     PRKAG2\n",
       "4       LRP6\n",
       "..       ...\n",
       "75      TRDN\n",
       "76   LDLRAP1\n",
       "77     PCSK9\n",
       "78      APOB\n",
       "79     STAP1\n",
       "\n",
       "[80 rows x 1 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/cardiac genes.xlsx')\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8d0c1fe5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygocity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['AAGAGAG', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPS1</td>\n",
       "      <td>rs10546472</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Trichorhinophalangeal_syndrome</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['ATCC', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KIF1A</td>\n",
       "      <td>rs10594016</td>\n",
       "      <td>inframe_insertion', 'SO:0001627</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>Hereditary_spastic_paraplegia_30|not_specified...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['ATCC', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>KIF1A</td>\n",
       "      <td>rs10594016</td>\n",
       "      <td>inframe_deletion', 'SO:0001627</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>578</th>\n",
       "      <td>['AACTT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>PLCB1</td>\n",
       "      <td>rs78940282</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Early_Infantile_Epileptic_Encephalopathy', '_A...</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>579</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>['AT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ATCAY</td>\n",
       "      <td>rs796133913</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Cayman_type_cerebellar_ataxia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>['GT', G]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CD2AP</td>\n",
       "      <td>rs797004904</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Focal_segmental_glomerulosclerosis</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>582</th>\n",
       "      <td>['TA', T]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs8176144</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Hereditary_breast_ovarian_cancer_syndrome|Brea...</td>\n",
       "      <td>reviewed_by_expert_panel</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>583 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             allele      zygocity Gene Name         rsid  \\\n",
       "0    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "1    ['AAGAGAG', A]  Heterozygous     TRPS1   rs10546472   \n",
       "2      ['ACCTT', A]  Heterozygous  ADAMTS17   rs10549565   \n",
       "3       ['ATCC', A]  Heterozygous     KIF1A   rs10594016   \n",
       "4       ['ATCC', A]  Heterozygous     KIF1A   rs10594016   \n",
       "..              ...           ...       ...          ...   \n",
       "578    ['AACTT', A]  Heterozygous     PLCB1   rs78940282   \n",
       "579       ['AT', A]  Heterozygous     ATCAY  rs796133913   \n",
       "580       ['AT', A]  Heterozygous     ATCAY  rs796133913   \n",
       "581       ['GT', G]  Heterozygous     CD2AP  rs797004904   \n",
       "582       ['TA', T]  Heterozygous     BRCA1    rs8176144   \n",
       "\n",
       "                         consequence  \\\n",
       "0                5_prime_UTR_variant   \n",
       "1                5_prime_UTR_variant   \n",
       "2                3_prime_UTR_variant   \n",
       "3    inframe_insertion', 'SO:0001627   \n",
       "4     inframe_deletion', 'SO:0001627   \n",
       "..                               ...   \n",
       "578              3_prime_UTR_variant   \n",
       "579              3_prime_UTR_variant   \n",
       "580              3_prime_UTR_variant   \n",
       "581              3_prime_UTR_variant   \n",
       "582                   intron_variant   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                          Uncertain_significance   \n",
       "1                          Uncertain_significance   \n",
       "2                                          Benign   \n",
       "3    Conflicting_interpretations_of_pathogenicity   \n",
       "4                                          Benign   \n",
       "..                                            ...   \n",
       "578                                 Likely_benign   \n",
       "579                                        Benign   \n",
       "580                        Uncertain_significance   \n",
       "581                                        Benign   \n",
       "582                                        Benign   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0                       Trichorhinophalangeal_syndrome   \n",
       "1                       Trichorhinophalangeal_syndrome   \n",
       "2    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "3    Hereditary_spastic_paraplegia_30|not_specified...   \n",
       "4                           not_specified|not_provided   \n",
       "..                                                 ...   \n",
       "578  Early_Infantile_Epileptic_Encephalopathy', '_A...   \n",
       "579                      Cayman_type_cerebellar_ataxia   \n",
       "580                      Cayman_type_cerebellar_ataxia   \n",
       "581                 Focal_segmental_glomerulosclerosis   \n",
       "582  Hereditary_breast_ovarian_cancer_syndrome|Brea...   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0               criteria_provided', '_single_submitter  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3    criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "4    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "578             criteria_provided', '_single_submitter  germline        indel   \n",
       "579             criteria_provided', '_single_submitter  germline        indel   \n",
       "580             criteria_provided', '_single_submitter  germline        indel   \n",
       "581             criteria_provided', '_single_submitter  germline        indel   \n",
       "582                           reviewed_by_expert_panel  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \\\n",
       "0         Insertion           MayBe No          No                  No   \n",
       "1          Deletion           MayBe No          No                  No   \n",
       "2          Deletion           MayBe No          No                  No   \n",
       "3         Insertion           MayBe No          No                 Yes   \n",
       "4          Deletion           MayBe No          No                 Yes   \n",
       "..              ...                ...         ...                 ...   \n",
       "578        Deletion           MayBe No          No                  No   \n",
       "579        Deletion           MayBe No          No                  No   \n",
       "580        Deletion           MayBe No          No                  No   \n",
       "581        Deletion           MayBe No          No                  No   \n",
       "582        Deletion           MayBe No          No                 Yes   \n",
       "\n",
       "    Gene_Match  \n",
       "0           No  \n",
       "1           No  \n",
       "2           No  \n",
       "3           No  \n",
       "4           No  \n",
       "..         ...  \n",
       "578         No  \n",
       "579         No  \n",
       "580         No  \n",
       "581         No  \n",
       "582         No  \n",
       "\n",
       "[583 rows x 15 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indel['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in indel['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            indel.loc[indel['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "indel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cb8090d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     556\n",
       "Yes     27\n",
       "Name: Gene_Match, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indel.Gene_Match.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a2b9243",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "834e518d",
   "metadata": {},
   "source": [
    "## InDel Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03d81d1d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d2185fc6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>SAMPLE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:150:29:29:1:28:96.55%:9.9776E-16:32:21:1:0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=65;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:109:65:65:35:30:46.15%:1.1519E-11:55:57:27...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:151:76:76:36:40:52.63%:7.6319E-16:58:56:28...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1398672</td>\n",
       "      <td>rs368050244;rs3831366</td>\n",
       "      <td>CTAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.925...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:131:60:60:26:34:56.67%:7.4725E-14:57:50:21...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1629572</td>\n",
       "      <td>rs112177324</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;G5;GENEINFO=M...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:50:22:21:9:13:59.09%:9.5813E-6:57:56:9:0:11:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2606</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=63;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.0002649...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:255:63:63:2:61:96.83%:3.4466E-34:74:55:2:0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2607</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:64:25:25:5:15:60%:3.8543E-7:64:55:5:0:14:1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2608</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155492733</td>\n",
       "      <td>rs1169019545;rs376271737</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=TMLH...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:81:17:17:1:16:94.12%:7.7134E-9:35:55:0:1:11:5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2609</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12786501</td>\n",
       "      <td>rs760255651</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=USP9...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>0/1:50:21:21:4:12:57.14%:8.0605E-6:42:49:3:1:10:2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2610</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...</td>\n",
       "      <td>1/1:71:18:18:2:15:83.33%:7.3277E-8:74:65:2:0:13:2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2611 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                      rsID     REF ALT QUAL FILTER  \\\n",
       "0     chr1    1043223                rs35881187     CCT   C    .   PASS   \n",
       "1     chr1    1299382               rs143128930      AG   A    .   PASS   \n",
       "2     chr1    1353987               rs140777846     CTG   C    .   PASS   \n",
       "3     chr1    1398672     rs368050244;rs3831366  CTAGAG   C    .   PASS   \n",
       "4     chr1    1629572               rs112177324      TG   T    .   PASS   \n",
       "...    ...        ...                       ...     ...  ..  ...    ...   \n",
       "2606  chrX  153913005      rs5904376;rs77485258       G  GC    .   PASS   \n",
       "2607  chrX  155228363               rs781912204       A  AT    .   PASS   \n",
       "2608  chrX  155492733  rs1169019545;rs376271737       T  TG    .   PASS   \n",
       "2609  chrY   12786501               rs760255651      CT   C    .   PASS   \n",
       "2610  chrY   21311242               rs202147921       A  AT    .   PASS   \n",
       "\n",
       "                                                   INFO  \\\n",
       "0     ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5208,0....   \n",
       "1     ADP=65;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....   \n",
       "2     ADP=76;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....   \n",
       "3     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.925...   \n",
       "4     ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;G5;GENEINFO=M...   \n",
       "...                                                 ...   \n",
       "2606  ADP=63;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.0002649...   \n",
       "2607  ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...   \n",
       "2608  ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=TMLH...   \n",
       "2609  ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=USP9...   \n",
       "2610  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...   \n",
       "\n",
       "                                                 FORMAT  \\\n",
       "0     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "1     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "3     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "4     GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "...                                                 ...   \n",
       "2606  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2607  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2608  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2609  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "2610  GT:GQ:SDP:DP:RD:AD:FREQ:PVAL:RBQ:ABQ:RDF:RDR:A...   \n",
       "\n",
       "                                                 SAMPLE  \n",
       "0     1/1:150:29:29:1:28:96.55%:9.9776E-16:32:21:1:0...  \n",
       "1     0/1:109:65:65:35:30:46.15%:1.1519E-11:55:57:27...  \n",
       "2     0/1:151:76:76:36:40:52.63%:7.6319E-16:58:56:28...  \n",
       "3     0/1:131:60:60:26:34:56.67%:7.4725E-14:57:50:21...  \n",
       "4     0/1:50:22:21:9:13:59.09%:9.5813E-6:57:56:9:0:11:2  \n",
       "...                                                 ...  \n",
       "2606  1/1:255:63:63:2:61:96.83%:3.4466E-34:74:55:2:0...  \n",
       "2607     0/1:64:25:25:5:15:60%:3.8543E-7:64:55:5:0:14:1  \n",
       "2608  1/1:81:17:17:1:16:94.12%:7.7134E-9:35:55:0:1:11:5  \n",
       "2609  0/1:50:21:21:4:12:57.14%:8.0605E-6:42:49:3:1:10:2  \n",
       "2610  1/1:71:18:18:2:15:83.33%:7.3277E-8:74:65:2:0:13:2  \n",
       "\n",
       "[2611 rows x 10 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHCDPRGPTTL6_annotated_indel.vcf', comment='#', sep= '\\t', header=None, low_memory=False)\n",
    "data.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "17fa70a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>150</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>96.55%</td>\n",
       "      <td>9.9776E-16</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=65;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>35</td>\n",
       "      <td>30</td>\n",
       "      <td>46.15%</td>\n",
       "      <td>1.1519E-11</td>\n",
       "      <td>27</td>\n",
       "      <td>8</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>151</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "      <td>36</td>\n",
       "      <td>40</td>\n",
       "      <td>52.63%</td>\n",
       "      <td>7.6319E-16</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>34</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1398672</td>\n",
       "      <td>rs368050244;rs3831366</td>\n",
       "      <td>CTAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.925...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>131</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>26</td>\n",
       "      <td>34</td>\n",
       "      <td>56.67%</td>\n",
       "      <td>7.4725E-14</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1629572</td>\n",
       "      <td>rs112177324</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;G5;GENEINFO=M...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>22</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>59.09%</td>\n",
       "      <td>9.5813E-6</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2606</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=63;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.0002649...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>2</td>\n",
       "      <td>61</td>\n",
       "      <td>96.83%</td>\n",
       "      <td>3.4466E-34</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2607</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>60%</td>\n",
       "      <td>3.8543E-7</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2608</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155492733</td>\n",
       "      <td>rs1169019545;rs376271737</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=TMLH...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>81</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>94.12%</td>\n",
       "      <td>7.7134E-9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2609</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12786501</td>\n",
       "      <td>rs760255651</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=USP9...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>57.14%</td>\n",
       "      <td>8.0605E-6</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2610</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>71</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>7.3277E-8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2611 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                      rsID     REF ALT QUAL FILTER  \\\n",
       "0     chr1    1043223                rs35881187     CCT   C    .   PASS   \n",
       "1     chr1    1299382               rs143128930      AG   A    .   PASS   \n",
       "2     chr1    1353987               rs140777846     CTG   C    .   PASS   \n",
       "3     chr1    1398672     rs368050244;rs3831366  CTAGAG   C    .   PASS   \n",
       "4     chr1    1629572               rs112177324      TG   T    .   PASS   \n",
       "...    ...        ...                       ...     ...  ..  ...    ...   \n",
       "2606  chrX  153913005      rs5904376;rs77485258       G  GC    .   PASS   \n",
       "2607  chrX  155228363               rs781912204       A  AT    .   PASS   \n",
       "2608  chrX  155492733  rs1169019545;rs376271737       T  TG    .   PASS   \n",
       "2609  chrY   12786501               rs760255651      CT   C    .   PASS   \n",
       "2610  chrY   21311242               rs202147921       A  AT    .   PASS   \n",
       "\n",
       "                                                   INFO   GT   GQ SDP  DP  RD  \\\n",
       "0     ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5208,0....  1/1  150  29  29   1   \n",
       "1     ADP=65;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....  0/1  109  65  65  35   \n",
       "2     ADP=76;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....  0/1  151  76  76  36   \n",
       "3     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.925...  0/1  131  60  60  26   \n",
       "4     ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;G5;GENEINFO=M...  0/1   50  22  21   9   \n",
       "...                                                 ...  ...  ...  ..  ..  ..   \n",
       "2606  ADP=63;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.0002649...  1/1  255  63  63   2   \n",
       "2607  ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...  0/1   64  25  25   5   \n",
       "2608  ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=TMLH...  1/1   81  17  17   1   \n",
       "2609  ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=USP9...  0/1   50  21  21   4   \n",
       "2610  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...  1/1   71  18  18   2   \n",
       "\n",
       "      AD    FREQ        PVAL RDF RDR ADF ADR  \n",
       "0     28  96.55%  9.9776E-16   1   0  25   3  \n",
       "1     30  46.15%  1.1519E-11  27   8  26   4  \n",
       "2     40  52.63%  7.6319E-16  28   8  34   6  \n",
       "3     34  56.67%  7.4725E-14  21   5  26   8  \n",
       "4     13  59.09%   9.5813E-6   9   0  11   2  \n",
       "...   ..     ...         ...  ..  ..  ..  ..  \n",
       "2606  61  96.83%  3.4466E-34   2   0  50  11  \n",
       "2607  15     60%   3.8543E-7   5   0  14   1  \n",
       "2608  16  94.12%   7.7134E-9   0   1  11   5  \n",
       "2609  12  57.14%   8.0605E-6   3   1  10   2  \n",
       "2610  15  83.33%   7.3277E-8   2   0  13   2  \n",
       "\n",
       "[2611 rows x 20 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_cols = data['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "data = pd.concat([data, sample_cols], axis=1)\n",
    "data = data[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "57a771e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5208,0.4792;COMMON=1;G5;GENEINFO=AGRN:375790;GNO;INT;KGPhase1;KGPhase3;PM;PMC;RS=35881187;RSPOS=1043224;RV;SAO=0;SLO;SSR=0;TOPMED=0.55968081039755351,0.44031918960244648;VC=DIV;VLD;VP=0x05012808000515013e000200;WGT=1;dbSNPBuildID=126'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.INFO.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "47911594",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>CDA</th>\n",
       "      <th>OTH</th>\n",
       "      <th>S3D</th>\n",
       "      <th>WTD</th>\n",
       "      <th>dbSNPBuildID</th>\n",
       "      <th>SLO</th>\n",
       "      <th>NSF</th>\n",
       "      <th>R3</th>\n",
       "      <th>R5</th>\n",
       "      <th>NSN</th>\n",
       "      <th>NSM</th>\n",
       "      <th>G5A</th>\n",
       "      <th>COMMON</th>\n",
       "      <th>RS</th>\n",
       "      <th>RV</th>\n",
       "      <th>TPA</th>\n",
       "      <th>CFL</th>\n",
       "      <th>GNO</th>\n",
       "      <th>VLD</th>\n",
       "      <th>ASP</th>\n",
       "      <th>ASS</th>\n",
       "      <th>Ref</th>\n",
       "      <th>U3</th>\n",
       "      <th>U5</th>\n",
       "      <th>TOPMED</th>\n",
       "      <th>WGT</th>\n",
       "      <th>MTP</th>\n",
       "      <th>LSD</th>\n",
       "      <th>NOC</th>\n",
       "      <th>DSS</th>\n",
       "      <th>SYN</th>\n",
       "      <th>KGPhase3</th>\n",
       "      <th>CAF</th>\n",
       "      <th>VC</th>\n",
       "      <th>MUT</th>\n",
       "      <th>KGPhase1</th>\n",
       "      <th>NOV</th>\n",
       "      <th>VP</th>\n",
       "      <th>SAO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5208,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>150</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>96.55%</td>\n",
       "      <td>9.9776E-16</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>3</td>\n",
       "      <td>ADP=29</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=126</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=35881187</td>\n",
       "      <td>RV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.55968081039755351,0.44031918960244648</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.5208,0.4792</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05012808000515013e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=AGRN:375790</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td>PMC</td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1043224</td>\n",
       "      <td></td>\n",
       "      <td>PM</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>AGRN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=65;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>35</td>\n",
       "      <td>30</td>\n",
       "      <td>46.15%</td>\n",
       "      <td>1.1519E-11</td>\n",
       "      <td>27</td>\n",
       "      <td>8</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>ADP=65</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=134</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=143128930</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.91126720183486238,0.08873279816513761</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.8806,0.1194</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x0500000a000515003e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=ACAP3:116983</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1299383</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ACAP3:116983</td>\n",
       "      <td>ACAP3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=76;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>151</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "      <td>36</td>\n",
       "      <td>40</td>\n",
       "      <td>52.63%</td>\n",
       "      <td>7.6319E-16</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>34</td>\n",
       "      <td>6</td>\n",
       "      <td>ADP=76</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=134</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>G5A</td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=140777846</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.13763857033639143,0.86236142966360856</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.1793,0.8207</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005170126000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=MXRA8:54587</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1353988</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>MXRA8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1398672</td>\n",
       "      <td>rs368050244;rs3831366</td>\n",
       "      <td>CTAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.925...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>131</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>26</td>\n",
       "      <td>34</td>\n",
       "      <td>56.67%</td>\n",
       "      <td>7.4725E-14</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>8</td>\n",
       "      <td>ADP=60</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=107,138</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1,1</td>\n",
       "      <td>RS=3831366,368050244</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td>ASS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.94513729612640163,0.05486270387359836</td>\n",
       "      <td>WGT=1,1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.9259,0.07408,0.9259,0.07408</td>\n",
       "      <td>VC=DIV,DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05012822000515013e000200,0x0500000a000500...</td>\n",
       "      <td>SAO=0,0</td>\n",
       "      <td>GENEINFO=LOC148413:148413|CCNL2:81669,LOC14841...</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td>PMC</td>\n",
       "      <td>SSR=0,0</td>\n",
       "      <td>RSPOS=1398673,1398677</td>\n",
       "      <td></td>\n",
       "      <td>PM</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>LOC148413:148413|CCNL2:81669,LOC148413:148413|...</td>\n",
       "      <td>LOC148413,CCNL2,CCNL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1629572</td>\n",
       "      <td>rs112177324</td>\n",
       "      <td>null</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;G5;GENEINFO=M...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>22</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>59.09%</td>\n",
       "      <td>9.5813E-6</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=21</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=132</td>\n",
       "      <td>SLO</td>\n",
       "      <td>NSF</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=112177324</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.75031058868501529,0.24968941131498470</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05010008120515013e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=MIB2:142678</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1629573</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>MIB2:142678</td>\n",
       "      <td>MIB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2606</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=63;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.0002649...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>2</td>\n",
       "      <td>61</td>\n",
       "      <td>96.83%</td>\n",
       "      <td>3.4466E-34</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>11</td>\n",
       "      <td>ADP=63</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=131,126</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=0,0</td>\n",
       "      <td>RS=77485258,5904376</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.01672400611620795,0.98327599388379204</td>\n",
       "      <td>WGT=1,1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.0002649,0.9997,0.0002649,0.9997</td>\n",
       "      <td>VC=DIV,DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05000008000501003e000200,0x05010008000500...</td>\n",
       "      <td>SAO=0,0</td>\n",
       "      <td>GENEINFO=ARHGAP4:393,ARHGAP4:393</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0,0</td>\n",
       "      <td>RSPOS=153913005,153913008</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ARHGAP4:393,ARHGAP4:393</td>\n",
       "      <td>ARHGAP4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2607</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>60%</td>\n",
       "      <td>3.8543E-7</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=25</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=144</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=781912204</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.43169438073394495,0.56314506880733944...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NOV</td>\n",
       "      <td>VP=0x050000080005000002000204</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=VBP1:7411</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=155228363</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VBP1:7411</td>\n",
       "      <td>VBP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2608</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155492733</td>\n",
       "      <td>rs1169019545;rs376271737</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=TMLH...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>81</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>94.12%</td>\n",
       "      <td>7.7134E-9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>ADP=17</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=151,138</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=1169019545,376271737</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.35031218144750254,0.64968781855249745</td>\n",
       "      <td>WGT=1,1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV,DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000080005000002000200,0x05000008000500...</td>\n",
       "      <td>SAO=0,0</td>\n",
       "      <td>GENEINFO=TMLHE:55217|TMLHE-AS1:100507404,TMLHE...</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0,0</td>\n",
       "      <td>RSPOS=155492733,155492734</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TMLHE:55217|TMLHE-AS1:100507404,TMLHE:55217|TM...</td>\n",
       "      <td>TMLHE,TMLHE-AS1,TMLHE-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2609</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12786501</td>\n",
       "      <td>rs760255651</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=USP9...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>57.14%</td>\n",
       "      <td>8.0605E-6</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=21</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=144</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=760255651</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000080005000002000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>GENEINFO=USP9Y:8287</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=12786503</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>USP9Y:8287</td>\n",
       "      <td>USP9Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2610</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>71</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>7.3277E-8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=18</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=137</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=202147921</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000000005000002000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=21311242</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2611 rows × 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                      rsID     REF ALT QUAL FILTER  \\\n",
       "0     chr1    1043223                rs35881187     CCT   C    .   PASS   \n",
       "1     chr1    1299382               rs143128930      AG   A    .   PASS   \n",
       "2     chr1    1353987               rs140777846     CTG   C    .   PASS   \n",
       "3     chr1    1398672     rs368050244;rs3831366  CTAGAG   C    .   PASS   \n",
       "4     chr1    1629572               rs112177324    null   T    .   PASS   \n",
       "...    ...        ...                       ...     ...  ..  ...    ...   \n",
       "2606  chrX  153913005      rs5904376;rs77485258       G  GC    .   PASS   \n",
       "2607  chrX  155228363               rs781912204       A  AT    .   PASS   \n",
       "2608  chrX  155492733  rs1169019545;rs376271737       T  TG    .   PASS   \n",
       "2609  chrY   12786501               rs760255651      CT   C    .   PASS   \n",
       "2610  chrY   21311242               rs202147921       A  AT    .   PASS   \n",
       "\n",
       "                                                   INFO   GT   GQ SDP  DP  RD  \\\n",
       "0     ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.5208,0....  1/1  150  29  29   1   \n",
       "1     ADP=65;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.8806,0....  0/1  109  65  65  35   \n",
       "2     ADP=76;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.1793,0....  0/1  151  76  76  36   \n",
       "3     ADP=60;WT=0;HET=1;HOM=0;NC=0;ASP;ASS;CAF=0.925...  0/1  131  60  60  26   \n",
       "4     ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;G5;GENEINFO=M...  0/1   50  22  21   9   \n",
       "...                                                 ...  ...  ...  ..  ..  ..   \n",
       "2606  ADP=63;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.0002649...  1/1  255  63  63   2   \n",
       "2607  ADP=25;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=VBP1...  0/1   64  25  25   5   \n",
       "2608  ADP=17;WT=0;HET=0;HOM=1;NC=0;ASP;GENEINFO=TMLH...  1/1   81  17  17   1   \n",
       "2609  ADP=21;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=USP9...  0/1   50  21  21   4   \n",
       "2610  ADP=18;WT=0;HET=0;HOM=1;NC=0;ASP;RS=202147921;...  1/1   71  18  18   2   \n",
       "\n",
       "      AD    FREQ        PVAL RDF RDR ADF ADR     ADP    WT    HET    HOM  \\\n",
       "0     28  96.55%  9.9776E-16   1   0  25   3  ADP=29  WT=0  HET=0  HOM=1   \n",
       "1     30  46.15%  1.1519E-11  27   8  26   4  ADP=65  WT=0  HET=1  HOM=0   \n",
       "2     40  52.63%  7.6319E-16  28   8  34   6  ADP=76  WT=0  HET=1  HOM=0   \n",
       "3     34  56.67%  7.4725E-14  21   5  26   8  ADP=60  WT=0  HET=1  HOM=0   \n",
       "4     13  59.09%   9.5813E-6   9   0  11   2  ADP=21  WT=0  HET=1  HOM=0   \n",
       "...   ..     ...         ...  ..  ..  ..  ..     ...   ...    ...    ...   \n",
       "2606  61  96.83%  3.4466E-34   2   0  50  11  ADP=63  WT=0  HET=0  HOM=1   \n",
       "2607  15     60%   3.8543E-7   5   0  14   1  ADP=25  WT=0  HET=1  HOM=0   \n",
       "2608  16  94.12%   7.7134E-9   0   1  11   5  ADP=17  WT=0  HET=0  HOM=1   \n",
       "2609  12  57.14%   8.0605E-6   3   1  10   2  ADP=21  WT=0  HET=1  HOM=0   \n",
       "2610  15  83.33%   7.3277E-8   2   0  13   2  ADP=18  WT=0  HET=0  HOM=1   \n",
       "\n",
       "        NC CDA OTH S3D WTD          dbSNPBuildID  SLO  NSF R3  R5 NSN NSM  \\\n",
       "0     NC=0                      dbSNPBuildID=126  SLO                       \n",
       "1     NC=0                      dbSNPBuildID=134               R5           \n",
       "2     NC=0                      dbSNPBuildID=134  SLO                       \n",
       "3     NC=0                  dbSNPBuildID=107,138  SLO          R5           \n",
       "4     NC=0                      dbSNPBuildID=132  SLO  NSF                  \n",
       "...    ...  ..  ..  ..  ..                   ...  ...  ... ..  ..  ..  ..   \n",
       "2606  NC=0                  dbSNPBuildID=131,126  SLO                       \n",
       "2607  NC=0                      dbSNPBuildID=144                            \n",
       "2608  NC=0                  dbSNPBuildID=151,138                            \n",
       "2609  NC=0                      dbSNPBuildID=144                            \n",
       "2610  NC=0                      dbSNPBuildID=137                            \n",
       "\n",
       "      G5A      COMMON                       RS  RV TPA CFL  GNO  VLD  ASP  \\\n",
       "0            COMMON=1              RS=35881187  RV          GNO  VLD  ASP   \n",
       "1            COMMON=1             RS=143128930                   VLD  ASP   \n",
       "2     G5A    COMMON=1             RS=140777846              GNO  VLD  ASP   \n",
       "3          COMMON=1,1     RS=3831366,368050244              GNO  VLD  ASP   \n",
       "4                                 RS=112177324              GNO  VLD  ASP   \n",
       "...   ...         ...                      ...  ..  ..  ..  ...  ...  ...   \n",
       "2606       COMMON=0,0      RS=77485258,5904376              GNO       ASP   \n",
       "2607                              RS=781912204                        ASP   \n",
       "2608                   RS=1169019545,376271737                        ASP   \n",
       "2609                              RS=760255651                        ASP   \n",
       "2610                              RS=202147921                        ASP   \n",
       "\n",
       "      ASS Ref U3 U5                                             TOPMED  \\\n",
       "0                       TOPMED=0.55968081039755351,0.44031918960244648   \n",
       "1                       TOPMED=0.91126720183486238,0.08873279816513761   \n",
       "2                       TOPMED=0.13763857033639143,0.86236142966360856   \n",
       "3     ASS               TOPMED=0.94513729612640163,0.05486270387359836   \n",
       "4                       TOPMED=0.75031058868501529,0.24968941131498470   \n",
       "...   ...  .. .. ..                                                ...   \n",
       "2606                    TOPMED=0.01672400611620795,0.98327599388379204   \n",
       "2607                 TOPMED=0.43169438073394495,0.56314506880733944...   \n",
       "2608                    TOPMED=0.35031218144750254,0.64968781855249745   \n",
       "2609                                                                     \n",
       "2610                                                                     \n",
       "\n",
       "          WGT MTP LSD NOC DSS SYN  KGPhase3  \\\n",
       "0       WGT=1                      KGPhase3   \n",
       "1       WGT=1                      KGPhase3   \n",
       "2       WGT=1                      KGPhase3   \n",
       "3     WGT=1,1                      KGPhase3   \n",
       "4       WGT=1                      KGPhase3   \n",
       "...       ...  ..  ..  ..  ..  ..       ...   \n",
       "2606  WGT=1,1                      KGPhase3   \n",
       "2607    WGT=1                                 \n",
       "2608  WGT=1,1                                 \n",
       "2609    WGT=1                                 \n",
       "2610    WGT=1                                 \n",
       "\n",
       "                                        CAF          VC MUT  KGPhase1  NOV  \\\n",
       "0                         CAF=0.5208,0.4792      VC=DIV      KGPhase1        \n",
       "1                         CAF=0.8806,0.1194      VC=DIV      KGPhase1        \n",
       "2                         CAF=0.1793,0.8207      VC=DIV                      \n",
       "3         CAF=0.9259,0.07408,0.9259,0.07408  VC=DIV,DIV      KGPhase1        \n",
       "4                                                VC=DIV      KGPhase1        \n",
       "...                                     ...         ...  ..       ...  ...   \n",
       "2606  CAF=0.0002649,0.9997,0.0002649,0.9997  VC=DIV,DIV      KGPhase1        \n",
       "2607                                             VC=DIV                NOV   \n",
       "2608                                         VC=DIV,DIV                      \n",
       "2609                                             VC=DIV                      \n",
       "2610                                             VC=DIV                      \n",
       "\n",
       "                                                     VP      SAO  \\\n",
       "0                         VP=0x05012808000515013e000200    SAO=0   \n",
       "1                         VP=0x0500000a000515003e000200    SAO=0   \n",
       "2                         VP=0x050100080005170126000200    SAO=0   \n",
       "3     VP=0x05012822000515013e000200,0x0500000a000500...  SAO=0,0   \n",
       "4                         VP=0x05010008120515013e000200    SAO=0   \n",
       "...                                                 ...      ...   \n",
       "2606  VP=0x05000008000501003e000200,0x05010008000500...  SAO=0,0   \n",
       "2607                      VP=0x050000080005000002000204    SAO=0   \n",
       "2608  VP=0x050000080005000002000200,0x05000008000500...  SAO=0,0   \n",
       "2609                      VP=0x050000080005000002000200    SAO=0   \n",
       "2610                      VP=0x050000000005000002000200    SAO=0   \n",
       "\n",
       "                                               GENEINFO  INT  G5 OM  PMC  \\\n",
       "0                                  GENEINFO=AGRN:375790  INT  G5     PMC   \n",
       "1                                 GENEINFO=ACAP3:116983  INT  G5           \n",
       "2                                  GENEINFO=MXRA8:54587  INT  G5           \n",
       "3     GENEINFO=LOC148413:148413|CCNL2:81669,LOC14841...  INT  G5     PMC   \n",
       "4                                  GENEINFO=MIB2:142678  INT  G5           \n",
       "...                                                 ...  ...  .. ..  ...   \n",
       "2606                   GENEINFO=ARHGAP4:393,ARHGAP4:393  INT  G5           \n",
       "2607                                 GENEINFO=VBP1:7411  INT               \n",
       "2608  GENEINFO=TMLHE:55217|TMLHE-AS1:100507404,TMLHE...  INT               \n",
       "2609                                GENEINFO=USP9Y:8287  INT               \n",
       "2610                                                                       \n",
       "\n",
       "          SSR                      RSPOS HD  PM ClinVar ClinVar_CLNSIG  \\\n",
       "0       SSR=0              RSPOS=1043224     PM                          \n",
       "1       SSR=0              RSPOS=1299383                                 \n",
       "2       SSR=0              RSPOS=1353988                                 \n",
       "3     SSR=0,0      RSPOS=1398673,1398677     PM                          \n",
       "4       SSR=0              RSPOS=1629573                                 \n",
       "...       ...                        ... ..  ..     ...            ...   \n",
       "2606  SSR=0,0  RSPOS=153913005,153913008                                 \n",
       "2607    SSR=0            RSPOS=155228363                                 \n",
       "2608  SSR=0,0  RSPOS=155492733,155492734                                 \n",
       "2609    SSR=0             RSPOS=12786503                                 \n",
       "2610    SSR=0             RSPOS=21311242                                 \n",
       "\n",
       "                                              Gene_Name  \\\n",
       "0                                           AGRN:375790   \n",
       "1                                          ACAP3:116983   \n",
       "2                                           MXRA8:54587   \n",
       "3     LOC148413:148413|CCNL2:81669,LOC148413:148413|...   \n",
       "4                                           MIB2:142678   \n",
       "...                                                 ...   \n",
       "2606                            ARHGAP4:393,ARHGAP4:393   \n",
       "2607                                          VBP1:7411   \n",
       "2608  TMLHE:55217|TMLHE-AS1:100507404,TMLHE:55217|TM...   \n",
       "2609                                         USP9Y:8287   \n",
       "2610                                                NaN   \n",
       "\n",
       "                      Gene Name  \n",
       "0                          AGRN  \n",
       "1                         ACAP3  \n",
       "2                         MXRA8  \n",
       "3         LOC148413,CCNL2,CCNL2  \n",
       "4                          MIB2  \n",
       "...                         ...  \n",
       "2606                    ARHGAP4  \n",
       "2607                       VBP1  \n",
       "2608  TMLHE,TMLHE-AS1,TMLHE-AS1  \n",
       "2609                      USP9Y  \n",
       "2610                             \n",
       "\n",
       "[2611 rows x 77 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create empty columns\n",
    "columns = ['ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD', 'dbSNPBuildID', 'SLO',\n",
    "           'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON', 'RS', 'RV', 'TPA', 'CFL', 'GNO',\n",
    "           'VLD', 'ASP', 'ASS', 'Ref', 'U3', 'U5', 'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC',\n",
    "           'DSS', 'SYN', 'KGPhase3', 'CAF', 'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO',\n",
    "           'GENEINFO', 'INT', 'G5', 'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar',\n",
    "           'ClinVar_CLNSIG']\n",
    "\n",
    "for col in columns:\n",
    "    data[col] = ''\n",
    "\n",
    "# Populate columns based on 'info' values\n",
    "for i, row in data.iterrows():\n",
    "    info = row['INFO']\n",
    "    items = info.split(';')\n",
    "    for item in items:\n",
    "        key_value = item.split('=')\n",
    "        key = key_value[0]\n",
    "        if key in columns:\n",
    "            if len(key_value) > 1:\n",
    "                value = key_value[1]\n",
    "                data.at[i, key] = f\"{key}={value}\"\n",
    "            else:\n",
    "                data.at[i, key] = key\n",
    "        else:\n",
    "            data.at[i, key] = 'null'\n",
    "\n",
    "\n",
    "            \n",
    "data[\"Gene_Name\"] = data[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "data['Gene Name'] = data['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "\n",
    "# Print the resulting DataFrame\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e2dfa005",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT',\n",
       "       'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
       "       'ADR', 'ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD',\n",
       "       'dbSNPBuildID', 'SLO', 'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON',\n",
       "       'RS', 'RV', 'TPA', 'CFL', 'GNO', 'VLD', 'ASP', 'ASS', 'Ref', 'U3', 'U5',\n",
       "       'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC', 'DSS', 'SYN', 'KGPhase3', 'CAF',\n",
       "       'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO', 'GENEINFO', 'INT', 'G5',\n",
       "       'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar', 'ClinVar_CLNSIG',\n",
       "       'Gene_Name', 'Gene Name'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "6d3a7a8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>CDA</th>\n",
       "      <th>OTH</th>\n",
       "      <th>S3D</th>\n",
       "      <th>WTD</th>\n",
       "      <th>dbSNPBuildID</th>\n",
       "      <th>SLO</th>\n",
       "      <th>NSF</th>\n",
       "      <th>R3</th>\n",
       "      <th>R5</th>\n",
       "      <th>NSN</th>\n",
       "      <th>NSM</th>\n",
       "      <th>G5A</th>\n",
       "      <th>COMMON</th>\n",
       "      <th>RS</th>\n",
       "      <th>RV</th>\n",
       "      <th>TPA</th>\n",
       "      <th>CFL</th>\n",
       "      <th>GNO</th>\n",
       "      <th>VLD</th>\n",
       "      <th>ASP</th>\n",
       "      <th>ASS</th>\n",
       "      <th>Ref</th>\n",
       "      <th>U3</th>\n",
       "      <th>U5</th>\n",
       "      <th>TOPMED</th>\n",
       "      <th>WGT</th>\n",
       "      <th>MTP</th>\n",
       "      <th>LSD</th>\n",
       "      <th>NOC</th>\n",
       "      <th>DSS</th>\n",
       "      <th>SYN</th>\n",
       "      <th>KGPhase3</th>\n",
       "      <th>CAF</th>\n",
       "      <th>VC</th>\n",
       "      <th>MUT</th>\n",
       "      <th>KGPhase1</th>\n",
       "      <th>NOV</th>\n",
       "      <th>VP</th>\n",
       "      <th>SAO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>150</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>96.55%</td>\n",
       "      <td>9.9776E-16</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>3</td>\n",
       "      <td>ADP=29</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=126</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=35881187</td>\n",
       "      <td>RV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.55968081039755351,0.44031918960244648</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.5208,0.4792</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05012808000515013e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td>PMC</td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1043224</td>\n",
       "      <td></td>\n",
       "      <td>PM</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>35</td>\n",
       "      <td>30</td>\n",
       "      <td>46.15%</td>\n",
       "      <td>1.1519E-11</td>\n",
       "      <td>27</td>\n",
       "      <td>8</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>ADP=65</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=134</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=143128930</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.91126720183486238,0.08873279816513761</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.8806,0.1194</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x0500000a000515003e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1299383</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>0/1</td>\n",
       "      <td>151</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "      <td>36</td>\n",
       "      <td>40</td>\n",
       "      <td>52.63%</td>\n",
       "      <td>7.6319E-16</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>34</td>\n",
       "      <td>6</td>\n",
       "      <td>ADP=76</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=134</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>G5A</td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=140777846</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.13763857033639143,0.86236142966360856</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.1793,0.8207</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005170126000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1353988</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1398672</td>\n",
       "      <td>LOC148413,CCNL2,CCNL2</td>\n",
       "      <td>rs368050244;rs3831366</td>\n",
       "      <td>CTAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>0/1</td>\n",
       "      <td>131</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>26</td>\n",
       "      <td>34</td>\n",
       "      <td>56.67%</td>\n",
       "      <td>7.4725E-14</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>8</td>\n",
       "      <td>ADP=60</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=107,138</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1,1</td>\n",
       "      <td>RS=3831366,368050244</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td>ASS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.94513729612640163,0.05486270387359836</td>\n",
       "      <td>WGT=1,1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.9259,0.07408,0.9259,0.07408</td>\n",
       "      <td>VC=DIV,DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05012822000515013e000200,0x0500000a000500...</td>\n",
       "      <td>SAO=0,0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td>PMC</td>\n",
       "      <td>SSR=0,0</td>\n",
       "      <td>RSPOS=1398673,1398677</td>\n",
       "      <td></td>\n",
       "      <td>PM</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1629572</td>\n",
       "      <td>MIB2</td>\n",
       "      <td>rs112177324</td>\n",
       "      <td>null</td>\n",
       "      <td>T</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>22</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>59.09%</td>\n",
       "      <td>9.5813E-6</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=21</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=132</td>\n",
       "      <td>SLO</td>\n",
       "      <td>NSF</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=112177324</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.75031058868501529,0.24968941131498470</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05010008120515013e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1629573</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2606</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>ARHGAP4</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>2</td>\n",
       "      <td>61</td>\n",
       "      <td>96.83%</td>\n",
       "      <td>3.4466E-34</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>11</td>\n",
       "      <td>ADP=63</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=131,126</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=0,0</td>\n",
       "      <td>RS=77485258,5904376</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.01672400611620795,0.98327599388379204</td>\n",
       "      <td>WGT=1,1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.0002649,0.9997,0.0002649,0.9997</td>\n",
       "      <td>VC=DIV,DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05000008000501003e000200,0x05010008000500...</td>\n",
       "      <td>SAO=0,0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0,0</td>\n",
       "      <td>RSPOS=153913005,153913008</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2607</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>VBP1</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>60%</td>\n",
       "      <td>3.8543E-7</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=25</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=144</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=781912204</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.43169438073394495,0.56314506880733944...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NOV</td>\n",
       "      <td>VP=0x050000080005000002000204</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=155228363</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2608</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155492733</td>\n",
       "      <td>TMLHE,TMLHE-AS1,TMLHE-AS1</td>\n",
       "      <td>rs1169019545;rs376271737</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>1/1</td>\n",
       "      <td>81</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>94.12%</td>\n",
       "      <td>7.7134E-9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>ADP=17</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=151,138</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=1169019545,376271737</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.35031218144750254,0.64968781855249745</td>\n",
       "      <td>WGT=1,1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV,DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000080005000002000200,0x05000008000500...</td>\n",
       "      <td>SAO=0,0</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0,0</td>\n",
       "      <td>RSPOS=155492733,155492734</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2609</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12786501</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>rs760255651</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>57.14%</td>\n",
       "      <td>8.0605E-6</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=21</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=144</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=760255651</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000080005000002000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=12786503</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2610</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td></td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>1/1</td>\n",
       "      <td>71</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>7.3277E-8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=18</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=137</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=202147921</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000000005000002000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=21311242</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2611 rows × 72 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                  Gene Name                      rsID  \\\n",
       "0     chr1    1043223                       AGRN                rs35881187   \n",
       "1     chr1    1299382                      ACAP3               rs143128930   \n",
       "2     chr1    1353987                      MXRA8               rs140777846   \n",
       "3     chr1    1398672      LOC148413,CCNL2,CCNL2     rs368050244;rs3831366   \n",
       "4     chr1    1629572                       MIB2               rs112177324   \n",
       "...    ...        ...                        ...                       ...   \n",
       "2606  chrX  153913005                    ARHGAP4      rs5904376;rs77485258   \n",
       "2607  chrX  155228363                       VBP1               rs781912204   \n",
       "2608  chrX  155492733  TMLHE,TMLHE-AS1,TMLHE-AS1  rs1169019545;rs376271737   \n",
       "2609  chrY   12786501                      USP9Y               rs760255651   \n",
       "2610  chrY   21311242                                          rs202147921   \n",
       "\n",
       "         REF ALT   GT   GQ SDP  DP  RD  AD    FREQ        PVAL RDF RDR ADF  \\\n",
       "0        CCT   C  1/1  150  29  29   1  28  96.55%  9.9776E-16   1   0  25   \n",
       "1         AG   A  0/1  109  65  65  35  30  46.15%  1.1519E-11  27   8  26   \n",
       "2        CTG   C  0/1  151  76  76  36  40  52.63%  7.6319E-16  28   8  34   \n",
       "3     CTAGAG   C  0/1  131  60  60  26  34  56.67%  7.4725E-14  21   5  26   \n",
       "4       null   T  0/1   50  22  21   9  13  59.09%   9.5813E-6   9   0  11   \n",
       "...      ...  ..  ...  ...  ..  ..  ..  ..     ...         ...  ..  ..  ..   \n",
       "2606       G  GC  1/1  255  63  63   2  61  96.83%  3.4466E-34   2   0  50   \n",
       "2607       A  AT  0/1   64  25  25   5  15     60%   3.8543E-7   5   0  14   \n",
       "2608       T  TG  1/1   81  17  17   1  16  94.12%   7.7134E-9   0   1  11   \n",
       "2609      CT   C  0/1   50  21  21   4  12  57.14%   8.0605E-6   3   1  10   \n",
       "2610       A  AT  1/1   71  18  18   2  15  83.33%   7.3277E-8   2   0  13   \n",
       "\n",
       "     ADR     ADP    WT    HET    HOM    NC CDA OTH S3D WTD  \\\n",
       "0      3  ADP=29  WT=0  HET=0  HOM=1  NC=0                   \n",
       "1      4  ADP=65  WT=0  HET=1  HOM=0  NC=0                   \n",
       "2      6  ADP=76  WT=0  HET=1  HOM=0  NC=0                   \n",
       "3      8  ADP=60  WT=0  HET=1  HOM=0  NC=0                   \n",
       "4      2  ADP=21  WT=0  HET=1  HOM=0  NC=0                   \n",
       "...   ..     ...   ...    ...    ...   ...  ..  ..  ..  ..   \n",
       "2606  11  ADP=63  WT=0  HET=0  HOM=1  NC=0                   \n",
       "2607   1  ADP=25  WT=0  HET=1  HOM=0  NC=0                   \n",
       "2608   5  ADP=17  WT=0  HET=0  HOM=1  NC=0                   \n",
       "2609   2  ADP=21  WT=0  HET=1  HOM=0  NC=0                   \n",
       "2610   2  ADP=18  WT=0  HET=0  HOM=1  NC=0                   \n",
       "\n",
       "              dbSNPBuildID  SLO  NSF R3  R5 NSN NSM  G5A      COMMON  \\\n",
       "0         dbSNPBuildID=126  SLO                             COMMON=1   \n",
       "1         dbSNPBuildID=134               R5                 COMMON=1   \n",
       "2         dbSNPBuildID=134  SLO                      G5A    COMMON=1   \n",
       "3     dbSNPBuildID=107,138  SLO          R5               COMMON=1,1   \n",
       "4         dbSNPBuildID=132  SLO  NSF                                   \n",
       "...                    ...  ...  ... ..  ..  ..  ..  ...         ...   \n",
       "2606  dbSNPBuildID=131,126  SLO                           COMMON=0,0   \n",
       "2607      dbSNPBuildID=144                                             \n",
       "2608  dbSNPBuildID=151,138                                             \n",
       "2609      dbSNPBuildID=144                                             \n",
       "2610      dbSNPBuildID=137                                             \n",
       "\n",
       "                           RS  RV TPA CFL  GNO  VLD  ASP  ASS Ref U3 U5  \\\n",
       "0                 RS=35881187  RV          GNO  VLD  ASP                  \n",
       "1                RS=143128930                   VLD  ASP                  \n",
       "2                RS=140777846              GNO  VLD  ASP                  \n",
       "3        RS=3831366,368050244              GNO  VLD  ASP  ASS             \n",
       "4                RS=112177324              GNO  VLD  ASP                  \n",
       "...                       ...  ..  ..  ..  ...  ...  ...  ...  .. .. ..   \n",
       "2606      RS=77485258,5904376              GNO       ASP                  \n",
       "2607             RS=781912204                        ASP                  \n",
       "2608  RS=1169019545,376271737                        ASP                  \n",
       "2609             RS=760255651                        ASP                  \n",
       "2610             RS=202147921                        ASP                  \n",
       "\n",
       "                                                 TOPMED      WGT MTP LSD NOC  \\\n",
       "0        TOPMED=0.55968081039755351,0.44031918960244648    WGT=1               \n",
       "1        TOPMED=0.91126720183486238,0.08873279816513761    WGT=1               \n",
       "2        TOPMED=0.13763857033639143,0.86236142966360856    WGT=1               \n",
       "3        TOPMED=0.94513729612640163,0.05486270387359836  WGT=1,1               \n",
       "4        TOPMED=0.75031058868501529,0.24968941131498470    WGT=1               \n",
       "...                                                 ...      ...  ..  ..  ..   \n",
       "2606     TOPMED=0.01672400611620795,0.98327599388379204  WGT=1,1               \n",
       "2607  TOPMED=0.43169438073394495,0.56314506880733944...    WGT=1               \n",
       "2608     TOPMED=0.35031218144750254,0.64968781855249745  WGT=1,1               \n",
       "2609                                                       WGT=1               \n",
       "2610                                                       WGT=1               \n",
       "\n",
       "     DSS SYN  KGPhase3                                    CAF          VC MUT  \\\n",
       "0             KGPhase3                      CAF=0.5208,0.4792      VC=DIV       \n",
       "1             KGPhase3                      CAF=0.8806,0.1194      VC=DIV       \n",
       "2             KGPhase3                      CAF=0.1793,0.8207      VC=DIV       \n",
       "3             KGPhase3      CAF=0.9259,0.07408,0.9259,0.07408  VC=DIV,DIV       \n",
       "4             KGPhase3                                             VC=DIV       \n",
       "...   ..  ..       ...                                    ...         ...  ..   \n",
       "2606          KGPhase3  CAF=0.0002649,0.9997,0.0002649,0.9997  VC=DIV,DIV       \n",
       "2607                                                               VC=DIV       \n",
       "2608                                                           VC=DIV,DIV       \n",
       "2609                                                               VC=DIV       \n",
       "2610                                                               VC=DIV       \n",
       "\n",
       "      KGPhase1  NOV                                                 VP  \\\n",
       "0     KGPhase1                           VP=0x05012808000515013e000200   \n",
       "1     KGPhase1                           VP=0x0500000a000515003e000200   \n",
       "2                                        VP=0x050100080005170126000200   \n",
       "3     KGPhase1       VP=0x05012822000515013e000200,0x0500000a000500...   \n",
       "4     KGPhase1                           VP=0x05010008120515013e000200   \n",
       "...        ...  ...                                                ...   \n",
       "2606  KGPhase1       VP=0x05000008000501003e000200,0x05010008000500...   \n",
       "2607            NOV                      VP=0x050000080005000002000204   \n",
       "2608                 VP=0x050000080005000002000200,0x05000008000500...   \n",
       "2609                                     VP=0x050000080005000002000200   \n",
       "2610                                     VP=0x050000000005000002000200   \n",
       "\n",
       "          SAO  INT  G5 OM  PMC      SSR                      RSPOS HD  PM  \\\n",
       "0       SAO=0  INT  G5     PMC    SSR=0              RSPOS=1043224     PM   \n",
       "1       SAO=0  INT  G5            SSR=0              RSPOS=1299383          \n",
       "2       SAO=0  INT  G5            SSR=0              RSPOS=1353988          \n",
       "3     SAO=0,0  INT  G5     PMC  SSR=0,0      RSPOS=1398673,1398677     PM   \n",
       "4       SAO=0  INT  G5            SSR=0              RSPOS=1629573          \n",
       "...       ...  ...  .. ..  ...      ...                        ... ..  ..   \n",
       "2606  SAO=0,0  INT  G5          SSR=0,0  RSPOS=153913005,153913008          \n",
       "2607    SAO=0  INT                SSR=0            RSPOS=155228363          \n",
       "2608  SAO=0,0  INT              SSR=0,0  RSPOS=155492733,155492734          \n",
       "2609    SAO=0  INT                SSR=0             RSPOS=12786503          \n",
       "2610    SAO=0                     SSR=0             RSPOS=21311242          \n",
       "\n",
       "     ClinVar ClinVar_CLNSIG  \n",
       "0                            \n",
       "1                            \n",
       "2                            \n",
       "3                            \n",
       "4                            \n",
       "...      ...            ...  \n",
       "2606                         \n",
       "2607                         \n",
       "2608                         \n",
       "2609                         \n",
       "2610                         \n",
       "\n",
       "[2611 rows x 72 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = data[['CHROM', 'POS', 'Gene Name', 'rsID', 'REF', 'ALT', 'GT',\n",
    "       'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'ADP', 'WT', 'HET', 'HOM', 'NC', 'CDA', 'OTH', 'S3D', 'WTD',\n",
    "       'dbSNPBuildID', 'SLO', 'NSF', 'R3', 'R5', 'NSN', 'NSM', 'G5A', 'COMMON',\n",
    "       'RS', 'RV', 'TPA', 'CFL', 'GNO', 'VLD', 'ASP', 'ASS', 'Ref', 'U3', 'U5',\n",
    "       'TOPMED', 'WGT', 'MTP', 'LSD', 'NOC', 'DSS', 'SYN', 'KGPhase3', 'CAF',\n",
    "       'VC', 'MUT', 'KGPhase1', 'NOV', 'VP', 'SAO', 'INT', 'G5',\n",
    "       'OM', 'PMC', 'SSR', 'RSPOS', 'HD', 'PM', 'ClinVar', 'ClinVar_CLNSIG']]\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9bfdcb51",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "    2611\n",
       "Name: ClinVar_CLNSIG, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.ClinVar_CLNSIG.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "00ec7866",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>CDA</th>\n",
       "      <th>OTH</th>\n",
       "      <th>S3D</th>\n",
       "      <th>WTD</th>\n",
       "      <th>dbSNPBuildID</th>\n",
       "      <th>SLO</th>\n",
       "      <th>NSF</th>\n",
       "      <th>R3</th>\n",
       "      <th>R5</th>\n",
       "      <th>NSN</th>\n",
       "      <th>NSM</th>\n",
       "      <th>G5A</th>\n",
       "      <th>COMMON</th>\n",
       "      <th>RS</th>\n",
       "      <th>RV</th>\n",
       "      <th>TPA</th>\n",
       "      <th>CFL</th>\n",
       "      <th>GNO</th>\n",
       "      <th>VLD</th>\n",
       "      <th>ASP</th>\n",
       "      <th>ASS</th>\n",
       "      <th>Ref</th>\n",
       "      <th>U3</th>\n",
       "      <th>U5</th>\n",
       "      <th>TOPMED</th>\n",
       "      <th>WGT</th>\n",
       "      <th>MTP</th>\n",
       "      <th>LSD</th>\n",
       "      <th>NOC</th>\n",
       "      <th>DSS</th>\n",
       "      <th>SYN</th>\n",
       "      <th>KGPhase3</th>\n",
       "      <th>CAF</th>\n",
       "      <th>VC</th>\n",
       "      <th>MUT</th>\n",
       "      <th>KGPhase1</th>\n",
       "      <th>NOV</th>\n",
       "      <th>VP</th>\n",
       "      <th>SAO</th>\n",
       "      <th>INT</th>\n",
       "      <th>G5</th>\n",
       "      <th>OM</th>\n",
       "      <th>PMC</th>\n",
       "      <th>SSR</th>\n",
       "      <th>RSPOS</th>\n",
       "      <th>HD</th>\n",
       "      <th>PM</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>ClinVar_CLNSIG</th>\n",
       "      <th>Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>rs35881187</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>1/1</td>\n",
       "      <td>150</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>96.55%</td>\n",
       "      <td>9.9776E-16</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>3</td>\n",
       "      <td>ADP=29</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=126</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=35881187</td>\n",
       "      <td>RV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.55968081039755351,0.44031918960244648</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.5208,0.4792</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05012808000515013e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td>PMC</td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1043224</td>\n",
       "      <td></td>\n",
       "      <td>PM</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[CCT, C]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1299382</td>\n",
       "      <td>ACAP3</td>\n",
       "      <td>rs143128930</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>0/1</td>\n",
       "      <td>109</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>35</td>\n",
       "      <td>30</td>\n",
       "      <td>46.15%</td>\n",
       "      <td>1.1519E-11</td>\n",
       "      <td>27</td>\n",
       "      <td>8</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>ADP=65</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=134</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=143128930</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.91126720183486238,0.08873279816513761</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.8806,0.1194</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x0500000a000515003e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1299383</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[AG, A]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1353987</td>\n",
       "      <td>MXRA8</td>\n",
       "      <td>rs140777846</td>\n",
       "      <td>CTG</td>\n",
       "      <td>C</td>\n",
       "      <td>0/1</td>\n",
       "      <td>151</td>\n",
       "      <td>76</td>\n",
       "      <td>76</td>\n",
       "      <td>36</td>\n",
       "      <td>40</td>\n",
       "      <td>52.63%</td>\n",
       "      <td>7.6319E-16</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>34</td>\n",
       "      <td>6</td>\n",
       "      <td>ADP=76</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=134</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>G5A</td>\n",
       "      <td>COMMON=1</td>\n",
       "      <td>RS=140777846</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.13763857033639143,0.86236142966360856</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.1793,0.8207</td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050100080005170126000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1353988</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[CTG, C]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1398672</td>\n",
       "      <td>LOC148413,CCNL2,CCNL2</td>\n",
       "      <td>rs368050244;rs3831366</td>\n",
       "      <td>CTAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>0/1</td>\n",
       "      <td>131</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>26</td>\n",
       "      <td>34</td>\n",
       "      <td>56.67%</td>\n",
       "      <td>7.4725E-14</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>8</td>\n",
       "      <td>ADP=60</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=107,138</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>R5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=1,1</td>\n",
       "      <td>RS=3831366,368050244</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td>ASS</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.94513729612640163,0.05486270387359836</td>\n",
       "      <td>WGT=1,1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.9259,0.07408,0.9259,0.07408</td>\n",
       "      <td>VC=DIV,DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05012822000515013e000200,0x0500000a000500...</td>\n",
       "      <td>SAO=0,0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td>PMC</td>\n",
       "      <td>SSR=0,0</td>\n",
       "      <td>RSPOS=1398673,1398677</td>\n",
       "      <td></td>\n",
       "      <td>PM</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[CTAGAG, C]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1629572</td>\n",
       "      <td>MIB2</td>\n",
       "      <td>rs112177324</td>\n",
       "      <td>null</td>\n",
       "      <td>T</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>22</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>59.09%</td>\n",
       "      <td>9.5813E-6</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=21</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=132</td>\n",
       "      <td>SLO</td>\n",
       "      <td>NSF</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=112177324</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td>VLD</td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.75031058868501529,0.24968941131498470</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05010008120515013e000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=1629573</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[null, T]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2606</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153913005</td>\n",
       "      <td>ARHGAP4</td>\n",
       "      <td>rs5904376;rs77485258</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>2</td>\n",
       "      <td>61</td>\n",
       "      <td>96.83%</td>\n",
       "      <td>3.4466E-34</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>11</td>\n",
       "      <td>ADP=63</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=131,126</td>\n",
       "      <td>SLO</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>COMMON=0,0</td>\n",
       "      <td>RS=77485258,5904376</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>GNO</td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.01672400611620795,0.98327599388379204</td>\n",
       "      <td>WGT=1,1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>KGPhase3</td>\n",
       "      <td>CAF=0.0002649,0.9997,0.0002649,0.9997</td>\n",
       "      <td>VC=DIV,DIV</td>\n",
       "      <td></td>\n",
       "      <td>KGPhase1</td>\n",
       "      <td></td>\n",
       "      <td>VP=0x05000008000501003e000200,0x05010008000500...</td>\n",
       "      <td>SAO=0,0</td>\n",
       "      <td>INT</td>\n",
       "      <td>G5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0,0</td>\n",
       "      <td>RSPOS=153913005,153913008</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[G, GC]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2607</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155228363</td>\n",
       "      <td>VBP1</td>\n",
       "      <td>rs781912204</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>0/1</td>\n",
       "      <td>64</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>60%</td>\n",
       "      <td>3.8543E-7</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>ADP=25</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=144</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=781912204</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.43169438073394495,0.56314506880733944...</td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NOV</td>\n",
       "      <td>VP=0x050000080005000002000204</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=155228363</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[A, AT]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2608</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155492733</td>\n",
       "      <td>TMLHE,TMLHE-AS1,TMLHE-AS1</td>\n",
       "      <td>rs1169019545;rs376271737</td>\n",
       "      <td>T</td>\n",
       "      <td>TG</td>\n",
       "      <td>1/1</td>\n",
       "      <td>81</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>94.12%</td>\n",
       "      <td>7.7134E-9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>ADP=17</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=151,138</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=1169019545,376271737</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TOPMED=0.35031218144750254,0.64968781855249745</td>\n",
       "      <td>WGT=1,1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV,DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000080005000002000200,0x05000008000500...</td>\n",
       "      <td>SAO=0,0</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0,0</td>\n",
       "      <td>RSPOS=155492733,155492734</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[T, TG]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2609</th>\n",
       "      <td>chrY</td>\n",
       "      <td>12786501</td>\n",
       "      <td>USP9Y</td>\n",
       "      <td>rs760255651</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>0/1</td>\n",
       "      <td>50</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>57.14%</td>\n",
       "      <td>8.0605E-6</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=21</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=1</td>\n",
       "      <td>HOM=0</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=144</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=760255651</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000080005000002000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td>INT</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=12786503</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[CT, C]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2610</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21311242</td>\n",
       "      <td></td>\n",
       "      <td>rs202147921</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>1/1</td>\n",
       "      <td>71</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>7.3277E-8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>ADP=18</td>\n",
       "      <td>WT=0</td>\n",
       "      <td>HET=0</td>\n",
       "      <td>HOM=1</td>\n",
       "      <td>NC=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>dbSNPBuildID=137</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RS=202147921</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ASP</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>WGT=1</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VC=DIV</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>VP=0x050000000005000002000200</td>\n",
       "      <td>SAO=0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>SSR=0</td>\n",
       "      <td>RSPOS=21311242</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>[A, AT]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2611 rows × 73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM        POS                  Gene Name                      rsID  \\\n",
       "0     chr1    1043223                       AGRN                rs35881187   \n",
       "1     chr1    1299382                      ACAP3               rs143128930   \n",
       "2     chr1    1353987                      MXRA8               rs140777846   \n",
       "3     chr1    1398672      LOC148413,CCNL2,CCNL2     rs368050244;rs3831366   \n",
       "4     chr1    1629572                       MIB2               rs112177324   \n",
       "...    ...        ...                        ...                       ...   \n",
       "2606  chrX  153913005                    ARHGAP4      rs5904376;rs77485258   \n",
       "2607  chrX  155228363                       VBP1               rs781912204   \n",
       "2608  chrX  155492733  TMLHE,TMLHE-AS1,TMLHE-AS1  rs1169019545;rs376271737   \n",
       "2609  chrY   12786501                      USP9Y               rs760255651   \n",
       "2610  chrY   21311242                                          rs202147921   \n",
       "\n",
       "         REF ALT   GT   GQ SDP  DP  RD  AD    FREQ        PVAL RDF RDR ADF  \\\n",
       "0        CCT   C  1/1  150  29  29   1  28  96.55%  9.9776E-16   1   0  25   \n",
       "1         AG   A  0/1  109  65  65  35  30  46.15%  1.1519E-11  27   8  26   \n",
       "2        CTG   C  0/1  151  76  76  36  40  52.63%  7.6319E-16  28   8  34   \n",
       "3     CTAGAG   C  0/1  131  60  60  26  34  56.67%  7.4725E-14  21   5  26   \n",
       "4       null   T  0/1   50  22  21   9  13  59.09%   9.5813E-6   9   0  11   \n",
       "...      ...  ..  ...  ...  ..  ..  ..  ..     ...         ...  ..  ..  ..   \n",
       "2606       G  GC  1/1  255  63  63   2  61  96.83%  3.4466E-34   2   0  50   \n",
       "2607       A  AT  0/1   64  25  25   5  15     60%   3.8543E-7   5   0  14   \n",
       "2608       T  TG  1/1   81  17  17   1  16  94.12%   7.7134E-9   0   1  11   \n",
       "2609      CT   C  0/1   50  21  21   4  12  57.14%   8.0605E-6   3   1  10   \n",
       "2610       A  AT  1/1   71  18  18   2  15  83.33%   7.3277E-8   2   0  13   \n",
       "\n",
       "     ADR     ADP    WT    HET    HOM    NC CDA OTH S3D WTD  \\\n",
       "0      3  ADP=29  WT=0  HET=0  HOM=1  NC=0                   \n",
       "1      4  ADP=65  WT=0  HET=1  HOM=0  NC=0                   \n",
       "2      6  ADP=76  WT=0  HET=1  HOM=0  NC=0                   \n",
       "3      8  ADP=60  WT=0  HET=1  HOM=0  NC=0                   \n",
       "4      2  ADP=21  WT=0  HET=1  HOM=0  NC=0                   \n",
       "...   ..     ...   ...    ...    ...   ...  ..  ..  ..  ..   \n",
       "2606  11  ADP=63  WT=0  HET=0  HOM=1  NC=0                   \n",
       "2607   1  ADP=25  WT=0  HET=1  HOM=0  NC=0                   \n",
       "2608   5  ADP=17  WT=0  HET=0  HOM=1  NC=0                   \n",
       "2609   2  ADP=21  WT=0  HET=1  HOM=0  NC=0                   \n",
       "2610   2  ADP=18  WT=0  HET=0  HOM=1  NC=0                   \n",
       "\n",
       "              dbSNPBuildID  SLO  NSF R3  R5 NSN NSM  G5A      COMMON  \\\n",
       "0         dbSNPBuildID=126  SLO                             COMMON=1   \n",
       "1         dbSNPBuildID=134               R5                 COMMON=1   \n",
       "2         dbSNPBuildID=134  SLO                      G5A    COMMON=1   \n",
       "3     dbSNPBuildID=107,138  SLO          R5               COMMON=1,1   \n",
       "4         dbSNPBuildID=132  SLO  NSF                                   \n",
       "...                    ...  ...  ... ..  ..  ..  ..  ...         ...   \n",
       "2606  dbSNPBuildID=131,126  SLO                           COMMON=0,0   \n",
       "2607      dbSNPBuildID=144                                             \n",
       "2608  dbSNPBuildID=151,138                                             \n",
       "2609      dbSNPBuildID=144                                             \n",
       "2610      dbSNPBuildID=137                                             \n",
       "\n",
       "                           RS  RV TPA CFL  GNO  VLD  ASP  ASS Ref U3 U5  \\\n",
       "0                 RS=35881187  RV          GNO  VLD  ASP                  \n",
       "1                RS=143128930                   VLD  ASP                  \n",
       "2                RS=140777846              GNO  VLD  ASP                  \n",
       "3        RS=3831366,368050244              GNO  VLD  ASP  ASS             \n",
       "4                RS=112177324              GNO  VLD  ASP                  \n",
       "...                       ...  ..  ..  ..  ...  ...  ...  ...  .. .. ..   \n",
       "2606      RS=77485258,5904376              GNO       ASP                  \n",
       "2607             RS=781912204                        ASP                  \n",
       "2608  RS=1169019545,376271737                        ASP                  \n",
       "2609             RS=760255651                        ASP                  \n",
       "2610             RS=202147921                        ASP                  \n",
       "\n",
       "                                                 TOPMED      WGT MTP LSD NOC  \\\n",
       "0        TOPMED=0.55968081039755351,0.44031918960244648    WGT=1               \n",
       "1        TOPMED=0.91126720183486238,0.08873279816513761    WGT=1               \n",
       "2        TOPMED=0.13763857033639143,0.86236142966360856    WGT=1               \n",
       "3        TOPMED=0.94513729612640163,0.05486270387359836  WGT=1,1               \n",
       "4        TOPMED=0.75031058868501529,0.24968941131498470    WGT=1               \n",
       "...                                                 ...      ...  ..  ..  ..   \n",
       "2606     TOPMED=0.01672400611620795,0.98327599388379204  WGT=1,1               \n",
       "2607  TOPMED=0.43169438073394495,0.56314506880733944...    WGT=1               \n",
       "2608     TOPMED=0.35031218144750254,0.64968781855249745  WGT=1,1               \n",
       "2609                                                       WGT=1               \n",
       "2610                                                       WGT=1               \n",
       "\n",
       "     DSS SYN  KGPhase3                                    CAF          VC MUT  \\\n",
       "0             KGPhase3                      CAF=0.5208,0.4792      VC=DIV       \n",
       "1             KGPhase3                      CAF=0.8806,0.1194      VC=DIV       \n",
       "2             KGPhase3                      CAF=0.1793,0.8207      VC=DIV       \n",
       "3             KGPhase3      CAF=0.9259,0.07408,0.9259,0.07408  VC=DIV,DIV       \n",
       "4             KGPhase3                                             VC=DIV       \n",
       "...   ..  ..       ...                                    ...         ...  ..   \n",
       "2606          KGPhase3  CAF=0.0002649,0.9997,0.0002649,0.9997  VC=DIV,DIV       \n",
       "2607                                                               VC=DIV       \n",
       "2608                                                           VC=DIV,DIV       \n",
       "2609                                                               VC=DIV       \n",
       "2610                                                               VC=DIV       \n",
       "\n",
       "      KGPhase1  NOV                                                 VP  \\\n",
       "0     KGPhase1                           VP=0x05012808000515013e000200   \n",
       "1     KGPhase1                           VP=0x0500000a000515003e000200   \n",
       "2                                        VP=0x050100080005170126000200   \n",
       "3     KGPhase1       VP=0x05012822000515013e000200,0x0500000a000500...   \n",
       "4     KGPhase1                           VP=0x05010008120515013e000200   \n",
       "...        ...  ...                                                ...   \n",
       "2606  KGPhase1       VP=0x05000008000501003e000200,0x05010008000500...   \n",
       "2607            NOV                      VP=0x050000080005000002000204   \n",
       "2608                 VP=0x050000080005000002000200,0x05000008000500...   \n",
       "2609                                     VP=0x050000080005000002000200   \n",
       "2610                                     VP=0x050000000005000002000200   \n",
       "\n",
       "          SAO  INT  G5 OM  PMC      SSR                      RSPOS HD  PM  \\\n",
       "0       SAO=0  INT  G5     PMC    SSR=0              RSPOS=1043224     PM   \n",
       "1       SAO=0  INT  G5            SSR=0              RSPOS=1299383          \n",
       "2       SAO=0  INT  G5            SSR=0              RSPOS=1353988          \n",
       "3     SAO=0,0  INT  G5     PMC  SSR=0,0      RSPOS=1398673,1398677     PM   \n",
       "4       SAO=0  INT  G5            SSR=0              RSPOS=1629573          \n",
       "...       ...  ...  .. ..  ...      ...                        ... ..  ..   \n",
       "2606  SAO=0,0  INT  G5          SSR=0,0  RSPOS=153913005,153913008          \n",
       "2607    SAO=0  INT                SSR=0            RSPOS=155228363          \n",
       "2608  SAO=0,0  INT              SSR=0,0  RSPOS=155492733,155492734          \n",
       "2609    SAO=0  INT                SSR=0             RSPOS=12786503          \n",
       "2610    SAO=0                     SSR=0             RSPOS=21311242          \n",
       "\n",
       "     ClinVar ClinVar_CLNSIG       Allele  \n",
       "0                               [CCT, C]  \n",
       "1                                [AG, A]  \n",
       "2                               [CTG, C]  \n",
       "3                            [CTAGAG, C]  \n",
       "4                              [null, T]  \n",
       "...      ...            ...          ...  \n",
       "2606                             [G, GC]  \n",
       "2607                             [A, AT]  \n",
       "2608                             [T, TG]  \n",
       "2609                             [CT, C]  \n",
       "2610                             [A, AT]  \n",
       "\n",
       "[2611 rows x 73 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['Allele'] = '[' + data['REF'] + ', ' + data['ALT'] + ']'\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "5e385982",
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/InDel_columns_Extracted.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "35da363e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele</th>\n",
       "      <th>zygosity</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsid</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>associated_diseases</th>\n",
       "      <th>review_status</th>\n",
       "      <th>origin</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>variant_subtype</th>\n",
       "      <th>Phargkb_ann_exists</th>\n",
       "      <th>is_mutation</th>\n",
       "      <th>Variant_is_precious</th>\n",
       "      <th>Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>['ACCTT', A]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>ADAMTS17</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10549565</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Weill-Marchesani_4_syndrome', '_recessive|not_...</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>[ACCTT, A]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>['CACA', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>GIGYF2</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10555297</td>\n",
       "      <td>non-coding_transcript_variant', 'SO:0001822</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Parkinson_disease_11', '_autosomal_dominant', ...</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>[CACA, C]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>['GCA', G]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TRPV3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10573788</td>\n",
       "      <td>5_prime_UTR_variant', 'SO:0001624</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Olmsted_syndrome_1|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>[GCA, G]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>['CAT', C]</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>PTGES3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10579382</td>\n",
       "      <td>initiatior_codon_variant', 'SO:0001589</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>[CAT, C]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>['CAA', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>JAK3</td>\n",
       "      <td>No</td>\n",
       "      <td>rs10580414</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Severe_combined_immunodeficiency_disease</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>[CAA, C]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>664</th>\n",
       "      <td>['AAAGGAAGT', A]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ATR</td>\n",
       "      <td>No</td>\n",
       "      <td>rs797045404</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>[AAAGGAAGT, A]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>665</th>\n",
       "      <td>['TA', T]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>WDR62</td>\n",
       "      <td>No</td>\n",
       "      <td>rs797046108</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>Conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>not_specified|Primary_Microcephaly_2_With_or_W...</td>\n",
       "      <td>criteria_provided', '_conflicting_interpretations</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>[TA, T]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666</th>\n",
       "      <td>['CTT', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>ANTXR2</td>\n",
       "      <td>No</td>\n",
       "      <td>rs80314910</td>\n",
       "      <td>5_prime_UTR_variant', 'SO:0001627</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Hyaline_fibromatosis_syndrome|not_provided</td>\n",
       "      <td>criteria_provided', '_multiple_submitters', '_...</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>[CTT, C]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667</th>\n",
       "      <td>['CCTTT', C]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CNGB1</td>\n",
       "      <td>No</td>\n",
       "      <td>rs866017919</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>Retinitis_Pigmentosa', '_Recessive</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>[CCTTT, C]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>668</th>\n",
       "      <td>['C', CA]</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>CHMP2B</td>\n",
       "      <td>No</td>\n",
       "      <td>rs886058906</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Frontotemporal_dementia</td>\n",
       "      <td>criteria_provided', '_single_submitter</td>\n",
       "      <td>germline</td>\n",
       "      <td>indel</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>MayBe No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>[C, CA]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>669 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               allele      zygosity Gene Name Gene_Match         rsid  \\\n",
       "0        ['ACCTT', A]    Homozygous  ADAMTS17         No   rs10549565   \n",
       "1         ['CACA', C]  Heterozygous    GIGYF2         No   rs10555297   \n",
       "2          ['GCA', G]  Heterozygous     TRPV3         No   rs10573788   \n",
       "3          ['CAT', C]    Homozygous    PTGES3         No   rs10579382   \n",
       "4          ['CAA', C]  Heterozygous      JAK3         No   rs10580414   \n",
       "..                ...           ...       ...        ...          ...   \n",
       "664  ['AAAGGAAGT', A]  Heterozygous       ATR         No  rs797045404   \n",
       "665         ['TA', T]  Heterozygous     WDR62         No  rs797046108   \n",
       "666        ['CTT', C]  Heterozygous    ANTXR2         No   rs80314910   \n",
       "667      ['CCTTT', C]  Heterozygous     CNGB1         No  rs866017919   \n",
       "668         ['C', CA]  Heterozygous    CHMP2B         No  rs886058906   \n",
       "\n",
       "                                     consequence Consequence_score  \\\n",
       "0                            3_prime_UTR_variant              3/10   \n",
       "1    non-coding_transcript_variant', 'SO:0001822              2/10   \n",
       "2              5_prime_UTR_variant', 'SO:0001624              3/10   \n",
       "3         initiatior_codon_variant', 'SO:0001589               NaN   \n",
       "4                                            NaN               NaN   \n",
       "..                                           ...               ...   \n",
       "664                               intron_variant              2/10   \n",
       "665                               intron_variant              2/10   \n",
       "666            5_prime_UTR_variant', 'SO:0001627              3/10   \n",
       "667                          3_prime_UTR_variant              3/10   \n",
       "668                          3_prime_UTR_variant              3/10   \n",
       "\n",
       "                            clinical_significance  \\\n",
       "0                                          Benign   \n",
       "1                                          Benign   \n",
       "2                                          Benign   \n",
       "3                                          Benign   \n",
       "4                                          Benign   \n",
       "..                                            ...   \n",
       "664                        Uncertain_significance   \n",
       "665  Conflicting_interpretations_of_pathogenicity   \n",
       "666                                        Benign   \n",
       "667                                 Likely_benign   \n",
       "668                        Uncertain_significance   \n",
       "\n",
       "                                   associated_diseases  \\\n",
       "0    Weill-Marchesani_4_syndrome', '_recessive|not_...   \n",
       "1    Parkinson_disease_11', '_autosomal_dominant', ...   \n",
       "2                      Olmsted_syndrome_1|not_provided   \n",
       "3                                         not_provided   \n",
       "4             Severe_combined_immunodeficiency_disease   \n",
       "..                                                 ...   \n",
       "664                         not_specified|not_provided   \n",
       "665  not_specified|Primary_Microcephaly_2_With_or_W...   \n",
       "666         Hyaline_fibromatosis_syndrome|not_provided   \n",
       "667                 Retinitis_Pigmentosa', '_Recessive   \n",
       "668                            Frontotemporal_dementia   \n",
       "\n",
       "                                         review_status    origin variant_type  \\\n",
       "0    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "1               criteria_provided', '_single_submitter  germline        indel   \n",
       "2    criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "3                       no_assertion_criteria_provided  germline        indel   \n",
       "4               criteria_provided', '_single_submitter  germline        indel   \n",
       "..                                                 ...       ...          ...   \n",
       "664  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "665  criteria_provided', '_conflicting_interpretations  germline        indel   \n",
       "666  criteria_provided', '_multiple_submitters', '_...  germline        indel   \n",
       "667             criteria_provided', '_single_submitter  germline        indel   \n",
       "668             criteria_provided', '_single_submitter  germline        indel   \n",
       "\n",
       "    variant_subtype Phargkb_ann_exists is_mutation Variant_is_precious  \\\n",
       "0          Deletion           MayBe No          No                  No   \n",
       "1          Deletion           MayBe No          No                 Yes   \n",
       "2          Deletion           MayBe No          No                  No   \n",
       "3          Deletion           MayBe No          No                  No   \n",
       "4          Deletion           MayBe No          No                  No   \n",
       "..              ...                ...         ...                 ...   \n",
       "664        Deletion           MayBe No          No                 Yes   \n",
       "665        Deletion           MayBe No          No                 Yes   \n",
       "666        Deletion           MayBe No          No                 Yes   \n",
       "667        Deletion           MayBe No          No                  No   \n",
       "668       Insertion           MayBe No          No                 Yes   \n",
       "\n",
       "             Allele  \n",
       "0        [ACCTT, A]  \n",
       "1         [CACA, C]  \n",
       "2          [GCA, G]  \n",
       "3          [CAT, C]  \n",
       "4          [CAA, C]  \n",
       "..              ...  \n",
       "664  [AAAGGAAGT, A]  \n",
       "665         [TA, T]  \n",
       "666        [CTT, C]  \n",
       "667      [CCTTT, C]  \n",
       "668         [C, CA]  \n",
       "\n",
       "[669 rows x 17 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/KHCDPRGPTTL6_InDel.xlsx')\n",
    "df = df.rename({'zygocity':'zygosity'}, axis = 1)\n",
    "df['Allele'] = df['allele'].str.replace('\\'', '')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27e39b3a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
