{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "260b2f61",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import polars as pl\n",
    "import sys\n",
    "import re\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import plotly\n",
    "import plotly.express as px\n",
    "\n",
    "\n",
    "pd.set_option('display.max_columns',None)\n",
    "import psycopg2\n",
    "\n",
    "\n",
    "#to scale the data using z-score \n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "#Algorithms to use\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier\n",
    "\n",
    "#Metrics to evaluate the model\n",
    "from sklearn.metrics import confusion_matrix, classification_report, precision_recall_curve\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "#importing PCA and TSNE\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "efdb282d",
   "metadata": {},
   "source": [
    "# VCF file Covered Positions Extraction From the Bed File "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "ab06e34c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_bed_file(bed_file):\n",
    "    bed_positions = set()\n",
    "    with open(bed_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Skip header lines if present\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 3:\n",
    "                chrom = fields[0]\n",
    "                try:\n",
    "                    start = int(fields[1])\n",
    "                    end = int(fields[2])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if start or end position is not an integer\n",
    "                for pos in range(start, end + 1):\n",
    "                    bed_positions.add((chrom, pos))\n",
    "    return bed_positions\n",
    "\n",
    "def normalize_chrom_name(chrom):\n",
    "    return chrom.split('_')[0]\n",
    "\n",
    "def filter_vcf_file(vcf_file, bed_positions):\n",
    "    filtered_vcf_records = []\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        for line in f:\n",
    "            if line.startswith('#'):  # Preserve header lines in the output\n",
    "                filtered_vcf_records.append(line)\n",
    "                continue\n",
    "            fields = line.strip().split('\\t')\n",
    "            if len(fields) >= 2:\n",
    "                raw_chrom = fields[0]\n",
    "                chrom = normalize_chrom_name(raw_chrom)\n",
    "                try:\n",
    "                    pos = int(fields[1])\n",
    "                except ValueError:\n",
    "                    continue  # Skip this line if 'POS' is not an integer\n",
    "                if (chrom, pos) in bed_positions:\n",
    "                    filtered_vcf_records.append(line)\n",
    "    return filtered_vcf_records\n",
    "\n",
    "def write_filtered_vcf(filtered_vcf_records, output_file):\n",
    "    with open(output_file, 'w') as f:\n",
    "        for record in filtered_vcf_records:\n",
    "            f.write(record)\n",
    "\n",
    "def main():\n",
    "    bed_file = r'C:/Users/GenepoweRx_Madhu/Desktop/Covered_regions.bed'\n",
    "    vcf_file = r'C:/Users/GenepoweRx_Madhu/Downloads/vcf_files_all/KHAIGHGPTTL187/KHAIGHGPTTL187_final.vcf'\n",
    "    output_file = r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHAIGHGPTTL187_final.vcf'\n",
    "\n",
    "    bed_positions = read_bed_file(bed_file)\n",
    "    filtered_vcf_records = filter_vcf_file(vcf_file, bed_positions)\n",
    "    write_filtered_vcf(filtered_vcf_records, output_file)\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    main()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "c688d818",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65529</td>\n",
       "      <td>rs1473003496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>31.82%</td>\n",
       "      <td>4.4505E-3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=102;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>132</td>\n",
       "      <td>102</td>\n",
       "      <td>102</td>\n",
       "      <td>64</td>\n",
       "      <td>38</td>\n",
       "      <td>37.25%</td>\n",
       "      <td>5.1448E-14</td>\n",
       "      <td>41</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=160;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>160</td>\n",
       "      <td>160</td>\n",
       "      <td>0</td>\n",
       "      <td>160</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.0505E-95</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69897</td>\n",
       "      <td>rs200676709</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=86;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....</td>\n",
       "      <td>0/1</td>\n",
       "      <td>165</td>\n",
       "      <td>86</td>\n",
       "      <td>86</td>\n",
       "      <td>42</td>\n",
       "      <td>44</td>\n",
       "      <td>51.16%</td>\n",
       "      <td>3.1151E-17</td>\n",
       "      <td>27</td>\n",
       "      <td>15</td>\n",
       "      <td>34</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=33;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>188</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>33</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.3852E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34676</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23179733</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=89;WT=0;HET=1;HOM=0;NC=0;CSQ=C|synonymous_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>91</td>\n",
       "      <td>89</td>\n",
       "      <td>89</td>\n",
       "      <td>62</td>\n",
       "      <td>27</td>\n",
       "      <td>30.34%</td>\n",
       "      <td>7.2581E-10</td>\n",
       "      <td>44</td>\n",
       "      <td>18</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34677</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23190581</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=67;WT=0;HET=1;HOM=0;NC=0;CSQ=C|synonymous_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>48</td>\n",
       "      <td>67</td>\n",
       "      <td>67</td>\n",
       "      <td>52</td>\n",
       "      <td>15</td>\n",
       "      <td>22.39%</td>\n",
       "      <td>1.2648E-5</td>\n",
       "      <td>31</td>\n",
       "      <td>21</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34678</th>\n",
       "      <td>chrY</td>\n",
       "      <td>23227961</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=G|synonymous_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>77</td>\n",
       "      <td>75</td>\n",
       "      <td>75</td>\n",
       "      <td>52</td>\n",
       "      <td>23</td>\n",
       "      <td>30.67%</td>\n",
       "      <td>1.6219E-8</td>\n",
       "      <td>41</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34679</th>\n",
       "      <td>chrY</td>\n",
       "      <td>24842348</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=65;WT=0;HET=1;HOM=0;NC=0;CSQ=G|synonymous_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>60</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>47</td>\n",
       "      <td>18</td>\n",
       "      <td>27.69%</td>\n",
       "      <td>9.7371E-7</td>\n",
       "      <td>31</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34680</th>\n",
       "      <td>chrY</td>\n",
       "      <td>24853196</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=1;HOM=0;NC=0;CSQ=G|synonymous_...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>53</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>29.63%</td>\n",
       "      <td>4.1444E-6</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>34681 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     65529  rs1473003496   C   T    .   PASS   \n",
       "1      chr1     69270   rs201219564   A   G    .   PASS   \n",
       "2      chr1     69511     rs2691305   A   G    .   PASS   \n",
       "3      chr1     69897   rs200676709   T   C    .   PASS   \n",
       "4      chr1    924533   rs112703963   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "34676  chrY  23179733             .   T   C    .   PASS   \n",
       "34677  chrY  23190581             .   T   C    .   PASS   \n",
       "34678  chrY  23227961             .   A   G    .   PASS   \n",
       "34679  chrY  24842348             .   A   G    .   PASS   \n",
       "34680  chrY  24853196             .   A   G    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=22;WT=0;HET=1;HOM=0;NC=0;ASP;RS=1473003496...  0/1   23   22   22   \n",
       "1      ADP=102;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  132  102  102   \n",
       "2      ADP=160;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  160  160   \n",
       "3      ADP=86;WT=0;HET=1;HOM=0;NC=0;ASP;CAF=0.3119,0....  0/1  165   86   86   \n",
       "4      ADP=33;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  1/1  188   33   33   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "34676  ADP=89;WT=0;HET=1;HOM=0;NC=0;CSQ=C|synonymous_...  0/1   91   89   89   \n",
       "34677  ADP=67;WT=0;HET=1;HOM=0;NC=0;CSQ=C|synonymous_...  0/1   48   67   67   \n",
       "34678  ADP=75;WT=0;HET=1;HOM=0;NC=0;CSQ=G|synonymous_...  0/1   77   75   75   \n",
       "34679  ADP=65;WT=0;HET=1;HOM=0;NC=0;CSQ=G|synonymous_...  0/1   60   65   65   \n",
       "34680  ADP=54;WT=0;HET=1;HOM=0;NC=0;CSQ=G|synonymous_...  0/1   53   55   54   \n",
       "\n",
       "       RD   AD    FREQ        PVAL RDF RDR  ADF ADR  \n",
       "0      15    7  31.82%   4.4505E-3  14   1    6   1  \n",
       "1      64   38  37.25%  5.1448E-14  41  23   28  10  \n",
       "2       0  160    100%  1.0505E-95   0   0  116  44  \n",
       "3      42   44  51.16%  3.1151E-17  27  15   34  10  \n",
       "4       0   33    100%  1.3852E-19   0   0   20  13  \n",
       "...    ..  ...     ...         ...  ..  ..  ...  ..  \n",
       "34676  62   27  30.34%  7.2581E-10  44  18   23   4  \n",
       "34677  52   15  22.39%   1.2648E-5  31  21   10   5  \n",
       "34678  52   23  30.67%   1.6219E-8  41  11   21   2  \n",
       "34679  47   18  27.69%   9.7371E-7  31  16   14   4  \n",
       "34680  38   16  29.63%   4.1444E-6  28  10   14   2  \n",
       "\n",
       "[34681 rows x 20 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf = pd.read_csv(r'C:/Users/GenepoweRx_Madhu/Downloads/COVERED_VCF_FILES_BED/KHAIGHGPTTL187_final.vcf', comment= '#', sep = '\\t', header=None, low_memory=False)\n",
    "vcf.columns = ['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE']\n",
    "\n",
    "sample_cols = vcf['SAMPLE'].str.split(':', expand=True)\n",
    "sample_cols.columns = ['GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RBQ', 'ABQ', 'RDF', 'RDR', 'ADF', 'ADR']\n",
    "\n",
    "# Assign the values to the newly created columns\n",
    "vcf = pd.concat([vcf, sample_cols], axis=1)\n",
    "vcf = vcf[['CHROM', 'POS', 'rsID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL','RDF', 'RDR', 'ADF', 'ADR']]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "90d1a40c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686266</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>54</td>\n",
       "      <td>25</td>\n",
       "      <td>31.65%</td>\n",
       "      <td>3.1101E-9</td>\n",
       "      <td>43</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36647</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9528488</td>\n",
       "      <td>rs1179775265</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>130</td>\n",
       "      <td>186</td>\n",
       "      <td>186</td>\n",
       "      <td>146</td>\n",
       "      <td>40</td>\n",
       "      <td>21.51%</td>\n",
       "      <td>8.6616E-14</td>\n",
       "      <td>77</td>\n",
       "      <td>69</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36648</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13360045</td>\n",
       "      <td>rs2032674</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>54</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.0229E-32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36649</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14941891</td>\n",
       "      <td>rs17269816</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>182</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4567E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36650</th>\n",
       "      <td>chrY</td>\n",
       "      <td>15174113</td>\n",
       "      <td>rs17307398</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>164</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.3259E-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36652 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270   rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511     rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290             .   A   C    .   PASS   \n",
       "3      chr1    686266             .   A   C    .   PASS   \n",
       "4      chr1    924533   rs112703963   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "36647  chrY   9528488  rs1179775265   T   G    .   PASS   \n",
       "36648  chrY  13360045     rs2032674   T   C    .   PASS   \n",
       "36649  chrY  14941891    rs17269816   C   T    .   PASS   \n",
       "36650  chrY  15174113    rs17307398   T   C    .   PASS   \n",
       "36651  chrY  20577481     rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "3      ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   85   79   79   \n",
       "4      ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  1/1   58   11   11   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36647  ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  130  186  186   \n",
       "36648  ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....  1/1  255   54   54   \n",
       "36649  ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...  1/1  182   32   32   \n",
       "36650  ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...  1/1  164   29   29   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous  \n",
       "1        0  114    100%   4.392E-68   0   0  98  16   0   1    Homozygous  \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous  \n",
       "3       54   25  31.65%   3.1101E-9  43  11  23   2   1   0  Heterozygous  \n",
       "4        0   11    100%   1.4176E-6   0   0  10   1   0   1    Homozygous  \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...  \n",
       "36647  146   40  21.51%  8.6616E-14  77  69  17  23   1   0  Heterozygous  \n",
       "36648    0   54    100%  4.0229E-32   0   0  35  19   0   1    Homozygous  \n",
       "36649    0   32    100%  5.4567E-19   0   0  28   4   0   1    Homozygous  \n",
       "36650    0   29    100%  3.3259E-17   0   0  24   5   0   1    Homozygous  \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous  \n",
       "\n",
       "[36652 rows x 23 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['HET'] = vcf['INFO'].str.extract(r'HET=(\\d)')\n",
    "vcf['HOM'] = vcf['INFO'].str.extract(r'HOM=(\\d)')\n",
    "\n",
    "# Create a new column 'Zygosity' based on conditions\n",
    "vcf['Zygosity'] = ''\n",
    "\n",
    "vcf.loc[vcf['HOM'] == '1', 'Zygosity'] = 'Homozygous'\n",
    "vcf.loc[vcf['HET'] == '1', 'Zygosity'] = 'Heterozygous'\n",
    "vcf['GT'] = vcf['GT'].astype(str)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3495a079",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686266</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>85</td>\n",
       "      <td>79</td>\n",
       "      <td>79</td>\n",
       "      <td>54</td>\n",
       "      <td>25</td>\n",
       "      <td>31.65%</td>\n",
       "      <td>3.1101E-9</td>\n",
       "      <td>43</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924533</td>\n",
       "      <td>rs112703963</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>58</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.4176E-6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>LOC107985728:107985728|SAMD11:148398</td>\n",
       "      <td>LOC107985728,SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36647</th>\n",
       "      <td>chrY</td>\n",
       "      <td>9528488</td>\n",
       "      <td>rs1179775265</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>130</td>\n",
       "      <td>186</td>\n",
       "      <td>186</td>\n",
       "      <td>146</td>\n",
       "      <td>40</td>\n",
       "      <td>21.51%</td>\n",
       "      <td>8.6616E-14</td>\n",
       "      <td>77</td>\n",
       "      <td>69</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>TSPY10:100289087</td>\n",
       "      <td>TSPY10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36648</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13360045</td>\n",
       "      <td>rs2032674</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>54</td>\n",
       "      <td>54</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.0229E-32</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>UTY:7404</td>\n",
       "      <td>UTY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36649</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14941891</td>\n",
       "      <td>rs17269816</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>182</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.4567E-19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36650</th>\n",
       "      <td>chrY</td>\n",
       "      <td>15174113</td>\n",
       "      <td>rs17307398</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>164</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>29</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.3259E-17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36652 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS          rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270   rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511     rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290             .   A   C    .   PASS   \n",
       "3      chr1    686266             .   A   C    .   PASS   \n",
       "4      chr1    924533   rs112703963   A   G    .   PASS   \n",
       "...     ...       ...           ...  ..  ..  ...    ...   \n",
       "36647  chrY   9528488  rs1179775265   T   G    .   PASS   \n",
       "36648  chrY  13360045     rs2032674   T   C    .   PASS   \n",
       "36649  chrY  14941891    rs17269816   C   T    .   PASS   \n",
       "36650  chrY  15174113    rs17307398   T   C    .   PASS   \n",
       "36651  chrY  20577481     rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "3      ADP=79;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   85   79   79   \n",
       "4      ADP=11;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.2502,0....  1/1   58   11   11   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36647  ADP=186;WT=0;HET=1;HOM=0;NC=0;ASP;GENEINFO=TSP...  0/1  130  186  186   \n",
       "36648  ADP=54;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.8094,0....  1/1  255   54   54   \n",
       "36649  ADP=32;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.91,0.09...  1/1  182   32   32   \n",
       "36650  ADP=29;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.266,0.7...  1/1  164   29   29   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "1        0  114    100%   4.392E-68   0   0  98  16   0   1    Homozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "3       54   25  31.65%   3.1101E-9  43  11  23   2   1   0  Heterozygous   \n",
       "4        0   11    100%   1.4176E-6   0   0  10   1   0   1    Homozygous   \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "36647  146   40  21.51%  8.6616E-14  77  69  17  23   1   0  Heterozygous   \n",
       "36648    0   54    100%  4.0229E-32   0   0  35  19   0   1    Homozygous   \n",
       "36649    0   32    100%  5.4567E-19   0   0  28   4   0   1    Homozygous   \n",
       "36650    0   29    100%  3.3259E-17   0   0  24   5   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "\n",
       "                                  Gene_Name            Gene Name  \n",
       "0                               OR4F5:79501                OR4F5  \n",
       "1                               OR4F5:79501                OR4F5  \n",
       "2                                       NaN                       \n",
       "3                                       NaN                       \n",
       "4      LOC107985728:107985728|SAMD11:148398  LOC107985728,SAMD11  \n",
       "...                                     ...                  ...  \n",
       "36647                      TSPY10:100289087               TSPY10  \n",
       "36648                              UTY:7404                  UTY  \n",
       "36649                                   NaN                       \n",
       "36650                                   NaN                       \n",
       "36651                           EIF1AY:9086               EIF1AY  \n",
       "\n",
       "[36652 rows x 25 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[\"Gene_Name\"] = vcf[\"INFO\"].str.extract('GENEINFO=(?P<GENEINFO>.+?);')\n",
    "vcf['Gene Name'] = vcf['Gene_Name'].apply(lambda x: ','.join([segment.split(':')[0] for segment in x.split('|')]) if pd.notnull(x) else '')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "32c6bd73",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "1        0  114    100%   4.392E-68   0   0  98  16   0   1    Homozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "\n",
       "         Gene_Name Gene Name  \\\n",
       "0      OR4F5:79501     OR4F5   \n",
       "0      OR4F5:79501     OR4F5   \n",
       "1      OR4F5:79501     OR4F5   \n",
       "2              NaN             \n",
       "2              NaN             \n",
       "...            ...       ...   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "...                                                  ...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "\n",
       "                                                     csq  \n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...  \n",
       "0      G|regulatory_region_variant|MODIFIER|||Regulat...  \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...  \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...  \n",
       "2      C|intron_variant&non_coding_transcript_variant...  \n",
       "...                                                  ...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...  \n",
       "\n",
       "[328334 rows x 27 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['CSQ'] = vcf['INFO'].str.extract(r'CSQ=(.*)')\n",
    "vcf['csq'] = vcf['CSQ'].str.split(',')\n",
    "vcf = vcf.explode('csq')\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "622ce58b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2:p.Ser81%3D</td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>81</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2:p.Thr162Ala</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4:c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1:p.Leu130Arg</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>130</td>\n",
       "      <td>L/R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0.01)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5:n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 45 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "1        0  114    100%   4.392E-68   0   0  98  16   0   1    Homozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "\n",
       "         Gene_Name Gene Name  \\\n",
       "0      OR4F5:79501     OR4F5   \n",
       "0      OR4F5:79501     OR4F5   \n",
       "1      OR4F5:79501     OR4F5   \n",
       "2              NaN             \n",
       "2              NaN             \n",
       "...            ...       ...   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "...                                                  ...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...                 \n",
       "0      G|regulatory_region_variant|MODIFIER|||Regulat...                 \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...                 \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...                 \n",
       "2      C|intron_variant&non_coding_transcript_variant...                 \n",
       "...                                                  ...           ...   \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar                             HGVSC  \\\n",
       "0                                                ENST00000641515.2:c.243A>G   \n",
       "0                                                                             \n",
       "1                                                ENST00000641515.2:c.484A>G   \n",
       "2                                                ENST00000426406.4:c.389T>G   \n",
       "2                                          ENST00000455207.5:n.169+33750T>G   \n",
       "...        ...                ...     ...                               ...   \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "\n",
       "                               HGVSP       PolyPhen          BIOTYPE EXON  \\\n",
       "0       ENSP00000493376.2:p.Ser81%3D                  protein_coding  3/3   \n",
       "0                                                    TF_binding_site        \n",
       "1      ENSP00000493376.2:p.Thr162Ala      benign(0)   protein_coding  3/3   \n",
       "2      ENSP00000409316.1:p.Leu130Arg  benign(0.336)   protein_coding  1/1   \n",
       "2                                                             lncRNA        \n",
       "...                              ...            ...              ...  ...   \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "\n",
       "      INTRON Protein_position Amino_acids   Codons STRAND  \\\n",
       "0                          81           S  tcA/tcG      1   \n",
       "0                                                           \n",
       "1                         162         T/A  Aca/Gca      1   \n",
       "2                         130         L/R  cTc/cGc     -1   \n",
       "2        1/2                                           -1   \n",
       "...      ...              ...         ...      ...    ...   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "\n",
       "                                                  PUBMED  \\\n",
       "0                                                          \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "2                                                          \n",
       "...                                                  ...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "\n",
       "                                        Consequence    IMPACT  \\\n",
       "0                                synonymous_variant       LOW   \n",
       "0                         regulatory_region_variant  MODIFIER   \n",
       "1                                  missense_variant  MODERATE   \n",
       "2                                  missense_variant  MODERATE   \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "...                                             ...       ...   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "\n",
       "                                   SIFT  \n",
       "0                                        \n",
       "0                                        \n",
       "1                       tolerated(0.92)  \n",
       "2      deleterious_low_confidence(0.01)  \n",
       "2                                        \n",
       "...                                 ...  \n",
       "36651                                    \n",
       "36651                                    \n",
       "36651                                    \n",
       "36651                                    \n",
       "36651                                    \n",
       "\n",
       "[328334 rows x 45 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['ClinVar_CLNDN'] = vcf['csq'].str.split('|').str[82]\n",
    "vcf['CLIN_SIG'] = vcf['csq'].str.split('|').str[70]\n",
    "vcf['ClinVar_CLNREVSTAT'] = vcf['csq'].str.split('|').str[81]\n",
    "vcf['ClinVar'] = vcf['csq'].str.split('|').str[79]\n",
    "vcf['HGVSC'] = vcf['csq'].str.split('|').str[10]\n",
    "vcf['HGVSP'] = vcf['csq'].str.split('|').str[11]\n",
    "vcf['PolyPhen'] = vcf['csq'].str.split('|').str[38]\n",
    "vcf['BIOTYPE'] = vcf['csq'].str.split('|').str[7]\n",
    "vcf['EXON'] = vcf['csq'].str.split('|').str[8]\n",
    "vcf['INTRON'] = vcf['csq'].str.split('|').str[9]\n",
    "vcf['Protein_position'] = vcf['csq'].str.split('|').str[14]\n",
    "vcf['Amino_acids'] = vcf['csq'].str.split('|').str[15]\n",
    "vcf['Codons'] = vcf['csq'].str.split('|').str[16]\n",
    "vcf['STRAND'] = vcf['csq'].str.split('|').str[19]\n",
    "vcf['PUBMED'] = vcf['csq'].str.split('|').str[73]\n",
    "vcf['Consequence'] = vcf['csq'].str.split('|').str[1]\n",
    "vcf['IMPACT'] = vcf['csq'].str.split('|').str[2]\n",
    "vcf['SIFT'] = vcf['csq'].str.split('|').str[37]\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "a8def5f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2:p.Ser81%3D</td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>81</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2:p.Thr162Ala</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>T162A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4:c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1:p.Leu130Arg</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>130</td>\n",
       "      <td>L/R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0.01)</td>\n",
       "      <td>L130R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5:n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "1        0  114    100%   4.392E-68   0   0  98  16   0   1    Homozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "\n",
       "         Gene_Name Gene Name  \\\n",
       "0      OR4F5:79501     OR4F5   \n",
       "0      OR4F5:79501     OR4F5   \n",
       "1      OR4F5:79501     OR4F5   \n",
       "2              NaN             \n",
       "2              NaN             \n",
       "...            ...       ...   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "...                                                  ...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...                 \n",
       "0      G|regulatory_region_variant|MODIFIER|||Regulat...                 \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...                 \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...                 \n",
       "2      C|intron_variant&non_coding_transcript_variant...                 \n",
       "...                                                  ...           ...   \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar                             HGVSC  \\\n",
       "0                                                ENST00000641515.2:c.243A>G   \n",
       "0                                                                             \n",
       "1                                                ENST00000641515.2:c.484A>G   \n",
       "2                                                ENST00000426406.4:c.389T>G   \n",
       "2                                          ENST00000455207.5:n.169+33750T>G   \n",
       "...        ...                ...     ...                               ...   \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "\n",
       "                               HGVSP       PolyPhen          BIOTYPE EXON  \\\n",
       "0       ENSP00000493376.2:p.Ser81%3D                  protein_coding  3/3   \n",
       "0                                                    TF_binding_site        \n",
       "1      ENSP00000493376.2:p.Thr162Ala      benign(0)   protein_coding  3/3   \n",
       "2      ENSP00000409316.1:p.Leu130Arg  benign(0.336)   protein_coding  1/1   \n",
       "2                                                             lncRNA        \n",
       "...                              ...            ...              ...  ...   \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "\n",
       "      INTRON Protein_position Amino_acids   Codons STRAND  \\\n",
       "0                          81           S  tcA/tcG      1   \n",
       "0                                                           \n",
       "1                         162         T/A  Aca/Gca      1   \n",
       "2                         130         L/R  cTc/cGc     -1   \n",
       "2        1/2                                           -1   \n",
       "...      ...              ...         ...      ...    ...   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "\n",
       "                                                  PUBMED  \\\n",
       "0                                                          \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "2                                                          \n",
       "...                                                  ...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "\n",
       "                                        Consequence    IMPACT  \\\n",
       "0                                synonymous_variant       LOW   \n",
       "0                         regulatory_region_variant  MODIFIER   \n",
       "1                                  missense_variant  MODERATE   \n",
       "2                                  missense_variant  MODERATE   \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "...                                             ...       ...   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "\n",
       "                                   SIFT Protein Position and Amino Acid  \n",
       "0                                                                   S81  \n",
       "0                                                                   NaN  \n",
       "1                       tolerated(0.92)                           T162A  \n",
       "2      deleterious_low_confidence(0.01)                           L130R  \n",
       "2                                                                   NaN  \n",
       "...                                 ...                             ...  \n",
       "36651                                                               NaN  \n",
       "36651                                                               NaN  \n",
       "36651                                                               NaN  \n",
       "36651                                                               NaN  \n",
       "36651                                                               NaN  \n",
       "\n",
       "[328334 rows x 46 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf['Protein Position and Amino Acid'] = vcf['Amino_acids'].str[0] + vcf['Protein_position'] + np.where(vcf['Amino_acids'].str[-1] == vcf['Amino_acids'].str[0], '', vcf['Amino_acids'].str[-1])\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "b800651c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>rsID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Gene_Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>CSQ</th>\n",
       "      <th>csq</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSC</th>\n",
       "      <th>HGVSP</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2:p.Ser81%3D</td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>81</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|synonymous_variant|LOW|OR4F5|ENSG00000186092...</td>\n",
       "      <td>G|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td>G|missense_variant|MODERATE|OR4F5|ENSG00000186...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2:c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2:p.Thr162Ala</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>162</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>T162A</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4:c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1:p.Leu130Arg</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>130</td>\n",
       "      <td>L/R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>deleterious_low_confidence(0.01)</td>\n",
       "      <td>L130R</td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>C|missense_variant|MODERATE|OR4F29|ENSG0000028...</td>\n",
       "      <td>C|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5:n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>EIF1AY:9086</td>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS         rsID REF ALT QUAL FILTER  \\\n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "0      chr1     69270  rs201219564   A   G    .   PASS   \n",
       "1      chr1     69511    rs2691305   A   G    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "2      chr1    451290            .   A   C    .   PASS   \n",
       "...     ...       ...          ...  ..  ..  ...    ...   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "36651  chrY  20577481    rs9786153   C   T    .   PASS   \n",
       "\n",
       "                                                    INFO   GT   GQ  SDP   DP  \\\n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "0      ADP=138;WT=0;HET=1;HOM=0;NC=0;ASP;G5;G5A;GENEI...  0/1  127  138  138   \n",
       "1      ADP=114;WT=0;HET=0;HOM=1;NC=0;ASP;G5;GENEINFO=...  1/1  255  114  114   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "2      ADP=64;WT=0;HET=1;HOM=0;NC=0;CSQ=C|missense_va...  0/1   56   64   64   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "36651  ADP=22;WT=0;HET=0;HOM=1;NC=0;ASP;CAF=0.1727,0....  1/1  123   22   22   \n",
       "\n",
       "        RD   AD    FREQ        PVAL RDF RDR ADF ADR HET HOM      Zygosity  \\\n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "0      100   38  27.54%  1.8855E-13  86  14  29   9   1   0  Heterozygous   \n",
       "1        0  114    100%   4.392E-68   0   0  98  16   0   1    Homozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "2       47   17  26.56%    2.243E-6  33  14  15   2   1   0  Heterozygous   \n",
       "...    ...  ...     ...         ...  ..  ..  ..  ..  ..  ..           ...   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "36651    0   22    100%  4.7526E-13   0   0  14   8   0   1    Homozygous   \n",
       "\n",
       "         Gene_Name Gene Name  \\\n",
       "0      OR4F5:79501     OR4F5   \n",
       "0      OR4F5:79501     OR4F5   \n",
       "1      OR4F5:79501     OR4F5   \n",
       "2              NaN             \n",
       "2              NaN             \n",
       "...            ...       ...   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "36651  EIF1AY:9086    EIF1AY   \n",
       "\n",
       "                                                     CSQ  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...   \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...   \n",
       "...                                                  ...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "36651  T|intron_variant|MODIFIER|EIF1AY|ENSG000001986...   \n",
       "\n",
       "                                                     csq ClinVar_CLNDN  \\\n",
       "0      G|synonymous_variant|LOW|OR4F5|ENSG00000186092...                 \n",
       "0      G|regulatory_region_variant|MODIFIER|||Regulat...                 \n",
       "1      G|missense_variant|MODERATE|OR4F5|ENSG00000186...                 \n",
       "2      C|missense_variant|MODERATE|OR4F29|ENSG0000028...                 \n",
       "2      C|intron_variant&non_coding_transcript_variant...                 \n",
       "...                                                  ...           ...   \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "36651  T|upstream_gene_variant|MODIFIER|TTTY10|ENSG00...                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar                             HGVSC  \\\n",
       "0                                                ENST00000641515.2:c.243A>G   \n",
       "0                                                                             \n",
       "1                                                ENST00000641515.2:c.484A>G   \n",
       "2                                                ENST00000426406.4:c.389T>G   \n",
       "2                                          ENST00000455207.5:n.169+33750T>G   \n",
       "...        ...                ...     ...                               ...   \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "36651                                                                         \n",
       "\n",
       "                               HGVSP       PolyPhen          BIOTYPE EXON  \\\n",
       "0       ENSP00000493376.2:p.Ser81%3D                  protein_coding  3/3   \n",
       "0                                                    TF_binding_site        \n",
       "1      ENSP00000493376.2:p.Thr162Ala      benign(0)   protein_coding  3/3   \n",
       "2      ENSP00000409316.1:p.Leu130Arg  benign(0.336)   protein_coding  1/1   \n",
       "2                                                             lncRNA        \n",
       "...                              ...            ...              ...  ...   \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "36651                                                         lncRNA        \n",
       "\n",
       "      INTRON Protein_position Amino_acids   Codons STRAND  \\\n",
       "0                          81           S  tcA/tcG      1   \n",
       "0                                                           \n",
       "1                         162         T/A  Aca/Gca      1   \n",
       "2                         130         L/R  cTc/cGc     -1   \n",
       "2        1/2                                           -1   \n",
       "...      ...              ...         ...      ...    ...   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "36651                                                  -1   \n",
       "\n",
       "                                                  PUBMED  \\\n",
       "0                                                          \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "2                                                          \n",
       "...                                                  ...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...   \n",
       "\n",
       "                                        Consequence    IMPACT  \\\n",
       "0                                synonymous_variant       LOW   \n",
       "0                         regulatory_region_variant  MODIFIER   \n",
       "1                                  missense_variant  MODERATE   \n",
       "2                                  missense_variant  MODERATE   \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER   \n",
       "...                                             ...       ...   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "36651                         upstream_gene_variant  MODIFIER   \n",
       "\n",
       "                                   SIFT Protein Position and Amino Acid  \\\n",
       "0                                                                   S81   \n",
       "0                                                                   NaN   \n",
       "1                       tolerated(0.92)                           T162A   \n",
       "2      deleterious_low_confidence(0.01)                           L130R   \n",
       "2                                                                   NaN   \n",
       "...                                 ...                             ...   \n",
       "36651                                                               NaN   \n",
       "36651                                                               NaN   \n",
       "36651                                                               NaN   \n",
       "36651                                                               NaN   \n",
       "36651                                                               NaN   \n",
       "\n",
       "                   HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0      ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "0                                      None                      \n",
       "1      ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "2      ENST00000426406.4           c.389T>G  ENSP00000409316.1   \n",
       "2      ENST00000455207.5     n.169+33750T>G                      \n",
       "...                  ...                ...                ...   \n",
       "36651                                  None                      \n",
       "36651                                  None                      \n",
       "36651                                  None                      \n",
       "36651                                  None                      \n",
       "36651                                  None                      \n",
       "\n",
       "      HGVSp (Transcript)  \n",
       "0             p.Ser81%3D  \n",
       "0                   None  \n",
       "1            p.Thr162Ala  \n",
       "2            p.Leu130Arg  \n",
       "2                   None  \n",
       "...                  ...  \n",
       "36651               None  \n",
       "36651               None  \n",
       "36651               None  \n",
       "36651               None  \n",
       "36651               None  \n",
       "\n",
       "[328334 rows x 50 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf[['HGVSc', 'HGVSc (Transcript)']] = vcf['HGVSC'].str.split(':', 1, expand=True)\n",
    "vcf[['HGVSp', 'HGVSp (Transcript)']] = vcf['HGVSP'].str.split(':', 1, expand=True)\n",
    "vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "ec49ebf4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious_low_confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "1         OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "...         ...          ...   ...       ...  ..  ..           ...   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "\n",
       "                                        Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                synonymous_variant       LOW                 \n",
       "0                         regulatory_region_variant  MODIFIER                 \n",
       "1                                  missense_variant  MODERATE                 \n",
       "2                                  missense_variant  MODERATE                 \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER                 \n",
       "...                                             ...       ...           ...   \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                          ENST00000641515.2   \n",
       "0                                                              \n",
       "1                                          ENST00000641515.2   \n",
       "2                                          ENST00000426406.4   \n",
       "2                                          ENST00000455207.5   \n",
       "...        ...                ...     ...                ...   \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "\n",
       "      HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  SDP  \\\n",
       "0               c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  127  138   \n",
       "0                   None                                  None  0/1  127  138   \n",
       "1               c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255  114   \n",
       "2               c.389T>G  ENSP00000409316.1        p.Leu130Arg  0/1   56   64   \n",
       "2         n.169+33750T>G                                  None  0/1   56   64   \n",
       "...                  ...                ...                ...  ...  ...  ...   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "\n",
       "        DP   RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "1      114    0  114    100%   4.392E-68   0   0  98  16   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "...    ...  ...  ...     ...         ...  ..  ..  ..  ..   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "\n",
       "                                   SIFT       PolyPhen          BIOTYPE EXON  \\\n",
       "0                                                        protein_coding  3/3   \n",
       "0                                                       TF_binding_site        \n",
       "1                       tolerated(0.92)      benign(0)   protein_coding  3/3   \n",
       "2      deleterious_low_confidence(0.01)  benign(0.336)   protein_coding  1/1   \n",
       "2                                                                lncRNA        \n",
       "...                                 ...            ...              ...  ...   \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "\n",
       "      INTRON Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                        S81  tcA/tcG      1   \n",
       "0                                        NaN                   \n",
       "1                                      T162A  Aca/Gca      1   \n",
       "2                                      L130R  cTc/cGc     -1   \n",
       "2        1/2                             NaN              -1   \n",
       "...      ...                             ...      ...    ...   \n",
       "36651                                    NaN              -1   \n",
       "36651                                    NaN              -1   \n",
       "36651                                    NaN              -1   \n",
       "36651                                    NaN              -1   \n",
       "36651                                    NaN              -1   \n",
       "\n",
       "                                                  PUBMED  \n",
       "0                                                         \n",
       "0                                                         \n",
       "1                                                         \n",
       "2                                                         \n",
       "2                                                         \n",
       "...                                                  ...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "\n",
       "[328334 rows x 38 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf[['Gene Name', 'rsID','CHROM', 'POS', 'REF', 'ALT', 'Zygosity', 'Consequence', 'IMPACT',\n",
    "          'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT',\n",
    "          'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF',\n",
    "       'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "          'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "c22dfc42",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious_low_confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron_variant&amp;non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044&amp;29391530&amp;24262073&amp;15896936&amp;32996047&amp;3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "1         OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "...         ...          ...   ...       ...  ..  ..           ...   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "\n",
       "                                        Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                synonymous_variant       LOW                 \n",
       "0                         regulatory_region_variant  MODIFIER                 \n",
       "1                                  missense_variant  MODERATE                 \n",
       "2                                  missense_variant  MODERATE                 \n",
       "2      intron_variant&non_coding_transcript_variant  MODIFIER                 \n",
       "...                                             ...       ...           ...   \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "36651                         upstream_gene_variant  MODIFIER                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                          ENST00000641515.2   \n",
       "0                                                              \n",
       "1                                          ENST00000641515.2   \n",
       "2                                          ENST00000426406.4   \n",
       "2                                          ENST00000455207.5   \n",
       "...        ...                ...     ...                ...   \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "\n",
       "      HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  SDP  \\\n",
       "0               c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  127  138   \n",
       "0                   None                                  None  0/1  127  138   \n",
       "1               c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255  114   \n",
       "2               c.389T>G  ENSP00000409316.1        p.Leu130Arg  0/1   56   64   \n",
       "2         n.169+33750T>G                                  None  0/1   56   64   \n",
       "...                  ...                ...                ...  ...  ...  ...   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "\n",
       "        DP   RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "1      114    0  114    100%   4.392E-68   0   0  98  16   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "...    ...  ...  ...     ...         ...  ..  ..  ..  ..   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "\n",
       "                                   SIFT       PolyPhen          BIOTYPE EXON  \\\n",
       "0                                                        protein_coding  3/3   \n",
       "0                                                       TF_binding_site        \n",
       "1                       tolerated(0.92)      benign(0)   protein_coding  3/3   \n",
       "2      deleterious_low_confidence(0.01)  benign(0.336)   protein_coding  1/1   \n",
       "2                                                                lncRNA        \n",
       "...                                 ...            ...              ...  ...   \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "\n",
       "      INTRON Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                        S81  tcA/tcG      1   \n",
       "0                                        NaN                   \n",
       "1                                      T162A  Aca/Gca      1   \n",
       "2                                      L130R  cTc/cGc     -1   \n",
       "2        1/2                             NaN              -1   \n",
       "...      ...                             ...      ...    ...   \n",
       "36651                                    NaN              -1   \n",
       "36651                                    NaN              -1   \n",
       "36651                                    NaN              -1   \n",
       "36651                                    NaN              -1   \n",
       "36651                                    NaN              -1   \n",
       "\n",
       "                                                  PUBMED  \n",
       "0                                                         \n",
       "0                                                         \n",
       "1                                                         \n",
       "2                                                         \n",
       "2                                                         \n",
       "...                                                  ...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "36651  22271044&29391530&24262073&15896936&32996047&3...  \n",
       "\n",
       "[328334 rows x 38 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Define the terms to remove\n",
    "remove_terms = set([\"not_specified\", \"not_provided\"])\n",
    "\n",
    "# Apply the filtering operation to 'Column1' only\n",
    "vcf_final['ClinVar_CLNDN'] = vcf_final['ClinVar_CLNDN'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['CLIN_SIG'] = vcf_final['CLIN_SIG'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "\n",
    "vcf_final['ClinVar_CLNREVSTAT'] = vcf_final['ClinVar_CLNREVSTAT'].apply(lambda row: \"&\".join(\n",
    "    [term for term in row.split(\"&\") if term not in remove_terms]\n",
    "    ) if isinstance(row, str) and not all(term in remove_terms for term in row.split(\"&\")) else row)\n",
    "\n",
    "# Print the modified DataFrame\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "021e6f7c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "1         OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "...         ...          ...   ...       ...  ..  ..           ...   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "\n",
       "                                        Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                synonymous variant       LOW                 \n",
       "0                         regulatory region variant  MODIFIER                 \n",
       "1                                  missense variant  MODERATE                 \n",
       "2                                  missense variant  MODERATE                 \n",
       "2      intron variant,non coding transcript variant  MODIFIER                 \n",
       "...                                             ...       ...           ...   \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                          ENST00000641515.2   \n",
       "0                                                              \n",
       "1                                          ENST00000641515.2   \n",
       "2                                          ENST00000426406.4   \n",
       "2                                          ENST00000455207.5   \n",
       "...        ...                ...     ...                ...   \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "\n",
       "      HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  SDP  \\\n",
       "0               c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  127  138   \n",
       "0                   None                                  None  0/1  127  138   \n",
       "1               c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255  114   \n",
       "2               c.389T>G  ENSP00000409316.1        p.Leu130Arg  0/1   56   64   \n",
       "2         n.169+33750T>G                                  None  0/1   56   64   \n",
       "...                  ...                ...                ...  ...  ...  ...   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "\n",
       "        DP   RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "1      114    0  114    100%   4.392E-68   0   0  98  16   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "...    ...  ...  ...     ...         ...  ..  ..  ..  ..   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "\n",
       "                                   SIFT       PolyPhen          BIOTYPE EXON  \\\n",
       "0                                                        protein coding  3/3   \n",
       "0                                                       TF binding site        \n",
       "1                       tolerated(0.92)      benign(0)   protein coding  3/3   \n",
       "2      deleterious low confidence(0.01)  benign(0.336)   protein coding  1/1   \n",
       "2                                                                lncRNA        \n",
       "...                                 ...            ...              ...  ...   \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "\n",
       "      INTRON Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                        S81  tcA/tcG      1   \n",
       "0                                        nan                   \n",
       "1                                      T162A  Aca/Gca      1   \n",
       "2                                      L130R  cTc/cGc     -1   \n",
       "2        1/2                             nan              -1   \n",
       "...      ...                             ...      ...    ...   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "\n",
       "                                                  PUBMED  \n",
       "0                                                         \n",
       "0                                                         \n",
       "1                                                         \n",
       "2                                                         \n",
       "2                                                         \n",
       "...                                                  ...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "\n",
       "[328334 rows x 38 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final = vcf_final.astype(str).applymap(lambda x: x.replace('&', ',').replace('_', ' '))\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e656d536",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "1         OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "...         ...          ...   ...       ...  ..  ..           ...   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "\n",
       "                                        Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                synonymous variant       LOW                 \n",
       "0                         regulatory region variant  MODIFIER                 \n",
       "1                                  missense variant  MODERATE                 \n",
       "2                                  missense variant  MODERATE                 \n",
       "2      intron variant,non coding transcript variant  MODIFIER                 \n",
       "...                                             ...       ...           ...   \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                          ENST00000641515.2   \n",
       "0                                                              \n",
       "1                                          ENST00000641515.2   \n",
       "2                                          ENST00000426406.4   \n",
       "2                                          ENST00000455207.5   \n",
       "...        ...                ...     ...                ...   \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "\n",
       "      HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  SDP  \\\n",
       "0               c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  127  138   \n",
       "0                   None                                  None  0/1  127  138   \n",
       "1               c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255  114   \n",
       "2               c.389T>G  ENSP00000409316.1        p.Leu130Arg  0/1   56   64   \n",
       "2         n.169+33750T>G                                  None  0/1   56   64   \n",
       "...                  ...                ...                ...  ...  ...  ...   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "\n",
       "        DP   RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "1      114    0  114    100%   4.392E-68   0   0  98  16   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "...    ...  ...  ...     ...         ...  ..  ..  ..  ..   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "\n",
       "                                   SIFT       PolyPhen          BIOTYPE EXON  \\\n",
       "0                                                        protein coding  3/3   \n",
       "0                                                       TF binding site        \n",
       "1                       tolerated(0.92)      benign(0)   protein coding  3/3   \n",
       "2      deleterious low confidence(0.01)  benign(0.336)   protein coding  1/1   \n",
       "2                                                                lncRNA        \n",
       "...                                 ...            ...              ...  ...   \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "\n",
       "      INTRON Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                        S81  tcA/tcG      1   \n",
       "0                                        nan                   \n",
       "1                                      T162A  Aca/Gca      1   \n",
       "2                                      L130R  cTc/cGc     -1   \n",
       "2        1/2                             nan              -1   \n",
       "...      ...                             ...      ...    ...   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "\n",
       "                                                  PUBMED  \n",
       "0                                                         \n",
       "0                                                         \n",
       "1                                                         \n",
       "2                                                         \n",
       "2                                                         \n",
       "...                                                  ...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "\n",
       "[328334 rows x 38 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final['POS'] = vcf_final['POS'].astype('int64')\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "b1a636a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36651</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "0         OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "1         OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "2                          .  chr1    451290   A   C  Heterozygous   \n",
       "...         ...          ...   ...       ...  ..  ..           ...   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "36651    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "\n",
       "                                        Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                synonymous variant       LOW                 \n",
       "0                         regulatory region variant  MODIFIER                 \n",
       "1                                  missense variant  MODERATE                 \n",
       "2                                  missense variant  MODERATE                 \n",
       "2      intron variant,non coding transcript variant  MODIFIER                 \n",
       "...                                             ...       ...           ...   \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "36651                         upstream gene variant  MODIFIER                 \n",
       "\n",
       "      CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                          ENST00000641515.2   \n",
       "0                                                              \n",
       "1                                          ENST00000641515.2   \n",
       "2                                          ENST00000426406.4   \n",
       "2                                          ENST00000455207.5   \n",
       "...        ...                ...     ...                ...   \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "36651                                                          \n",
       "\n",
       "      HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  SDP  \\\n",
       "0               c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  127  138   \n",
       "0                   None                                  None  0/1  127  138   \n",
       "1               c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255  114   \n",
       "2               c.389T>G  ENSP00000409316.1        p.Leu130Arg  0/1   56   64   \n",
       "2         n.169+33750T>G                                  None  0/1   56   64   \n",
       "...                  ...                ...                ...  ...  ...  ...   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "36651               None                                  None  1/1  123   22   \n",
       "\n",
       "        DP   RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "0      138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "1      114    0  114    100%   4.392E-68   0   0  98  16   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "2       64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "...    ...  ...  ...     ...         ...  ..  ..  ..  ..   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "36651   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "\n",
       "                                   SIFT       PolyPhen          BIOTYPE EXON  \\\n",
       "0                                                        protein coding  3/3   \n",
       "0                                                       TF binding site        \n",
       "1                       tolerated(0.92)      benign(0)   protein coding  3/3   \n",
       "2      deleterious low confidence(0.01)  benign(0.336)   protein coding  1/1   \n",
       "2                                                                lncRNA        \n",
       "...                                 ...            ...              ...  ...   \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "36651                                                            lncRNA        \n",
       "\n",
       "      INTRON Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                        S81  tcA/tcG      1   \n",
       "0                                        nan                   \n",
       "1                                      T162A  Aca/Gca      1   \n",
       "2                                      L130R  cTc/cGc     -1   \n",
       "2        1/2                             nan              -1   \n",
       "...      ...                             ...      ...    ...   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "36651                                    nan              -1   \n",
       "\n",
       "                                                  PUBMED  \\\n",
       "0                                                          \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "2                                                          \n",
       "...                                                  ...   \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "36651  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "\n",
       "                     consequence  \n",
       "0             synonymous variant  \n",
       "0      regulatory region variant  \n",
       "1               missense variant  \n",
       "2               missense variant  \n",
       "2                 intron variant  \n",
       "...                          ...  \n",
       "36651      upstream gene variant  \n",
       "36651      upstream gene variant  \n",
       "36651      upstream gene variant  \n",
       "36651      upstream gene variant  \n",
       "36651      upstream gene variant  \n",
       "\n",
       "[328334 rows x 39 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_final['consequence'] = vcf_final['Consequence'].str.split(',').str[0]\n",
    "vcf_final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "ea413954",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript ablation</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>splice acceptor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>splice donor variant</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>stop gained</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>frameshift variant</td>\n",
       "      <td>10/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>stop lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>start lost</td>\n",
       "      <td>9/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>transcript amplification</td>\n",
       "      <td>8/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>inframe insertion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>inframe deletion</td>\n",
       "      <td>6/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>protein altering variant</td>\n",
       "      <td>5/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>splice region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>splice donor 5th base variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>splice donor region variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>splice polypyrimidine tract variant</td>\n",
       "      <td>4/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>incomplete terminal codon variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>start retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>stop retained variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>coding sequence variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>mature miRNA variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>5 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>non coding transcript exon variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NMD transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TFBS ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TFBS amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TF binding site variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>regulatory region ablation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>regulatory region amplification</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>feature elongation</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>feature truncation</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>intergenic variant</td>\n",
       "      <td>1/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            consequence Consequence_score\n",
       "0                   transcript ablation             10/10\n",
       "1               splice acceptor variant              8/10\n",
       "2                  splice donor variant              8/10\n",
       "3                           stop gained             10/10\n",
       "4                    frameshift variant             10/10\n",
       "5                             stop lost              9/10\n",
       "6                            start lost              9/10\n",
       "7              transcript amplification              8/10\n",
       "8                     inframe insertion              6/10\n",
       "9                      inframe deletion              6/10\n",
       "10                     missense variant              7/10\n",
       "11             protein altering variant              5/10\n",
       "12                splice region variant              4/10\n",
       "13        splice donor 5th base variant              4/10\n",
       "14          splice donor region variant              4/10\n",
       "15  splice polypyrimidine tract variant              4/10\n",
       "16    incomplete terminal codon variant              3/10\n",
       "17               start retained variant              3/10\n",
       "18                stop retained variant              3/10\n",
       "19                   synonymous variant              3/10\n",
       "20              coding sequence variant              3/10\n",
       "21                 mature miRNA variant              3/10\n",
       "22                  5 prime UTR variant              3/10\n",
       "23                  3 prime UTR variant              3/10\n",
       "24   non coding transcript exon variant              2/10\n",
       "25                       intron variant              2/10\n",
       "26               NMD transcript variant              2/10\n",
       "27        non coding transcript variant              2/10\n",
       "28                upstream gene variant              2/10\n",
       "29              downstream gene variant              2/10\n",
       "30                        TFBS ablation              2/10\n",
       "31                   TFBS amplification              2/10\n",
       "32              TF binding site variant              2/10\n",
       "33           regulatory region ablation              2/10\n",
       "34      regulatory region amplification              2/10\n",
       "35                   feature elongation              2/10\n",
       "36            regulatory region variant              2/10\n",
       "37                   feature truncation              1/10\n",
       "38                   intergenic variant              1/10"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/consequence.xlsx')\n",
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "9cd823b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328329</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328330</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328331</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328332</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328333</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0          OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "1          OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "2          OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "3                           .  chr1    451290   A   C  Heterozygous   \n",
       "4                           .  chr1    451290   A   C  Heterozygous   \n",
       "...          ...          ...   ...       ...  ..  ..           ...   \n",
       "328329    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328330    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328331    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328332    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328333    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "\n",
       "                                         Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                 synonymous variant       LOW                 \n",
       "1                          regulatory region variant  MODIFIER                 \n",
       "2                                   missense variant  MODERATE                 \n",
       "3                                   missense variant  MODERATE                 \n",
       "4       intron variant,non coding transcript variant  MODIFIER                 \n",
       "...                                              ...       ...           ...   \n",
       "328329                         upstream gene variant  MODIFIER                 \n",
       "328330                         upstream gene variant  MODIFIER                 \n",
       "328331                         upstream gene variant  MODIFIER                 \n",
       "328332                         upstream gene variant  MODIFIER                 \n",
       "328333                         upstream gene variant  MODIFIER                 \n",
       "\n",
       "       CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                           ENST00000641515.2   \n",
       "1                                                               \n",
       "2                                           ENST00000641515.2   \n",
       "3                                           ENST00000426406.4   \n",
       "4                                           ENST00000455207.5   \n",
       "...         ...                ...     ...                ...   \n",
       "328329                                                          \n",
       "328330                                                          \n",
       "328331                                                          \n",
       "328332                                                          \n",
       "328333                                                          \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  127   \n",
       "1                    None                                  None  0/1  127   \n",
       "2                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "3                c.389T>G  ENSP00000409316.1        p.Leu130Arg  0/1   56   \n",
       "4          n.169+33750T>G                                  None  0/1   56   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "328329               None                                  None  1/1  123   \n",
       "328330               None                                  None  1/1  123   \n",
       "328331               None                                  None  1/1  123   \n",
       "328332               None                                  None  1/1  123   \n",
       "328333               None                                  None  1/1  123   \n",
       "\n",
       "        SDP   DP   RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       138  138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "1       138  138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "2       114  114    0  114    100%   4.392E-68   0   0  98  16   \n",
       "3        64   64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "4        64   64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "...     ...  ...  ...  ...     ...         ...  ..  ..  ..  ..   \n",
       "328329   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "328330   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "328331   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "328332   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "328333   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "\n",
       "                                    SIFT       PolyPhen          BIOTYPE EXON  \\\n",
       "0                                                         protein coding  3/3   \n",
       "1                                                        TF binding site        \n",
       "2                        tolerated(0.92)      benign(0)   protein coding  3/3   \n",
       "3       deleterious low confidence(0.01)  benign(0.336)   protein coding  1/1   \n",
       "4                                                                 lncRNA        \n",
       "...                                  ...            ...              ...  ...   \n",
       "328329                                                            lncRNA        \n",
       "328330                                                            lncRNA        \n",
       "328331                                                            lncRNA        \n",
       "328332                                                            lncRNA        \n",
       "328333                                                            lncRNA        \n",
       "\n",
       "       INTRON Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                         S81  tcA/tcG      1   \n",
       "1                                         nan                   \n",
       "2                                       T162A  Aca/Gca      1   \n",
       "3                                       L130R  cTc/cGc     -1   \n",
       "4         1/2                             nan              -1   \n",
       "...       ...                             ...      ...    ...   \n",
       "328329                                    nan              -1   \n",
       "328330                                    nan              -1   \n",
       "328331                                    nan              -1   \n",
       "328332                                    nan              -1   \n",
       "328333                                    nan              -1   \n",
       "\n",
       "                                                   PUBMED  \\\n",
       "0                                                           \n",
       "1                                                           \n",
       "2                                                           \n",
       "3                                                           \n",
       "4                                                           \n",
       "...                                                   ...   \n",
       "328329  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "328330  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "328331  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "328332  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "328333  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "\n",
       "                      consequence Consequence_score  \n",
       "0              synonymous variant              3/10  \n",
       "1       regulatory region variant              2/10  \n",
       "2                missense variant              7/10  \n",
       "3                missense variant              7/10  \n",
       "4                  intron variant              2/10  \n",
       "...                           ...               ...  \n",
       "328329      upstream gene variant              2/10  \n",
       "328330      upstream gene variant              2/10  \n",
       "328331      upstream gene variant              2/10  \n",
       "328332      upstream gene variant              2/10  \n",
       "328333      upstream gene variant              2/10  \n",
       "\n",
       "[328334 rows x 40 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_1 = pd.merge(vcf_final, df_1, on='consequence', how='left', sort=False)\n",
    "merged_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "aef93493",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HIGH</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     IMPACT  IMPACT_score\n",
       "0      HIGH          10.0\n",
       "1  MODERATE           5.0\n",
       "2       LOW           2.5\n",
       "3  MODIFIER           1.5"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_2 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Madhu_folder_04_07_2023/kidney_health_final.vcf/IMPACT.xlsx')\n",
    "df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "3cb687c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>2.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>intron variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328329</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328330</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328331</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328332</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328333</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>1.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0          OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "1          OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "2          OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "3                           .  chr1    451290   A   C  Heterozygous   \n",
       "4                           .  chr1    451290   A   C  Heterozygous   \n",
       "...          ...          ...   ...       ...  ..  ..           ...   \n",
       "328329    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328330    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328331    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328332    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328333    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "\n",
       "                                         Consequence    IMPACT ClinVar_CLNDN  \\\n",
       "0                                 synonymous variant       LOW                 \n",
       "1                          regulatory region variant  MODIFIER                 \n",
       "2                                   missense variant  MODERATE                 \n",
       "3                                   missense variant  MODERATE                 \n",
       "4       intron variant,non coding transcript variant  MODIFIER                 \n",
       "...                                              ...       ...           ...   \n",
       "328329                         upstream gene variant  MODIFIER                 \n",
       "328330                         upstream gene variant  MODIFIER                 \n",
       "328331                         upstream gene variant  MODIFIER                 \n",
       "328332                         upstream gene variant  MODIFIER                 \n",
       "328333                         upstream gene variant  MODIFIER                 \n",
       "\n",
       "       CLIN_SIG ClinVar_CLNREVSTAT ClinVar              HGVSc  \\\n",
       "0                                           ENST00000641515.2   \n",
       "1                                                               \n",
       "2                                           ENST00000641515.2   \n",
       "3                                           ENST00000426406.4   \n",
       "4                                           ENST00000455207.5   \n",
       "...         ...                ...     ...                ...   \n",
       "328329                                                          \n",
       "328330                                                          \n",
       "328331                                                          \n",
       "328332                                                          \n",
       "328333                                                          \n",
       "\n",
       "       HGVSc (Transcript)              HGVSp HGVSp (Transcript)   GT   GQ  \\\n",
       "0                c.243A>G  ENSP00000493376.2         p.Ser81%3D  0/1  127   \n",
       "1                    None                                  None  0/1  127   \n",
       "2                c.484A>G  ENSP00000493376.2        p.Thr162Ala  1/1  255   \n",
       "3                c.389T>G  ENSP00000409316.1        p.Leu130Arg  0/1   56   \n",
       "4          n.169+33750T>G                                  None  0/1   56   \n",
       "...                   ...                ...                ...  ...  ...   \n",
       "328329               None                                  None  1/1  123   \n",
       "328330               None                                  None  1/1  123   \n",
       "328331               None                                  None  1/1  123   \n",
       "328332               None                                  None  1/1  123   \n",
       "328333               None                                  None  1/1  123   \n",
       "\n",
       "        SDP   DP   RD   AD    FREQ        PVAL RDF RDR ADF ADR  \\\n",
       "0       138  138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "1       138  138  100   38  27.54%  1.8855E-13  86  14  29   9   \n",
       "2       114  114    0  114    100%   4.392E-68   0   0  98  16   \n",
       "3        64   64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "4        64   64   47   17  26.56%    2.243E-6  33  14  15   2   \n",
       "...     ...  ...  ...  ...     ...         ...  ..  ..  ..  ..   \n",
       "328329   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "328330   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "328331   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "328332   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "328333   22   22    0   22    100%  4.7526E-13   0   0  14   8   \n",
       "\n",
       "                                    SIFT       PolyPhen          BIOTYPE EXON  \\\n",
       "0                                                         protein coding  3/3   \n",
       "1                                                        TF binding site        \n",
       "2                        tolerated(0.92)      benign(0)   protein coding  3/3   \n",
       "3       deleterious low confidence(0.01)  benign(0.336)   protein coding  1/1   \n",
       "4                                                                 lncRNA        \n",
       "...                                  ...            ...              ...  ...   \n",
       "328329                                                            lncRNA        \n",
       "328330                                                            lncRNA        \n",
       "328331                                                            lncRNA        \n",
       "328332                                                            lncRNA        \n",
       "328333                                                            lncRNA        \n",
       "\n",
       "       INTRON Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                         S81  tcA/tcG      1   \n",
       "1                                         nan                   \n",
       "2                                       T162A  Aca/Gca      1   \n",
       "3                                       L130R  cTc/cGc     -1   \n",
       "4         1/2                             nan              -1   \n",
       "...       ...                             ...      ...    ...   \n",
       "328329                                    nan              -1   \n",
       "328330                                    nan              -1   \n",
       "328331                                    nan              -1   \n",
       "328332                                    nan              -1   \n",
       "328333                                    nan              -1   \n",
       "\n",
       "                                                   PUBMED  \\\n",
       "0                                                           \n",
       "1                                                           \n",
       "2                                                           \n",
       "3                                                           \n",
       "4                                                           \n",
       "...                                                   ...   \n",
       "328329  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "328330  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "328331  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "328332  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "328333  22271044,29391530,24262073,15896936,32996047,3...   \n",
       "\n",
       "                      consequence Consequence_score  IMPACT_score  \n",
       "0              synonymous variant              3/10           2.5  \n",
       "1       regulatory region variant              2/10           1.5  \n",
       "2                missense variant              7/10           5.0  \n",
       "3                missense variant              7/10           5.0  \n",
       "4                  intron variant              2/10           1.5  \n",
       "...                           ...               ...           ...  \n",
       "328329      upstream gene variant              2/10           1.5  \n",
       "328330      upstream gene variant              2/10           1.5  \n",
       "328331      upstream gene variant              2/10           1.5  \n",
       "328332      upstream gene variant              2/10           1.5  \n",
       "328333      upstream gene variant              2/10           1.5  \n",
       "\n",
       "[328334 rows x 41 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = pd.merge(merged_1, df_2, on = 'IMPACT', how='left', sort=False)\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "335fcc8b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328329</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328330</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328331</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328332</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328333</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0          OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "1          OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "2          OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "3                           .  chr1    451290   A   C  Heterozygous   \n",
       "4                           .  chr1    451290   A   C  Heterozygous   \n",
       "...          ...          ...   ...       ...  ..  ..           ...   \n",
       "328329    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328330    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328331    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328332    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328333    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                 synonymous variant              3/10   \n",
       "1                          regulatory region variant              2/10   \n",
       "2                                   missense variant              7/10   \n",
       "3                                   missense variant              7/10   \n",
       "4       intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "328329                         upstream gene variant              2/10   \n",
       "328330                         upstream gene variant              2/10   \n",
       "328331                         upstream gene variant              2/10   \n",
       "328332                         upstream gene variant              2/10   \n",
       "328333                         upstream gene variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  \\\n",
       "0            LOW           2.5                                             \n",
       "1       MODIFIER           1.5                                             \n",
       "2       MODERATE           5.0                                             \n",
       "3       MODERATE           5.0                                             \n",
       "4       MODIFIER           1.5                                             \n",
       "...          ...           ...           ...      ...                ...   \n",
       "328329  MODIFIER           1.5                                             \n",
       "328330  MODIFIER           1.5                                             \n",
       "328331  MODIFIER           1.5                                             \n",
       "328332  MODIFIER           1.5                                             \n",
       "328333  MODIFIER           1.5                                             \n",
       "\n",
       "       ClinVar              HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0               ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "1                                               None                      \n",
       "2               ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "3               ENST00000426406.4           c.389T>G  ENSP00000409316.1   \n",
       "4               ENST00000455207.5     n.169+33750T>G                      \n",
       "...        ...                ...                ...                ...   \n",
       "328329                                          None                      \n",
       "328330                                          None                      \n",
       "328331                                          None                      \n",
       "328332                                          None                      \n",
       "328333                                          None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD    FREQ        PVAL  \\\n",
       "0              p.Ser81%3D  0/1  127  138  138  100   38  27.54%  1.8855E-13   \n",
       "1                    None  0/1  127  138  138  100   38  27.54%  1.8855E-13   \n",
       "2             p.Thr162Ala  1/1  255  114  114    0  114    100%   4.392E-68   \n",
       "3             p.Leu130Arg  0/1   56   64   64   47   17  26.56%    2.243E-6   \n",
       "4                    None  0/1   56   64   64   47   17  26.56%    2.243E-6   \n",
       "...                   ...  ...  ...  ...  ...  ...  ...     ...         ...   \n",
       "328329               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "328330               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "328331               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "328332               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "328333               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "\n",
       "       RDF RDR ADF ADR                              SIFT       PolyPhen  \\\n",
       "0       86  14  29   9                                                    \n",
       "1       86  14  29   9                                                    \n",
       "2        0   0  98  16                   tolerated(0.92)      benign(0)   \n",
       "3       33  14  15   2  deleterious low confidence(0.01)  benign(0.336)   \n",
       "4       33  14  15   2                                                    \n",
       "...     ..  ..  ..  ..                               ...            ...   \n",
       "328329   0   0  14   8                                                    \n",
       "328330   0   0  14   8                                                    \n",
       "328331   0   0  14   8                                                    \n",
       "328332   0   0  14   8                                                    \n",
       "328333   0   0  14   8                                                    \n",
       "\n",
       "                BIOTYPE EXON INTRON Protein Position and Amino Acid   Codons  \\\n",
       "0        protein coding  3/3                                    S81  tcA/tcG   \n",
       "1       TF binding site                                         nan            \n",
       "2        protein coding  3/3                                  T162A  Aca/Gca   \n",
       "3        protein coding  1/1                                  L130R  cTc/cGc   \n",
       "4                lncRNA         1/2                             nan            \n",
       "...                 ...  ...    ...                             ...      ...   \n",
       "328329           lncRNA                                         nan            \n",
       "328330           lncRNA                                         nan            \n",
       "328331           lncRNA                                         nan            \n",
       "328332           lncRNA                                         nan            \n",
       "328333           lncRNA                                         nan            \n",
       "\n",
       "       STRAND                                             PUBMED  \n",
       "0           1                                                     \n",
       "1                                                                 \n",
       "2           1                                                     \n",
       "3          -1                                                     \n",
       "4          -1                                                     \n",
       "...       ...                                                ...  \n",
       "328329     -1  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328330     -1  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328331     -1  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328332     -1  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328333     -1  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "\n",
       "[328334 rows x 40 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "f33e14af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LDLR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LDLR-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NOS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRKAG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LRP6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>TRDN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>LDLRAP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>PCSK9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>APOB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>STAP1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Name\n",
       "0       LDLR\n",
       "1   LDLR-AS1\n",
       "2       NOS3\n",
       "3     PRKAG2\n",
       "4       LRP6\n",
       "..       ...\n",
       "75      TRDN\n",
       "76   LDLRAP1\n",
       "77     PCSK9\n",
       "78      APOB\n",
       "79     STAP1\n",
       "\n",
       "[80 rows x 1 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/cardiac genes.xlsx')\n",
    "df_gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "ebdbac68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Gene_Match</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328329</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328330</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328331</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328332</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328333</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name         rsID CHROM       POS REF ALT      Zygosity  \\\n",
       "0          OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "1          OR4F5  rs201219564  chr1     69270   A   G  Heterozygous   \n",
       "2          OR4F5    rs2691305  chr1     69511   A   G    Homozygous   \n",
       "3                           .  chr1    451290   A   C  Heterozygous   \n",
       "4                           .  chr1    451290   A   C  Heterozygous   \n",
       "...          ...          ...   ...       ...  ..  ..           ...   \n",
       "328329    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328330    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328331    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328332    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "328333    EIF1AY    rs9786153  chrY  20577481   C   T    Homozygous   \n",
       "\n",
       "                                         Consequence Consequence_score  \\\n",
       "0                                 synonymous variant              3/10   \n",
       "1                          regulatory region variant              2/10   \n",
       "2                                   missense variant              7/10   \n",
       "3                                   missense variant              7/10   \n",
       "4       intron variant,non coding transcript variant              2/10   \n",
       "...                                              ...               ...   \n",
       "328329                         upstream gene variant              2/10   \n",
       "328330                         upstream gene variant              2/10   \n",
       "328331                         upstream gene variant              2/10   \n",
       "328332                         upstream gene variant              2/10   \n",
       "328333                         upstream gene variant              2/10   \n",
       "\n",
       "          IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG ClinVar_CLNREVSTAT  \\\n",
       "0            LOW           2.5                                             \n",
       "1       MODIFIER           1.5                                             \n",
       "2       MODERATE           5.0                                             \n",
       "3       MODERATE           5.0                                             \n",
       "4       MODIFIER           1.5                                             \n",
       "...          ...           ...           ...      ...                ...   \n",
       "328329  MODIFIER           1.5                                             \n",
       "328330  MODIFIER           1.5                                             \n",
       "328331  MODIFIER           1.5                                             \n",
       "328332  MODIFIER           1.5                                             \n",
       "328333  MODIFIER           1.5                                             \n",
       "\n",
       "       ClinVar              HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "0               ENST00000641515.2           c.243A>G  ENSP00000493376.2   \n",
       "1                                               None                      \n",
       "2               ENST00000641515.2           c.484A>G  ENSP00000493376.2   \n",
       "3               ENST00000426406.4           c.389T>G  ENSP00000409316.1   \n",
       "4               ENST00000455207.5     n.169+33750T>G                      \n",
       "...        ...                ...                ...                ...   \n",
       "328329                                          None                      \n",
       "328330                                          None                      \n",
       "328331                                          None                      \n",
       "328332                                          None                      \n",
       "328333                                          None                      \n",
       "\n",
       "       HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD    FREQ        PVAL  \\\n",
       "0              p.Ser81%3D  0/1  127  138  138  100   38  27.54%  1.8855E-13   \n",
       "1                    None  0/1  127  138  138  100   38  27.54%  1.8855E-13   \n",
       "2             p.Thr162Ala  1/1  255  114  114    0  114    100%   4.392E-68   \n",
       "3             p.Leu130Arg  0/1   56   64   64   47   17  26.56%    2.243E-6   \n",
       "4                    None  0/1   56   64   64   47   17  26.56%    2.243E-6   \n",
       "...                   ...  ...  ...  ...  ...  ...  ...     ...         ...   \n",
       "328329               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "328330               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "328331               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "328332               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "328333               None  1/1  123   22   22    0   22    100%  4.7526E-13   \n",
       "\n",
       "       RDF RDR ADF ADR                              SIFT       PolyPhen  \\\n",
       "0       86  14  29   9                                                    \n",
       "1       86  14  29   9                                                    \n",
       "2        0   0  98  16                   tolerated(0.92)      benign(0)   \n",
       "3       33  14  15   2  deleterious low confidence(0.01)  benign(0.336)   \n",
       "4       33  14  15   2                                                    \n",
       "...     ..  ..  ..  ..                               ...            ...   \n",
       "328329   0   0  14   8                                                    \n",
       "328330   0   0  14   8                                                    \n",
       "328331   0   0  14   8                                                    \n",
       "328332   0   0  14   8                                                    \n",
       "328333   0   0  14   8                                                    \n",
       "\n",
       "                BIOTYPE EXON INTRON Protein Position and Amino Acid   Codons  \\\n",
       "0        protein coding  3/3                                    S81  tcA/tcG   \n",
       "1       TF binding site                                         nan            \n",
       "2        protein coding  3/3                                  T162A  Aca/Gca   \n",
       "3        protein coding  1/1                                  L130R  cTc/cGc   \n",
       "4                lncRNA         1/2                             nan            \n",
       "...                 ...  ...    ...                             ...      ...   \n",
       "328329           lncRNA                                         nan            \n",
       "328330           lncRNA                                         nan            \n",
       "328331           lncRNA                                         nan            \n",
       "328332           lncRNA                                         nan            \n",
       "328333           lncRNA                                         nan            \n",
       "\n",
       "       STRAND                                             PUBMED Gene_Match  \n",
       "0           1                                                            No  \n",
       "1                                                                        No  \n",
       "2           1                                                            No  \n",
       "3          -1                                                            No  \n",
       "4          -1                                                            No  \n",
       "...       ...                                                ...        ...  \n",
       "328329     -1  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "328330     -1  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "328331     -1  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "328332     -1  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "328333     -1  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "\n",
       "[328334 rows x 41 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2['Gene_Match'] = 'No'\n",
    "\n",
    "# Iterate through each gene in df1\n",
    "for genes in merged_2['Gene Name']:\n",
    "    if isinstance(genes, str):  # Check if the gene value is a non-null string\n",
    "        gene_list = genes.split(',')  # Split the genes by comma to create a list\n",
    "        match = any(gene in df_gene['Gene Name'].values for gene in gene_list)  # Check if any gene in the list exists in df2\n",
    "        if match:\n",
    "            merged_2.loc[merged_2['Gene Name'] == genes, 'Gene_Match'] = 'Yes'\n",
    "            \n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "95832285",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     322967\n",
       "Yes      5367\n",
       "Name: Gene_Match, dtype: int64"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2.Gene_Match.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "33e5c65a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328329</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328330</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328331</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328332</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328333</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match         rsID CHROM       POS REF ALT  \\\n",
       "0          OR4F5         No  rs201219564  chr1     69270   A   G   \n",
       "1          OR4F5         No  rs201219564  chr1     69270   A   G   \n",
       "2          OR4F5         No    rs2691305  chr1     69511   A   G   \n",
       "3                        No            .  chr1    451290   A   C   \n",
       "4                        No            .  chr1    451290   A   C   \n",
       "...          ...        ...          ...   ...       ...  ..  ..   \n",
       "328329    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "328330    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "328331    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "328332    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "328333    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "\n",
       "            Zygosity                                   Consequence  \\\n",
       "0       Heterozygous                            synonymous variant   \n",
       "1       Heterozygous                     regulatory region variant   \n",
       "2         Homozygous                              missense variant   \n",
       "3       Heterozygous                              missense variant   \n",
       "4       Heterozygous  intron variant,non coding transcript variant   \n",
       "...              ...                                           ...   \n",
       "328329    Homozygous                         upstream gene variant   \n",
       "328330    Homozygous                         upstream gene variant   \n",
       "328331    Homozygous                         upstream gene variant   \n",
       "328332    Homozygous                         upstream gene variant   \n",
       "328333    Homozygous                         upstream gene variant   \n",
       "\n",
       "       Consequence_score    IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0                   3/10       LOW           2.5                          \n",
       "1                   2/10  MODIFIER           1.5                          \n",
       "2                   7/10  MODERATE           5.0                          \n",
       "3                   7/10  MODERATE           5.0                          \n",
       "4                   2/10  MODIFIER           1.5                          \n",
       "...                  ...       ...           ...           ...      ...   \n",
       "328329              2/10  MODIFIER           1.5                          \n",
       "328330              2/10  MODIFIER           1.5                          \n",
       "328331              2/10  MODIFIER           1.5                          \n",
       "328332              2/10  MODIFIER           1.5                          \n",
       "328333              2/10  MODIFIER           1.5                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                  ENST00000641515.2           c.243A>G   \n",
       "1                                                                  None   \n",
       "2                                  ENST00000641515.2           c.484A>G   \n",
       "3                                  ENST00000426406.4           c.389T>G   \n",
       "4                                  ENST00000455207.5     n.169+33750T>G   \n",
       "...                   ...     ...                ...                ...   \n",
       "328329                                                             None   \n",
       "328330                                                             None   \n",
       "328331                                                             None   \n",
       "328332                                                             None   \n",
       "328333                                                             None   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD  \\\n",
       "0       ENSP00000493376.2         p.Ser81%3D  0/1  127  138  138  100   38   \n",
       "1                                       None  0/1  127  138  138  100   38   \n",
       "2       ENSP00000493376.2        p.Thr162Ala  1/1  255  114  114    0  114   \n",
       "3       ENSP00000409316.1        p.Leu130Arg  0/1   56   64   64   47   17   \n",
       "4                                       None  0/1   56   64   64   47   17   \n",
       "...                   ...                ...  ...  ...  ...  ...  ...  ...   \n",
       "328329                                  None  1/1  123   22   22    0   22   \n",
       "328330                                  None  1/1  123   22   22    0   22   \n",
       "328331                                  None  1/1  123   22   22    0   22   \n",
       "328332                                  None  1/1  123   22   22    0   22   \n",
       "328333                                  None  1/1  123   22   22    0   22   \n",
       "\n",
       "          FREQ        PVAL RDF RDR ADF ADR                              SIFT  \\\n",
       "0       27.54%  1.8855E-13  86  14  29   9                                     \n",
       "1       27.54%  1.8855E-13  86  14  29   9                                     \n",
       "2         100%   4.392E-68   0   0  98  16                   tolerated(0.92)   \n",
       "3       26.56%    2.243E-6  33  14  15   2  deleterious low confidence(0.01)   \n",
       "4       26.56%    2.243E-6  33  14  15   2                                     \n",
       "...        ...         ...  ..  ..  ..  ..                               ...   \n",
       "328329    100%  4.7526E-13   0   0  14   8                                     \n",
       "328330    100%  4.7526E-13   0   0  14   8                                     \n",
       "328331    100%  4.7526E-13   0   0  14   8                                     \n",
       "328332    100%  4.7526E-13   0   0  14   8                                     \n",
       "328333    100%  4.7526E-13   0   0  14   8                                     \n",
       "\n",
       "             PolyPhen          BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                      TF binding site               \n",
       "2           benign(0)   protein coding  3/3          \n",
       "3       benign(0.336)   protein coding  1/1          \n",
       "4                               lncRNA         1/2   \n",
       "...               ...              ...  ...    ...   \n",
       "328329                          lncRNA               \n",
       "328330                          lncRNA               \n",
       "328331                          lncRNA               \n",
       "328332                          lncRNA               \n",
       "328333                          lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                  S81  tcA/tcG      1   \n",
       "1                                  nan                   \n",
       "2                                T162A  Aca/Gca      1   \n",
       "3                                L130R  cTc/cGc     -1   \n",
       "4                                  nan              -1   \n",
       "...                                ...      ...    ...   \n",
       "328329                             nan              -1   \n",
       "328330                             nan              -1   \n",
       "328331                             nan              -1   \n",
       "328332                             nan              -1   \n",
       "328333                             nan              -1   \n",
       "\n",
       "                                                   PUBMED  \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "3                                                          \n",
       "4                                                          \n",
       "...                                                   ...  \n",
       "328329  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328330  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328331  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328332  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328333  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "\n",
       "[328334 rows x 41 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_2 = merged_2[['Gene Name', 'Gene_Match', 'rsID', 'CHROM', 'POS', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence', 'Consequence_score', 'IMPACT', 'IMPACT_score',\n",
    "       'ClinVar_CLNDN', 'CLIN_SIG', 'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc',\n",
    "       'HGVSc (Transcript)', 'HGVSp', 'HGVSp (Transcript)', 'GT', 'GQ', 'SDP',\n",
    "       'DP', 'RD', 'AD', 'FREQ', 'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT',\n",
    "       'PolyPhen', 'BIOTYPE', 'EXON', 'INTRON',\n",
    "       'Protein Position and Amino Acid', 'Codons', 'STRAND', 'PUBMED']]\n",
    "merged_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "33d91f84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>104858586</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>104826974</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>104824472</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>104858554</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>104794495</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1861</th>\n",
       "      <td>31592980</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1862</th>\n",
       "      <td>31595223</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1863</th>\n",
       "      <td>31593004</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1864</th>\n",
       "      <td>31598691</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1865</th>\n",
       "      <td>31595256</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1866 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            POS Literature\n",
       "0     104858586        Yes\n",
       "1     104826974        Yes\n",
       "2     104824472        Yes\n",
       "3     104858554        Yes\n",
       "4     104794495        Yes\n",
       "...         ...        ...\n",
       "1861   31592980        Yes\n",
       "1862   31595223        Yes\n",
       "1863   31593004        Yes\n",
       "1864   31598691        Yes\n",
       "1865   31595256        Yes\n",
       "\n",
       "[1866 rows x 2 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3 = pd.read_excel(r'C:/Users/GenepoweRx_Madhu/Desktop/Condition_pos/Cardiac_pos.xlsx')\n",
    "df_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "dbf9a23f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328329</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328330</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328331</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328332</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328333</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match         rsID CHROM       POS REF ALT  \\\n",
       "0          OR4F5         No  rs201219564  chr1     69270   A   G   \n",
       "1          OR4F5         No  rs201219564  chr1     69270   A   G   \n",
       "2          OR4F5         No    rs2691305  chr1     69511   A   G   \n",
       "3                        No            .  chr1    451290   A   C   \n",
       "4                        No            .  chr1    451290   A   C   \n",
       "...          ...        ...          ...   ...       ...  ..  ..   \n",
       "328329    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "328330    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "328331    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "328332    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "328333    EIF1AY         No    rs9786153  chrY  20577481   C   T   \n",
       "\n",
       "            Zygosity                                   Consequence  \\\n",
       "0       Heterozygous                            synonymous variant   \n",
       "1       Heterozygous                     regulatory region variant   \n",
       "2         Homozygous                              missense variant   \n",
       "3       Heterozygous                              missense variant   \n",
       "4       Heterozygous  intron variant,non coding transcript variant   \n",
       "...              ...                                           ...   \n",
       "328329    Homozygous                         upstream gene variant   \n",
       "328330    Homozygous                         upstream gene variant   \n",
       "328331    Homozygous                         upstream gene variant   \n",
       "328332    Homozygous                         upstream gene variant   \n",
       "328333    Homozygous                         upstream gene variant   \n",
       "\n",
       "       Consequence_score    IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0                   3/10       LOW           2.5                          \n",
       "1                   2/10  MODIFIER           1.5                          \n",
       "2                   7/10  MODERATE           5.0                          \n",
       "3                   7/10  MODERATE           5.0                          \n",
       "4                   2/10  MODIFIER           1.5                          \n",
       "...                  ...       ...           ...           ...      ...   \n",
       "328329              2/10  MODIFIER           1.5                          \n",
       "328330              2/10  MODIFIER           1.5                          \n",
       "328331              2/10  MODIFIER           1.5                          \n",
       "328332              2/10  MODIFIER           1.5                          \n",
       "328333              2/10  MODIFIER           1.5                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                  ENST00000641515.2           c.243A>G   \n",
       "1                                                                  None   \n",
       "2                                  ENST00000641515.2           c.484A>G   \n",
       "3                                  ENST00000426406.4           c.389T>G   \n",
       "4                                  ENST00000455207.5     n.169+33750T>G   \n",
       "...                   ...     ...                ...                ...   \n",
       "328329                                                             None   \n",
       "328330                                                             None   \n",
       "328331                                                             None   \n",
       "328332                                                             None   \n",
       "328333                                                             None   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD  \\\n",
       "0       ENSP00000493376.2         p.Ser81%3D  0/1  127  138  138  100   38   \n",
       "1                                       None  0/1  127  138  138  100   38   \n",
       "2       ENSP00000493376.2        p.Thr162Ala  1/1  255  114  114    0  114   \n",
       "3       ENSP00000409316.1        p.Leu130Arg  0/1   56   64   64   47   17   \n",
       "4                                       None  0/1   56   64   64   47   17   \n",
       "...                   ...                ...  ...  ...  ...  ...  ...  ...   \n",
       "328329                                  None  1/1  123   22   22    0   22   \n",
       "328330                                  None  1/1  123   22   22    0   22   \n",
       "328331                                  None  1/1  123   22   22    0   22   \n",
       "328332                                  None  1/1  123   22   22    0   22   \n",
       "328333                                  None  1/1  123   22   22    0   22   \n",
       "\n",
       "          FREQ        PVAL RDF RDR ADF ADR                              SIFT  \\\n",
       "0       27.54%  1.8855E-13  86  14  29   9                                     \n",
       "1       27.54%  1.8855E-13  86  14  29   9                                     \n",
       "2         100%   4.392E-68   0   0  98  16                   tolerated(0.92)   \n",
       "3       26.56%    2.243E-6  33  14  15   2  deleterious low confidence(0.01)   \n",
       "4       26.56%    2.243E-6  33  14  15   2                                     \n",
       "...        ...         ...  ..  ..  ..  ..                               ...   \n",
       "328329    100%  4.7526E-13   0   0  14   8                                     \n",
       "328330    100%  4.7526E-13   0   0  14   8                                     \n",
       "328331    100%  4.7526E-13   0   0  14   8                                     \n",
       "328332    100%  4.7526E-13   0   0  14   8                                     \n",
       "328333    100%  4.7526E-13   0   0  14   8                                     \n",
       "\n",
       "             PolyPhen          BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                      TF binding site               \n",
       "2           benign(0)   protein coding  3/3          \n",
       "3       benign(0.336)   protein coding  1/1          \n",
       "4                               lncRNA         1/2   \n",
       "...               ...              ...  ...    ...   \n",
       "328329                          lncRNA               \n",
       "328330                          lncRNA               \n",
       "328331                          lncRNA               \n",
       "328332                          lncRNA               \n",
       "328333                          lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                  S81  tcA/tcG      1   \n",
       "1                                  nan                   \n",
       "2                                T162A  Aca/Gca      1   \n",
       "3                                L130R  cTc/cGc     -1   \n",
       "4                                  nan              -1   \n",
       "...                                ...      ...    ...   \n",
       "328329                             nan              -1   \n",
       "328330                             nan              -1   \n",
       "328331                             nan              -1   \n",
       "328332                             nan              -1   \n",
       "328333                             nan              -1   \n",
       "\n",
       "                                                   PUBMED Literature  \n",
       "0                                                                 No  \n",
       "1                                                                 No  \n",
       "2                                                                 No  \n",
       "3                                                                 No  \n",
       "4                                                                 No  \n",
       "...                                                   ...        ...  \n",
       "328329  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "328330  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "328331  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "328332  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "328333  22271044,29391530,24262073,15896936,32996047,3...         No  \n",
       "\n",
       "[328334 rows x 42 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = pd.merge(merged_2, df_3, on = 'POS', how = 'left', sort=False)\n",
    "merged_3['Literature'] = merged_3['Literature'].fillna('No')\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "0b101b9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     328255\n",
       "Yes        79\n",
       "Name: Literature, dtype: int64"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3.Literature.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "4ca2a81b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>Literature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15189</th>\n",
       "      <td>CASQ2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs7521023</td>\n",
       "      <td>chr1</td>\n",
       "      <td>115700759</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>3 prime UTR variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Caudal regression sequence,Catecholaminergic p...</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>292122</td>\n",
       "      <td>ENST00000261448.6</td>\n",
       "      <td>c.*482C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>53.85%</td>\n",
       "      <td>5.4595E-6</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>11/11</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>27899944,24444446,31053292,32765588</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15190</th>\n",
       "      <td>CASQ2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs7521023</td>\n",
       "      <td>chr1</td>\n",
       "      <td>115700759</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Caudal regression sequence,Catecholaminergic p...</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>292122</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>53.85%</td>\n",
       "      <td>5.4595E-6</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>27899944,24444446,31053292,32765588</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15191</th>\n",
       "      <td>CASQ2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs7521023</td>\n",
       "      <td>chr1</td>\n",
       "      <td>115700759</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Caudal regression sequence,Catecholaminergic p...</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>292122</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>52</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>53.85%</td>\n",
       "      <td>5.4595E-6</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>27899944,24444446,31053292,32765588</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23277</th>\n",
       "      <td>TNNT2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs3730238</td>\n",
       "      <td>chr1</td>\n",
       "      <td>201361301</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Dilated Cardiomyopathy, Dominant,Cardiovascula...</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>43669</td>\n",
       "      <td>ENST00000236918.11</td>\n",
       "      <td>c.788A&gt;G</td>\n",
       "      <td>ENSP00000236918.8</td>\n",
       "      <td>p.Lys263Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>68</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>56.25%</td>\n",
       "      <td>1.3089E-7</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious low confidence(0.04)</td>\n",
       "      <td>benign(0.007)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>14/16</td>\n",
       "      <td></td>\n",
       "      <td>K263R</td>\n",
       "      <td>aAg/aGg</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,15542288,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23278</th>\n",
       "      <td>TNNT2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs3730238</td>\n",
       "      <td>chr1</td>\n",
       "      <td>201361301</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Dilated Cardiomyopathy, Dominant,Cardiovascula...</td>\n",
       "      <td>benign,likely benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>43669</td>\n",
       "      <td>ENST00000360372.8</td>\n",
       "      <td>c.659A&gt;G</td>\n",
       "      <td>ENSP00000353535.5</td>\n",
       "      <td>p.Lys220Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>68</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>56.25%</td>\n",
       "      <td>1.3089E-7</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>benign(0.388)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>12/14</td>\n",
       "      <td></td>\n",
       "      <td>K220R</td>\n",
       "      <td>aAg/aGg</td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,24033266,18414213,23861362,15542288,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297400</th>\n",
       "      <td>KCNH2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1805123</td>\n",
       "      <td>chr7</td>\n",
       "      <td>150948446</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Cardiac arrhythmia,Cardiovascular phenotype,At...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>67427</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>93</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>94.44%</td>\n",
       "      <td>4.2852E-10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,21056700,24033266,20850564,23820649,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297401</th>\n",
       "      <td>KCNH2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1805123</td>\n",
       "      <td>chr7</td>\n",
       "      <td>150948446</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Cardiac arrhythmia,Cardiovascular phenotype,At...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>67427</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>93</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>94.44%</td>\n",
       "      <td>4.2852E-10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,21056700,24033266,20850564,23820649,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297402</th>\n",
       "      <td>KCNH2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1805123</td>\n",
       "      <td>chr7</td>\n",
       "      <td>150948446</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Cardiac arrhythmia,Cardiovascular phenotype,At...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>67427</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>93</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>94.44%</td>\n",
       "      <td>4.2852E-10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,21056700,24033266,20850564,23820649,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297403</th>\n",
       "      <td>KCNH2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1805123</td>\n",
       "      <td>chr7</td>\n",
       "      <td>150948446</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>downstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td>Cardiac arrhythmia,Cardiovascular phenotype,At...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>67427</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>93</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>94.44%</td>\n",
       "      <td>4.2852E-10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,21056700,24033266,20850564,23820649,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297404</th>\n",
       "      <td>KCNH2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>rs1805123</td>\n",
       "      <td>chr7</td>\n",
       "      <td>150948446</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>splice region variant,non coding transcript ex...</td>\n",
       "      <td>4/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td>Cardiac arrhythmia,Cardiovascular phenotype,At...</td>\n",
       "      <td>benign</td>\n",
       "      <td>criteria provided, multiple submitters, no con...</td>\n",
       "      <td>67427</td>\n",
       "      <td>ENST00000684241.1</td>\n",
       "      <td>n.3523A&gt;C</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>93</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>94.44%</td>\n",
       "      <td>4.2852E-10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>retained intron</td>\n",
       "      <td>9/13</td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>25741868,21056700,24033266,20850564,23820649,2...</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match       rsID CHROM        POS REF ALT      Zygosity  \\\n",
       "15189      CASQ2        Yes  rs7521023  chr1  115700759   G   A  Heterozygous   \n",
       "15190      CASQ2        Yes  rs7521023  chr1  115700759   G   A  Heterozygous   \n",
       "15191      CASQ2        Yes  rs7521023  chr1  115700759   G   A  Heterozygous   \n",
       "23277      TNNT2        Yes  rs3730238  chr1  201361301   T   C  Heterozygous   \n",
       "23278      TNNT2        Yes  rs3730238  chr1  201361301   T   C  Heterozygous   \n",
       "...          ...        ...        ...   ...        ...  ..  ..           ...   \n",
       "297400     KCNH2        Yes  rs1805123  chr7  150948446   T   G    Homozygous   \n",
       "297401     KCNH2        Yes  rs1805123  chr7  150948446   T   G    Homozygous   \n",
       "297402     KCNH2        Yes  rs1805123  chr7  150948446   T   G    Homozygous   \n",
       "297403     KCNH2        Yes  rs1805123  chr7  150948446   T   G    Homozygous   \n",
       "297404     KCNH2        Yes  rs1805123  chr7  150948446   T   G    Homozygous   \n",
       "\n",
       "                                              Consequence Consequence_score  \\\n",
       "15189                                 3 prime UTR variant              3/10   \n",
       "15190                             downstream gene variant              2/10   \n",
       "15191                             downstream gene variant              2/10   \n",
       "23277                                    missense variant              7/10   \n",
       "23278                                    missense variant              7/10   \n",
       "...                                                   ...               ...   \n",
       "297400                            downstream gene variant              2/10   \n",
       "297401                            downstream gene variant              2/10   \n",
       "297402                            downstream gene variant              2/10   \n",
       "297403                            downstream gene variant              2/10   \n",
       "297404  splice region variant,non coding transcript ex...              4/10   \n",
       "\n",
       "          IMPACT  IMPACT_score  \\\n",
       "15189   MODIFIER           1.5   \n",
       "15190   MODIFIER           1.5   \n",
       "15191   MODIFIER           1.5   \n",
       "23277   MODERATE           5.0   \n",
       "23278   MODERATE           5.0   \n",
       "...          ...           ...   \n",
       "297400  MODIFIER           1.5   \n",
       "297401  MODIFIER           1.5   \n",
       "297402  MODIFIER           1.5   \n",
       "297403  MODIFIER           1.5   \n",
       "297404       LOW           2.5   \n",
       "\n",
       "                                            ClinVar_CLNDN  \\\n",
       "15189   Caudal regression sequence,Catecholaminergic p...   \n",
       "15190   Caudal regression sequence,Catecholaminergic p...   \n",
       "15191   Caudal regression sequence,Catecholaminergic p...   \n",
       "23277   Dilated Cardiomyopathy, Dominant,Cardiovascula...   \n",
       "23278   Dilated Cardiomyopathy, Dominant,Cardiovascula...   \n",
       "...                                                   ...   \n",
       "297400  Cardiac arrhythmia,Cardiovascular phenotype,At...   \n",
       "297401  Cardiac arrhythmia,Cardiovascular phenotype,At...   \n",
       "297402  Cardiac arrhythmia,Cardiovascular phenotype,At...   \n",
       "297403  Cardiac arrhythmia,Cardiovascular phenotype,At...   \n",
       "297404  Cardiac arrhythmia,Cardiovascular phenotype,At...   \n",
       "\n",
       "                    CLIN_SIG  \\\n",
       "15189   benign,likely benign   \n",
       "15190   benign,likely benign   \n",
       "15191   benign,likely benign   \n",
       "23277   benign,likely benign   \n",
       "23278   benign,likely benign   \n",
       "...                      ...   \n",
       "297400                benign   \n",
       "297401                benign   \n",
       "297402                benign   \n",
       "297403                benign   \n",
       "297404                benign   \n",
       "\n",
       "                                       ClinVar_CLNREVSTAT ClinVar  \\\n",
       "15189                 criteria provided, single submitter  292122   \n",
       "15190                 criteria provided, single submitter  292122   \n",
       "15191                 criteria provided, single submitter  292122   \n",
       "23277   criteria provided, multiple submitters, no con...   43669   \n",
       "23278   criteria provided, multiple submitters, no con...   43669   \n",
       "...                                                   ...     ...   \n",
       "297400  criteria provided, multiple submitters, no con...   67427   \n",
       "297401  criteria provided, multiple submitters, no con...   67427   \n",
       "297402  criteria provided, multiple submitters, no con...   67427   \n",
       "297403  criteria provided, multiple submitters, no con...   67427   \n",
       "297404  criteria provided, multiple submitters, no con...   67427   \n",
       "\n",
       "                     HGVSc HGVSc (Transcript)              HGVSp  \\\n",
       "15189    ENST00000261448.6          c.*482C>T                      \n",
       "15190                                    None                      \n",
       "15191                                    None                      \n",
       "23277   ENST00000236918.11           c.788A>G  ENSP00000236918.8   \n",
       "23278    ENST00000360372.8           c.659A>G  ENSP00000353535.5   \n",
       "...                    ...                ...                ...   \n",
       "297400                                   None                      \n",
       "297401                                   None                      \n",
       "297402                                   None                      \n",
       "297403                                   None                      \n",
       "297404   ENST00000684241.1          n.3523A>C                      \n",
       "\n",
       "       HGVSp (Transcript)   GT  GQ SDP  DP  RD  AD    FREQ        PVAL RDF  \\\n",
       "15189                None  0/1  52  26  26  12  14  53.85%   5.4595E-6  10   \n",
       "15190                None  0/1  52  26  26  12  14  53.85%   5.4595E-6  10   \n",
       "15191                None  0/1  52  26  26  12  14  53.85%   5.4595E-6  10   \n",
       "23277         p.Lys263Arg  0/1  68  32  32  14  18  56.25%   1.3089E-7  14   \n",
       "23278         p.Lys220Arg  0/1  68  32  32  14  18  56.25%   1.3089E-7  14   \n",
       "...                   ...  ...  ..  ..  ..  ..  ..     ...         ...  ..   \n",
       "297400               None  1/1  93  18  18   0  17  94.44%  4.2852E-10   0   \n",
       "297401               None  1/1  93  18  18   0  17  94.44%  4.2852E-10   0   \n",
       "297402               None  1/1  93  18  18   0  17  94.44%  4.2852E-10   0   \n",
       "297403               None  1/1  93  18  18   0  17  94.44%  4.2852E-10   0   \n",
       "297404               None  1/1  93  18  18   0  17  94.44%  4.2852E-10   0   \n",
       "\n",
       "       RDR ADF ADR                              SIFT       PolyPhen  \\\n",
       "15189    2  11   3                                                    \n",
       "15190    2  11   3                                                    \n",
       "15191    2  11   3                                                    \n",
       "23277    0  12   6  deleterious low confidence(0.04)  benign(0.007)   \n",
       "23278    0  12   6                 deleterious(0.02)  benign(0.388)   \n",
       "...     ..  ..  ..                               ...            ...   \n",
       "297400   0  12   5                                                    \n",
       "297401   0  12   5                                                    \n",
       "297402   0  12   5                                                    \n",
       "297403   0  12   5                                                    \n",
       "297404   0  12   5                                                    \n",
       "\n",
       "                BIOTYPE   EXON INTRON Protein Position and Amino Acid  \\\n",
       "15189    protein coding  11/11                                    nan   \n",
       "15190    protein coding                                           nan   \n",
       "15191    protein coding                                           nan   \n",
       "23277    protein coding  14/16                                  K263R   \n",
       "23278    protein coding  12/14                                  K220R   \n",
       "...                 ...    ...    ...                             ...   \n",
       "297400  retained intron                                           nan   \n",
       "297401  retained intron                                           nan   \n",
       "297402  retained intron                                           nan   \n",
       "297403  retained intron                                           nan   \n",
       "297404  retained intron   9/13                                    nan   \n",
       "\n",
       "         Codons STRAND                                             PUBMED  \\\n",
       "15189               -1                27899944,24444446,31053292,32765588   \n",
       "15190                1                27899944,24444446,31053292,32765588   \n",
       "15191                1                27899944,24444446,31053292,32765588   \n",
       "23277   aAg/aGg     -1  25741868,24033266,18414213,23861362,15542288,2...   \n",
       "23278   aAg/aGg     -1  25741868,24033266,18414213,23861362,15542288,2...   \n",
       "...         ...    ...                                                ...   \n",
       "297400              -1  25741868,21056700,24033266,20850564,23820649,2...   \n",
       "297401              -1  25741868,21056700,24033266,20850564,23820649,2...   \n",
       "297402              -1  25741868,21056700,24033266,20850564,23820649,2...   \n",
       "297403              -1  25741868,21056700,24033266,20850564,23820649,2...   \n",
       "297404              -1  25741868,21056700,24033266,20850564,23820649,2...   \n",
       "\n",
       "       Literature  \n",
       "15189         Yes  \n",
       "15190         Yes  \n",
       "15191         Yes  \n",
       "23277         Yes  \n",
       "23278         Yes  \n",
       "...           ...  \n",
       "297400        Yes  \n",
       "297401        Yes  \n",
       "297402        Yes  \n",
       "297403        Yes  \n",
       "297404        Yes  \n",
       "\n",
       "[79 rows x 42 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3[merged_3['Literature'] == 'Yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "b971f36b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene_Match</th>\n",
       "      <th>rsID</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>Literature</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Consequence_score</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>IMPACT_score</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSc (Transcript)</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp (Transcript)</th>\n",
       "      <th>GT</th>\n",
       "      <th>GQ</th>\n",
       "      <th>SDP</th>\n",
       "      <th>DP</th>\n",
       "      <th>RD</th>\n",
       "      <th>AD</th>\n",
       "      <th>FREQ</th>\n",
       "      <th>PVAL</th>\n",
       "      <th>RDF</th>\n",
       "      <th>RDR</th>\n",
       "      <th>ADF</th>\n",
       "      <th>ADR</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>Protein Position and Amino Acid</th>\n",
       "      <th>Codons</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>PUBMED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td>3/10</td>\n",
       "      <td>LOW</td>\n",
       "      <td>2.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.243A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Ser81%3D</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>S81</td>\n",
       "      <td>tcA/tcG</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs201219564</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69270</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>regulatory region variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>127</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>100</td>\n",
       "      <td>38</td>\n",
       "      <td>27.54%</td>\n",
       "      <td>1.8855E-13</td>\n",
       "      <td>86</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>TF binding site</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OR4F5</td>\n",
       "      <td>No</td>\n",
       "      <td>rs2691305</td>\n",
       "      <td>chr1</td>\n",
       "      <td>69511</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>c.484A&gt;G</td>\n",
       "      <td>ENSP00000493376.2</td>\n",
       "      <td>p.Thr162Ala</td>\n",
       "      <td>1/1</td>\n",
       "      <td>255</td>\n",
       "      <td>114</td>\n",
       "      <td>114</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.392E-68</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>98</td>\n",
       "      <td>16</td>\n",
       "      <td>tolerated(0.92)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>3/3</td>\n",
       "      <td></td>\n",
       "      <td>T162A</td>\n",
       "      <td>Aca/Gca</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>7/10</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>5.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000426406.4</td>\n",
       "      <td>c.389T&gt;G</td>\n",
       "      <td>ENSP00000409316.1</td>\n",
       "      <td>p.Leu130Arg</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>deleterious low confidence(0.01)</td>\n",
       "      <td>benign(0.336)</td>\n",
       "      <td>protein coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>L130R</td>\n",
       "      <td>cTc/cGc</td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>No</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>451290</td>\n",
       "      <td>No</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>intron variant,non coding transcript variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>ENST00000455207.5</td>\n",
       "      <td>n.169+33750T&gt;G</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>0/1</td>\n",
       "      <td>56</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>47</td>\n",
       "      <td>17</td>\n",
       "      <td>26.56%</td>\n",
       "      <td>2.243E-6</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td>1/2</td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328329</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328330</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328331</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328332</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>328333</th>\n",
       "      <td>EIF1AY</td>\n",
       "      <td>No</td>\n",
       "      <td>rs9786153</td>\n",
       "      <td>chrY</td>\n",
       "      <td>20577481</td>\n",
       "      <td>No</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Homozygous</td>\n",
       "      <td>upstream gene variant</td>\n",
       "      <td>2/10</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1.5</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>1/1</td>\n",
       "      <td>123</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>100%</td>\n",
       "      <td>4.7526E-13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>nan</td>\n",
       "      <td></td>\n",
       "      <td>-1</td>\n",
       "      <td>22271044,29391530,24262073,15896936,32996047,3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>328334 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Name Gene_Match         rsID CHROM       POS Literature REF ALT  \\\n",
       "0          OR4F5         No  rs201219564  chr1     69270         No   A   G   \n",
       "1          OR4F5         No  rs201219564  chr1     69270         No   A   G   \n",
       "2          OR4F5         No    rs2691305  chr1     69511         No   A   G   \n",
       "3                        No            .  chr1    451290         No   A   C   \n",
       "4                        No            .  chr1    451290         No   A   C   \n",
       "...          ...        ...          ...   ...       ...        ...  ..  ..   \n",
       "328329    EIF1AY         No    rs9786153  chrY  20577481         No   C   T   \n",
       "328330    EIF1AY         No    rs9786153  chrY  20577481         No   C   T   \n",
       "328331    EIF1AY         No    rs9786153  chrY  20577481         No   C   T   \n",
       "328332    EIF1AY         No    rs9786153  chrY  20577481         No   C   T   \n",
       "328333    EIF1AY         No    rs9786153  chrY  20577481         No   C   T   \n",
       "\n",
       "            Zygosity                                   Consequence  \\\n",
       "0       Heterozygous                            synonymous variant   \n",
       "1       Heterozygous                     regulatory region variant   \n",
       "2         Homozygous                              missense variant   \n",
       "3       Heterozygous                              missense variant   \n",
       "4       Heterozygous  intron variant,non coding transcript variant   \n",
       "...              ...                                           ...   \n",
       "328329    Homozygous                         upstream gene variant   \n",
       "328330    Homozygous                         upstream gene variant   \n",
       "328331    Homozygous                         upstream gene variant   \n",
       "328332    Homozygous                         upstream gene variant   \n",
       "328333    Homozygous                         upstream gene variant   \n",
       "\n",
       "       Consequence_score    IMPACT  IMPACT_score ClinVar_CLNDN CLIN_SIG  \\\n",
       "0                   3/10       LOW           2.5                          \n",
       "1                   2/10  MODIFIER           1.5                          \n",
       "2                   7/10  MODERATE           5.0                          \n",
       "3                   7/10  MODERATE           5.0                          \n",
       "4                   2/10  MODIFIER           1.5                          \n",
       "...                  ...       ...           ...           ...      ...   \n",
       "328329              2/10  MODIFIER           1.5                          \n",
       "328330              2/10  MODIFIER           1.5                          \n",
       "328331              2/10  MODIFIER           1.5                          \n",
       "328332              2/10  MODIFIER           1.5                          \n",
       "328333              2/10  MODIFIER           1.5                          \n",
       "\n",
       "       ClinVar_CLNREVSTAT ClinVar              HGVSc HGVSc (Transcript)  \\\n",
       "0                                  ENST00000641515.2           c.243A>G   \n",
       "1                                                                  None   \n",
       "2                                  ENST00000641515.2           c.484A>G   \n",
       "3                                  ENST00000426406.4           c.389T>G   \n",
       "4                                  ENST00000455207.5     n.169+33750T>G   \n",
       "...                   ...     ...                ...                ...   \n",
       "328329                                                             None   \n",
       "328330                                                             None   \n",
       "328331                                                             None   \n",
       "328332                                                             None   \n",
       "328333                                                             None   \n",
       "\n",
       "                    HGVSp HGVSp (Transcript)   GT   GQ  SDP   DP   RD   AD  \\\n",
       "0       ENSP00000493376.2         p.Ser81%3D  0/1  127  138  138  100   38   \n",
       "1                                       None  0/1  127  138  138  100   38   \n",
       "2       ENSP00000493376.2        p.Thr162Ala  1/1  255  114  114    0  114   \n",
       "3       ENSP00000409316.1        p.Leu130Arg  0/1   56   64   64   47   17   \n",
       "4                                       None  0/1   56   64   64   47   17   \n",
       "...                   ...                ...  ...  ...  ...  ...  ...  ...   \n",
       "328329                                  None  1/1  123   22   22    0   22   \n",
       "328330                                  None  1/1  123   22   22    0   22   \n",
       "328331                                  None  1/1  123   22   22    0   22   \n",
       "328332                                  None  1/1  123   22   22    0   22   \n",
       "328333                                  None  1/1  123   22   22    0   22   \n",
       "\n",
       "          FREQ        PVAL RDF RDR ADF ADR                              SIFT  \\\n",
       "0       27.54%  1.8855E-13  86  14  29   9                                     \n",
       "1       27.54%  1.8855E-13  86  14  29   9                                     \n",
       "2         100%   4.392E-68   0   0  98  16                   tolerated(0.92)   \n",
       "3       26.56%    2.243E-6  33  14  15   2  deleterious low confidence(0.01)   \n",
       "4       26.56%    2.243E-6  33  14  15   2                                     \n",
       "...        ...         ...  ..  ..  ..  ..                               ...   \n",
       "328329    100%  4.7526E-13   0   0  14   8                                     \n",
       "328330    100%  4.7526E-13   0   0  14   8                                     \n",
       "328331    100%  4.7526E-13   0   0  14   8                                     \n",
       "328332    100%  4.7526E-13   0   0  14   8                                     \n",
       "328333    100%  4.7526E-13   0   0  14   8                                     \n",
       "\n",
       "             PolyPhen          BIOTYPE EXON INTRON  \\\n",
       "0                       protein coding  3/3          \n",
       "1                      TF binding site               \n",
       "2           benign(0)   protein coding  3/3          \n",
       "3       benign(0.336)   protein coding  1/1          \n",
       "4                               lncRNA         1/2   \n",
       "...               ...              ...  ...    ...   \n",
       "328329                          lncRNA               \n",
       "328330                          lncRNA               \n",
       "328331                          lncRNA               \n",
       "328332                          lncRNA               \n",
       "328333                          lncRNA               \n",
       "\n",
       "       Protein Position and Amino Acid   Codons STRAND  \\\n",
       "0                                  S81  tcA/tcG      1   \n",
       "1                                  nan                   \n",
       "2                                T162A  Aca/Gca      1   \n",
       "3                                L130R  cTc/cGc     -1   \n",
       "4                                  nan              -1   \n",
       "...                                ...      ...    ...   \n",
       "328329                             nan              -1   \n",
       "328330                             nan              -1   \n",
       "328331                             nan              -1   \n",
       "328332                             nan              -1   \n",
       "328333                             nan              -1   \n",
       "\n",
       "                                                   PUBMED  \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "3                                                          \n",
       "4                                                          \n",
       "...                                                   ...  \n",
       "328329  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328330  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328331  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328332  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "328333  22271044,29391530,24262073,15896936,32996047,3...  \n",
       "\n",
       "[328334 rows x 42 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_3 = merged_3[['Gene Name', 'Gene_Match', 'rsID', 'CHROM', 'POS', 'Literature', 'REF', 'ALT', 'Zygosity',\n",
    "       'Consequence','Consequence_score', 'IMPACT', 'IMPACT_score', 'ClinVar_CLNDN', 'CLIN_SIG',\n",
    "       'ClinVar_CLNREVSTAT', 'ClinVar', 'HGVSc', 'HGVSc (Transcript)', 'HGVSp',\n",
    "       'HGVSp (Transcript)', 'GT', 'GQ', 'SDP', 'DP', 'RD', 'AD', 'FREQ',\n",
    "       'PVAL', 'RDF', 'RDR', 'ADF', 'ADR', 'SIFT', 'PolyPhen', 'BIOTYPE',\n",
    "       'EXON', 'INTRON', 'Protein Position and Amino Acid', 'Codons', 'STRAND',\n",
    "       'PUBMED']]\n",
    "merged_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "63c7d933",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_3.to_excel(r'C:/Users/GenepoweRx_Madhu/Downloads/Processed_vcf_files/KHCDPRGPTTL6_depth_vcf_processed.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1bcb1295",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
